{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "aab540c1-4038-4035-97ff-5bc89b9837f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "### Imports\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import pyranges as pr\n",
    "import os\n",
    "import string\n",
    "import pysam\n",
    "#import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7c49818c-2b48-4a55-a568-dbbc80cdc744",
   "metadata": {},
   "outputs": [],
   "source": [
    "# def hb_maker(phased_vars_file):\n",
    "#     ### Read the filtered vcf file (filtered on ROI of the OMIM genes & QUAL >= 40) and get the desired columns\n",
    "#     df_vcf = pd.read_csv(phased_vars_file, sep=\"\\t\",comment='#', names=['CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT', \"sample\"])\n",
    "    \n",
    "#     df_vcf[\"GENE\"] = df_vcf['INFO'].str.split(\"|\").str[4]\n",
    "#     df_splitted = df_vcf['sample'].str.split(\":\")\n",
    "#     df_vcf[\"GT\"] = df_splitted.str[0]\n",
    "#     df_vcf[\"DP\"] = df_splitted.str[2]\n",
    "#     df_vcf[\"PS\"] = df_splitted.str[5]\n",
    "\n",
    "#     ### Only keep the phased variants (the variants with a phase tag)\n",
    "#     phased = df_vcf.loc[df_vcf['PS'].notna(), ['CHROM','POS','QUAL','GENE','GT','DP','PS','REF','ALT']]\n",
    "\n",
    "#     ### Group by phase tag and get the min, max, and amount of variants per haploblock\n",
    "#     grouped = phased.groupby(['PS']).agg({'POS': ['min','max', 'count'],\n",
    "#                                                     'CHROM': 'first',\n",
    "#                                                     'QUAL': 'mean',\n",
    "#                                                     'GENE': 'unique'}).reset_index()\n",
    "\n",
    "#     ### Change the column names, type, and add desired columns\n",
    "#     grouped.columns = ['_'.join(col).rstrip('_') if isinstance(col, tuple) else col\n",
    "#                    for col in grouped.columns]\n",
    "#     grouped.rename(columns={\"PS\":\"PS_tag\", \"POS_min\":\"START_HB\", \"POS_max\": \"END_HB\", \"POS_count\":\"informative_variants\", \"CHROM_first\":\"chromosome\", \"GENE_unique\":\"GENES_in_HB\"}, inplace=True)\n",
    "#     grouped[\"PS_tag\"] = grouped[\"PS_tag\"].astype(int)\n",
    "#     grouped[\"HB_length\"]=grouped[\"END_HB\"]-grouped[\"START_HB\"]\n",
    "#     grouped[\"total_VARS\"] = None\n",
    "#     return grouped\n",
    "\n",
    "\n",
    "def hb_maker(phased_vars_file, bed_f):\n",
    "    ### Read the filtered vcf file (filtered on ROI of the OMIM genes & QUAL >= 40) and get the desired columns\n",
    "    df_vcf = pd.read_csv(phased_vars_file, sep=\"\\t\", comment='#', names=['CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT', \"sample\"])\n",
    "    df_splitted = df_vcf['sample'].str.split(\":\")\n",
    "    df_vcf[\"GT\"] = df_splitted.str[0]\n",
    "    df_vcf[\"GQ\"] = df_splitted.str[1]\n",
    "    df_vcf[\"DP\"] = df_splitted.str[2]\n",
    "    df_vcf[\"PS\"] = df_splitted.str[5]\n",
    "\n",
    "    ### Only keep the phased variants (the variants with a phase tag)\n",
    "    phased = df_vcf.loc[df_vcf['PS'].notna(), ['CHROM','POS','QUAL','GT','GQ','DP','PS','REF','ALT']]\n",
    "    ### Group by phase tag and get the min, max, and amount of variants per haploblock\n",
    "    phased['POS'] = pd.to_numeric(phased['POS'], errors='coerce')\n",
    "    phased['GQ'] = pd.to_numeric(phased['GQ'], errors='coerce')\n",
    "\n",
    "    phased = create_sub_ps(phased, bed_f)\n",
    "    return phased\n",
    "    # grouped = phased.groupby(['PS_R']).agg({'POS': ['min','max', 'count'],\n",
    "    #                                                 'Chromosome': 'first',\n",
    "    #                                                 'GQ': 'mean'}).reset_index()\n",
    "\n",
    "    # ### Change the column names, type, and add desired columns\n",
    "    # grouped.columns = ['_'.join(col).rstrip('_') if isinstance(col, tuple) else col for col in grouped.columns]\n",
    "    # grouped.rename(columns={\"POS_min\":\"START_HB\", \"POS_max\": \"END_HB\", \"POS_count\":\"phased_variants\", \"Chromosome_first\":\"chromosome\"}, inplace=True)\n",
    "    # #grouped[\"PS_tag\"] = grouped[\"PS_tag\"].astype(int)\n",
    "    # grouped[\"HB_length\"]=grouped[\"END_HB\"]-grouped[\"START_HB\"]\n",
    "    # grouped[\"total_VARS\"] = None\n",
    "    # grouped[\"PS_tag\"] = grouped[\"chromosome\"].astype(str)+\"_\"+grouped[\"PS_R\"].astype(str)\n",
    "    # return grouped\n",
    "\n",
    "\n",
    "def run_switch(haploblock_table, all_vars, sample, version, ph_vars):\n",
    "    ### A counter to check how many haploblocks are already compared to BM\n",
    "    cnt = 0 \n",
    "\n",
    "    ### Variables and Files for the output\n",
    "    folder = \"/hpc/umc_laat/gvandersluis/data/\"    \n",
    "    BM_vars = f\"{folder}Ont_data_nhung/HG00{sample}/HG00{sample}_BM_SSANDT_rn.vcf\"\n",
    "    BM_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/ROI_eval.tsv\"\n",
    "    open(BM_ROI, \"w\").close()\n",
    "    switches_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/switches_ROI.bed\"\n",
    "    open(switches_ROI, \"w\").close()\n",
    "\n",
    "    ### Loop through every haploblock\n",
    "    for idx, item in haploblock_table.iterrows():\n",
    "        ### Get the haploblock region\n",
    "        reg = item[\"chromosome\"]+\":\"+str(item[\"START_HB\"])+\"-\"+str(item[\"END_HB\"])\n",
    "        print(reg)\n",
    "        ### ADD TOTAL VARIANTS in the region from the unfiltered vcf file\n",
    "        cmd = f\"apptainer exec -B /hpc/:/hpc/ /hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif bcftools view -H -r {reg} {all_vars} | wc -l\"\n",
    "        ### Run the command\n",
    "        result = subprocess.run(cmd, shell=True, check=True, capture_output=True, text=True)\n",
    "        ### Get the count as integer and add the value to the column\n",
    "        count = int(result.stdout.strip())\n",
    "        grouped.loc[idx, \"total_VARS\"] = count\n",
    "        print(count, \"HII\")\n",
    "        \n",
    "        ### WHATSHAP COMPARE\n",
    "        PStag = str(item[\"PS_tag\"])\n",
    "        if \"NaN\" not in reg:\n",
    "            ### Prepare the benchmark file for whatshap compare\n",
    "            ### make an output file for the filtered Benchmark file (filtered on region of interest) and run that code + index the vcf\n",
    "            out_vcf = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{PStag}_BM.vcf\"\n",
    "            cmd = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", reg, BM_vars, \"-Oz\", \"-o\", out_vcf]\n",
    "            subprocess.run(cmd, check=True)\n",
    "            indexer = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"index\", \"-t\", out_vcf]\n",
    "            subprocess.run(indexer, check=True)\n",
    "\n",
    "            ### Make specific region version of the phased vcf file (Prepare file for whatshap compare)\n",
    "            out_ph_vars_vcf = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{PStag}_phased_vars.vcf\"\n",
    "            cmd_1 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", reg, ph_vars, \"-Oz\", \"-o\", out_ph_vars_vcf]\n",
    "            subprocess.run(cmd_1, check=True)\n",
    "            indexer_1 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"index\", \"-t\", out_ph_vars_vcf]\n",
    "            subprocess.run(indexer_1, check=True)\n",
    "\n",
    "            ### Prepare output files for WHATSHAP compare\n",
    "            indexed = out_vcf+\".tbi\"\n",
    "            indexed_1 = out_ph_vars_vcf+\".tbi\"\n",
    "            tsv_out = f\"{PStag}_eval.tsv\"\n",
    "            open(tsv_out, \"a+\").write(\"\")\n",
    "\n",
    "            ### If the output vcf file has variants (If there are variants found in the benchmark file of this particular region, then:)\n",
    "            if sum(1 for _ in pysam.VariantFile(out_vcf)) != 0:\n",
    "                ### Make file for the switch error bed file\n",
    "                bed_out = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{PStag}_switch.bed\"\n",
    "                ### RUN WHATSHAP COMPARE\n",
    "                cmd2 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/whatshap_v1.sif\", \"whatshap\", \"compare\",\"--switch-error-bed\",bed_out,\"--tsv-pairwise\", tsv_out, \"--names\", f\"BENCHMARK,{PStag}\", out_vcf, out_ph_vars_vcf]\n",
    "                subprocess.run(cmd2, check=True, capture_output=True)\n",
    "\n",
    "                ### If the whatsap compare output file is empty (first run), then write output\n",
    "                if os.path.getsize(BM_ROI) == 0:\n",
    "                    with open(BM_ROI, \"w\") as out:\n",
    "                        with open(tsv_out, \"r\") as inp:\n",
    "                            out.write(inp.read())\n",
    "                ### If whatshap compare output file is not empty anymore (every run after the first), then append output\n",
    "                else:\n",
    "                    with open(BM_ROI, \"a\") as out:\n",
    "                        with open(tsv_out, \"r\") as inp:\n",
    "                            next(inp) # Skip header\n",
    "                            out.write(inp.read())\n",
    "                ### Write or append switch error locations to the bed file\n",
    "                with open(switches_ROI, \"a+\") as out_b:\n",
    "                    with open(bed_out, \"r\") as inp_b:\n",
    "                        out_b.write(inp_b.read())\n",
    "            ### REMOVE temporary needed files\n",
    "                os.remove(bed_out)\n",
    "            os.remove(out_vcf)\n",
    "            os.remove(indexed)\n",
    "            os.remove(out_ph_vars_vcf)\n",
    "            os.remove(indexed_1)\n",
    "            os.remove(tsv_out)\n",
    "        print(PStag, \"\\t Done\")\n",
    "        \n",
    "        cnt += 1\n",
    "        print(cnt)\n",
    "    return BM_ROI, grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "55ebd74a-af9e-4e76-bd2c-c748c41347f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def create_sub_ps(variants, bed_file):\n",
    "    ### Convert to BED-like intervals\n",
    "    variants[\"start\"] = variants[\"POS\"] - 1\n",
    "    variants[\"end\"] = variants[\"POS\"]\n",
    "    ### Load Bed file \n",
    "    bed = pr.read_bed(bed_file)\n",
    "    ### Convert variants to Pyranges\n",
    "    var_pr = pr.PyRanges(variants.rename(columns={\n",
    "            \"CHROM\": \"Chromosome\",\n",
    "            \"start\": \"Start\",\n",
    "            \"end\": \"End\"}))\n",
    "    ### Intersect with BED\n",
    "    overlap = var_pr.join(bed, how=\"left\")\n",
    "    df = overlap.df\n",
    "    print(\"HIHIHIHIHHI\\n\",df)\n",
    "    ### Define ROI by coordinates\n",
    "    df[\"ROI\"] = (df[\"Chromosome\"].astype(str) + \":\" + df[\"Start_b\"].astype(str) + \"-\" + df[\"End_b\"].astype(str))\n",
    "    print(\"AGAIN\\n\", df)\n",
    "    ### assign the suffixes\n",
    "    df[\"PS_R\"] = split_ps_tags(df)\n",
    "    return df\n",
    "\n",
    "def split_ps_tags(df):\n",
    "    ### This function adds A B C etc to sub-haploblocks (haploblocks that are defined by multiple ROIs)\n",
    "    new_ps = [\"\"] * len(df)\n",
    "    print(df)\n",
    "    for ps, group in df.groupby(\"PS\", dropna=False):\n",
    "        print(ps)\n",
    "        print(group)\n",
    "        rois = group[\"ROI\"].dropna().unique()\n",
    "        print(rois)\n",
    "        ### Single or no ROI → PS unchanged\n",
    "        if len(rois) <= 1:\n",
    "            print(group.index)\n",
    "            for idx in group.index:\n",
    "                print(idx)\n",
    "                print(ps)\n",
    "                new_ps[idx] = ps\n",
    "            continue\n",
    "        ### Multiple ROIs → assign suffixes A, B, C...\n",
    "        roi_to_suffix = {\n",
    "            roi: string.ascii_uppercase[i]\n",
    "            for i, roi in enumerate(sorted(rois))}\n",
    "        for idx, row in group.iterrows():\n",
    "            if pd.isna(row[\"ROI\"]):\n",
    "                new_ps[idx] = ps\n",
    "            else:\n",
    "                new_ps[idx] = f\"{ps}.{roi_to_suffix[row['ROI']]}\"\n",
    "    return new_ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c35e5870-dbbd-49f6-9635-c9e8bf3540d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def excel_style_suffix(n):\n",
    "    print(n)\n",
    "    result = \"\"\n",
    "    while n > 0:\n",
    "        n, remainder = divmod(n - 1, 26)\n",
    "        result = string.ascii_uppercase[remainder] + result\n",
    "    return result\n",
    "\n",
    "\n",
    "def split_ps_tags(df):\n",
    "    df[\"PS_R\"]= \"None\"\n",
    "    for ps, group in df.groupby(\"PS\", dropna=False):\n",
    "        chrm = group[\"CHROM\"].unique()[0]\n",
    "        rois = group[\"ROI\"].dropna().unique()\n",
    "        if len(rois) <=1:\n",
    "            df.loc[df[\"PS\"]==ps, \"PS_R\"] = chrm+\"_\"+ps\n",
    "        else:\n",
    "            roi_to_suffix = {\n",
    "                roi: excel_style_suffix(i)\n",
    "                for i, roi in enumerate(sorted(rois))}\n",
    "            for idx, row in group.iterrows():\n",
    "                if row[\"ROI\"] == \"None\":\n",
    "                    df.loc[idx, \"PS_R\"] = ps\n",
    "                else:\n",
    "                    df.loc[idx, \"PS_R\"] = f\"{chrm}_{ps}.{roi_to_suffix[row['ROI']]}\"\n",
    "    return df\n",
    "\n",
    "\n",
    "def create_sub_ps(variants, bed_file):\n",
    "    variants[\"ROI\"]=\"None\"\n",
    "    bdf = open(bed_file, \"r\").readlines()\n",
    "    for loc in bdf:\n",
    "        chrom = loc.strip(\"\\n\").split(\"\\t\")[0]\n",
    "        start = int(loc.strip(\"\\n\").split(\"\\t\")[1])\n",
    "        stop = int(loc.strip(\"\\n\").split(\"\\t\")[2])\n",
    "        variants.loc[(variants[\"CHROM\"].astype(str) == chrom) & (variants[\"POS\"].astype(int) >= start) & (variants[\"POS\"].astype(int) <= stop), \"ROI\"] = chrom+\":\"+str(start)+\"-\"+str(stop)\n",
    "    variants[\"PS_R\"] = split_ps_tags(variants)\n",
    "    return variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9e78426e-eba2-43a6-8bcd-cbb52882b1fd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        CHROM        POS   QUAL   GT  GQ  DP         PS REF ALT  \\\n",
      "3       chr12      10183   8.77  1|0   8  52      10183   T   A   \n",
      "4       chr12      10185   7.52  1|0   7  52      10183   C   A   \n",
      "5       chr12      10194  10.42  1|0  10  52      10183   A   T   \n",
      "17      chr12      10491   5.05  1|0   5  55      10183   A   T   \n",
      "18      chr12      10649  14.34  1|0  14  71      10183   G   A   \n",
      "...       ...        ...    ...  ...  ..  ..        ...  ..  ..   \n",
      "268495  chr12  133241504  54.35  0|1  54  44  133158779   A   G   \n",
      "268496  chr12  133244280  58.20  0|1  58  47  133158779   G   A   \n",
      "268498  chr12  133249425  53.87  0|1  53  49  133158779   G   T   \n",
      "268499  chr12  133253066  55.29  0|1  55  49  133158779   G   T   \n",
      "268505  chr12  133264934  11.47  1|0  11  44  133158779   T   C   \n",
      "\n",
      "                              ROI  \n",
      "3                chr12:1-11895386  \n",
      "4                chr12:1-11895386  \n",
      "5                chr12:1-11895386  \n",
      "17               chr12:1-11895386  \n",
      "18               chr12:1-11895386  \n",
      "...                           ...  \n",
      "268495  chr12:128308614-133275309  \n",
      "268496  chr12:128308614-133275309  \n",
      "268498  chr12:128308614-133275309  \n",
      "268499  chr12:128308614-133275309  \n",
      "268505  chr12:128308614-133275309  \n",
      "\n",
      "[124679 rows x 10 columns]\n",
      "101571074\n",
      "        CHROM        POS   QUAL   GT  GQ  DP         PS REF ALT  \\\n",
      "210805  chr12  101571074  58.71  0|1  58  48  101571074   T   C   \n",
      "210806  chr12  101571172  67.78  0|1  67  48  101571074   G   A   \n",
      "210807  chr12  101572235  61.99  0|1  61  43  101571074   A   G   \n",
      "210810  chr12  101573710  62.33  0|1  62  46  101571074   C   T   \n",
      "210811  chr12  101575664  63.92  0|1  63  55  101571074   A   G   \n",
      "...       ...        ...    ...  ...  ..  ..        ...  ..  ..   \n",
      "211452  chr12  101996979  60.26  0|1  60  57  101571074   C   T   \n",
      "211456  chr12  101998244  58.90  1|0  58  67  101571074   C   T   \n",
      "211458  chr12  102000705  65.09  0|1  65  61  101571074   C   T   \n",
      "211462  chr12  102021456  55.71  1|0  55  52  101571074   C   A   \n",
      "211463  chr12  102028122  63.00  0|1  63  55  101571074   A   G   \n",
      "\n",
      "                             ROI  \n",
      "210805  chr12:98032956-115177215  \n",
      "210806  chr12:98032956-115177215  \n",
      "210807  chr12:98032956-115177215  \n",
      "210810  chr12:98032956-115177215  \n",
      "210811  chr12:98032956-115177215  \n",
      "...                          ...  \n",
      "211452  chr12:98032956-115177215  \n",
      "211456  chr12:98032956-115177215  \n",
      "211458  chr12:98032956-115177215  \n",
      "211462  chr12:98032956-115177215  \n",
      "211463  chr12:98032956-115177215  \n",
      "\n",
      "[351 rows x 10 columns]\n",
      "['chr12:98032956-115177215']\n",
      "Index([210805, 210806, 210807, 210810, 210811, 210812, 210813, 210815, 210820,\n",
      "       210821,\n",
      "       ...\n",
      "       211395, 211399, 211401, 211405, 211434, 211452, 211456, 211458, 211462,\n",
      "       211463],\n",
      "      dtype='int64', length=351)\n",
      "210805\n",
      "101571074\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list assignment index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mIndexError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[15]\u001b[39m\u001b[32m, line 10\u001b[39m\n\u001b[32m      8\u001b[39m samp=\u001b[33m\"\u001b[39m\u001b[33m2\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m      9\u001b[39m vers=\u001b[33m\"\u001b[39m\u001b[33mOMIM\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m---> \u001b[39m\u001b[32m10\u001b[39m grouped = \u001b[43mhb_maker\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvariants\u001b[49m\u001b[43m,\u001b[49m\u001b[43mbed\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;66;03m# maar van all_variants de header version van de variants\u001b[39;00m\n\u001b[32m     11\u001b[39m grouped\n\u001b[32m     12\u001b[39m \u001b[38;5;66;03m# grouped.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{vers}_ROI/final_df.csv\", index=False)\u001b[39;00m\n\u001b[32m     13\u001b[39m \n\u001b[32m     14\u001b[39m \u001b[38;5;66;03m# grouped.sort_values([\"chromosome\", \"PS_tag\"]).head(20)\u001b[39;00m\n\u001b[32m     15\u001b[39m \u001b[38;5;66;03m# grouped\u001b[39;00m\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[12]\u001b[39m\u001b[32m, line 45\u001b[39m, in \u001b[36mhb_maker\u001b[39m\u001b[34m(phased_vars_file, bed_f)\u001b[39m\n\u001b[32m     42\u001b[39m phased[\u001b[33m'\u001b[39m\u001b[33mPOS\u001b[39m\u001b[33m'\u001b[39m] = pd.to_numeric(phased[\u001b[33m'\u001b[39m\u001b[33mPOS\u001b[39m\u001b[33m'\u001b[39m], errors=\u001b[33m'\u001b[39m\u001b[33mcoerce\u001b[39m\u001b[33m'\u001b[39m)\n\u001b[32m     43\u001b[39m phased[\u001b[33m'\u001b[39m\u001b[33mGQ\u001b[39m\u001b[33m'\u001b[39m] = pd.to_numeric(phased[\u001b[33m'\u001b[39m\u001b[33mGQ\u001b[39m\u001b[33m'\u001b[39m], errors=\u001b[33m'\u001b[39m\u001b[33mcoerce\u001b[39m\u001b[33m'\u001b[39m)\n\u001b[32m---> \u001b[39m\u001b[32m45\u001b[39m phased = \u001b[43mcreate_sub_ps\u001b[49m\u001b[43m(\u001b[49m\u001b[43mphased\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbed_f\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     46\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m phased\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[14]\u001b[39m\u001b[32m, line 12\u001b[39m, in \u001b[36mcreate_sub_ps\u001b[39m\u001b[34m(variants, bed_file)\u001b[39m\n\u001b[32m      9\u001b[39m     variants.loc[(variants[\u001b[33m\"\u001b[39m\u001b[33mCHROM\u001b[39m\u001b[33m\"\u001b[39m].astype(\u001b[38;5;28mstr\u001b[39m) == chrom) & (variants[\u001b[33m\"\u001b[39m\u001b[33mPOS\u001b[39m\u001b[33m\"\u001b[39m].astype(\u001b[38;5;28mint\u001b[39m) >= start) & (variants[\u001b[33m\"\u001b[39m\u001b[33mPOS\u001b[39m\u001b[33m\"\u001b[39m].astype(\u001b[38;5;28mint\u001b[39m) <= stop), \u001b[33m\"\u001b[39m\u001b[33mROI\u001b[39m\u001b[33m\"\u001b[39m] = chrom+\u001b[33m\"\u001b[39m\u001b[33m:\u001b[39m\u001b[33m\"\u001b[39m+\u001b[38;5;28mstr\u001b[39m(start)+\u001b[33m\"\u001b[39m\u001b[33m-\u001b[39m\u001b[33m\"\u001b[39m+\u001b[38;5;28mstr\u001b[39m(stop)\n\u001b[32m     10\u001b[39m \u001b[38;5;66;03m# print(variants)\u001b[39;00m\n\u001b[32m     11\u001b[39m \u001b[38;5;66;03m# variants = variants.reset_index(drop=True)\u001b[39;00m\n\u001b[32m---> \u001b[39m\u001b[32m12\u001b[39m variants[\u001b[33m\"\u001b[39m\u001b[33mPS_R\u001b[39m\u001b[33m\"\u001b[39m] = \u001b[43msplit_ps_tags\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvariants\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     13\u001b[39m \u001b[38;5;66;03m# print(variants)\u001b[39;00m\n\u001b[32m     14\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m variants\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[13]\u001b[39m\u001b[32m, line 38\u001b[39m, in \u001b[36msplit_ps_tags\u001b[39m\u001b[34m(df)\u001b[39m\n\u001b[32m     36\u001b[39m         \u001b[38;5;28mprint\u001b[39m(idx)\n\u001b[32m     37\u001b[39m         \u001b[38;5;28mprint\u001b[39m(ps)\n\u001b[32m---> \u001b[39m\u001b[32m38\u001b[39m         \u001b[43mnew_ps\u001b[49m\u001b[43m[\u001b[49m\u001b[43midx\u001b[49m\u001b[43m]\u001b[49m = ps\n\u001b[32m     39\u001b[39m     \u001b[38;5;28;01mcontinue\u001b[39;00m\n\u001b[32m     40\u001b[39m \u001b[38;5;66;03m### Multiple ROIs → assign suffixes A, B, C...\u001b[39;00m\n",
      "\u001b[31mIndexError\u001b[39m: list assignment index out of range"
     ]
    }
   ],
   "source": [
    "### RUN scripts\n",
    "\n",
    "# OMIM test set of 50000 variants\n",
    "variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/test_nh.vcf\"\n",
    "all_variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/SAMPLE_renamed.vcf.gz\"\n",
    "phased_variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/phased_ROI_HG2.vcf.gz\"\n",
    "bed=\"/hpc/umc_laat/gvandersluis/data/Ref_HG/HG_OMIM_ROI_merged.bed\"\n",
    "samp=\"2\"\n",
    "vers=\"OMIM\"\n",
    "grouped = hb_maker(variants,bed) # maar van all_variants de header version van de variants\n",
    "grouped\n",
    "# grouped.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{vers}_ROI/final_df.csv\", index=False)\n",
    "\n",
    "# grouped.sort_values([\"chromosome\", \"PS_tag\"]).head(20)\n",
    "# grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90a42c35-4de4-4e17-ae3d-132b1c388f17",
   "metadata": {},
   "outputs": [],
   "source": [
    "chrom = \"chr12\"\n",
    "start = \"1\"\n",
    "stop = \n",
    "\n",
    "variants.loc[(variants[\"CHROM\"].astype(str) == chrom) & (variants[\"POS\"].astype(str) >= start) & (variants[\"POS\"].astype(str) <= stop), \"ROI\"] = str(chrom+\":\"+start+\"-\"+stop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1987218f-f6b1-4ff4-93bb-c9d9411b184f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>PS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ROI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHROM, POS, QUAL, GT, GQ, DP, PS, REF, ALT, ROI]\n",
       "Index: []"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped[(grouped[\"PS\"] == '10183') & (grouped[\"ROI\"] ==\"None\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "88eeebf2-3f5c-48ef-ac8a-ccb2002cd55c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n",
      "0\n",
      "1\n",
      "2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>PS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ROI</th>\n",
       "      <th>PS_R</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10183</td>\n",
       "      <td>8.77</td>\n",
       "      <td>1|0</td>\n",
       "      <td>8</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "      <td>chr12_10183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10185</td>\n",
       "      <td>7.52</td>\n",
       "      <td>1|0</td>\n",
       "      <td>7</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "      <td>chr12_10183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10194</td>\n",
       "      <td>10.42</td>\n",
       "      <td>1|0</td>\n",
       "      <td>10</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "      <td>chr12_10183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10491</td>\n",
       "      <td>5.05</td>\n",
       "      <td>1|0</td>\n",
       "      <td>5</td>\n",
       "      <td>55</td>\n",
       "      <td>10183</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "      <td>chr12_10183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10649</td>\n",
       "      <td>14.34</td>\n",
       "      <td>1|0</td>\n",
       "      <td>14</td>\n",
       "      <td>71</td>\n",
       "      <td>10183</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "      <td>chr12_10183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268495</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133241504</td>\n",
       "      <td>54.35</td>\n",
       "      <td>0|1</td>\n",
       "      <td>54</td>\n",
       "      <td>44</td>\n",
       "      <td>133158779</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "      <td>chr12_133158779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268496</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133244280</td>\n",
       "      <td>58.20</td>\n",
       "      <td>0|1</td>\n",
       "      <td>58</td>\n",
       "      <td>47</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "      <td>chr12_133158779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268498</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133249425</td>\n",
       "      <td>53.87</td>\n",
       "      <td>0|1</td>\n",
       "      <td>53</td>\n",
       "      <td>49</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "      <td>chr12_133158779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268499</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133253066</td>\n",
       "      <td>55.29</td>\n",
       "      <td>0|1</td>\n",
       "      <td>55</td>\n",
       "      <td>49</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "      <td>chr12_133158779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268505</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133264934</td>\n",
       "      <td>11.47</td>\n",
       "      <td>1|0</td>\n",
       "      <td>11</td>\n",
       "      <td>44</td>\n",
       "      <td>133158779</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "      <td>chr12_133158779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>124679 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS   QUAL   GT  GQ  DP         PS REF ALT  \\\n",
       "3       chr12      10183   8.77  1|0   8  52      10183   T   A   \n",
       "4       chr12      10185   7.52  1|0   7  52      10183   C   A   \n",
       "5       chr12      10194  10.42  1|0  10  52      10183   A   T   \n",
       "17      chr12      10491   5.05  1|0   5  55      10183   A   T   \n",
       "18      chr12      10649  14.34  1|0  14  71      10183   G   A   \n",
       "...       ...        ...    ...  ...  ..  ..        ...  ..  ..   \n",
       "268495  chr12  133241504  54.35  0|1  54  44  133158779   A   G   \n",
       "268496  chr12  133244280  58.20  0|1  58  47  133158779   G   A   \n",
       "268498  chr12  133249425  53.87  0|1  53  49  133158779   G   T   \n",
       "268499  chr12  133253066  55.29  0|1  55  49  133158779   G   T   \n",
       "268505  chr12  133264934  11.47  1|0  11  44  133158779   T   C   \n",
       "\n",
       "                              ROI             PS_R  \n",
       "3                chr12:1-11895386      chr12_10183  \n",
       "4                chr12:1-11895386      chr12_10183  \n",
       "5                chr12:1-11895386      chr12_10183  \n",
       "17               chr12:1-11895386      chr12_10183  \n",
       "18               chr12:1-11895386      chr12_10183  \n",
       "...                           ...              ...  \n",
       "268495  chr12:128308614-133275309  chr12_133158779  \n",
       "268496  chr12:128308614-133275309  chr12_133158779  \n",
       "268498  chr12:128308614-133275309  chr12_133158779  \n",
       "268499  chr12:128308614-133275309  chr12_133158779  \n",
       "268505  chr12:128308614-133275309  chr12_133158779  \n",
       "\n",
       "[124679 rows x 11 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def excel_style_suffix(n):\n",
    "    print(n)\n",
    "    result = \"\"\n",
    "    while n > 0:\n",
    "        n, remainder = divmod(n - 1, 26)\n",
    "        result = string.ascii_uppercase[remainder] + result\n",
    "    return result\n",
    "\n",
    "\n",
    "def split_ps_tags(df):\n",
    "    df[\"PS_R\"]= \"None\"\n",
    "    for ps, group in df.groupby(\"PS\", dropna=False):\n",
    "        chrm = group[\"CHROM\"].unique()[0]\n",
    "        rois = group[\"ROI\"].dropna().unique()\n",
    "        if len(rois) <=1:\n",
    "            df.loc[df[\"PS\"]==ps, \"PS_R\"] = chrm+\"_\"+ps\n",
    "        else:\n",
    "            roi_to_suffix = {\n",
    "                roi: excel_style_suffix(i)\n",
    "                for i, roi in enumerate(sorted(rois))}\n",
    "            for idx, row in group.iterrows():\n",
    "                if row[\"ROI\"] == \"None\":\n",
    "                    df.loc[idx, \"PS_R\"] = ps\n",
    "                else:\n",
    "                    df.loc[idx, \"PS_R\"] = f\"{chrm}_{ps}.{roi_to_suffix[row['ROI']]}\"\n",
    "    return df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b8804ca0-52d9-49a6-95d7-210a2d002c4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>PS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ROI</th>\n",
       "      <th>PS_R</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>205158</th>\n",
       "      <td>chr12</td>\n",
       "      <td>97397768</td>\n",
       "      <td>62.16</td>\n",
       "      <td>0|1</td>\n",
       "      <td>62</td>\n",
       "      <td>64</td>\n",
       "      <td>97397768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:94108839-97430502</td>\n",
       "      <td>chr12_97397768.A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205159</th>\n",
       "      <td>chr12</td>\n",
       "      <td>97397787</td>\n",
       "      <td>66.45</td>\n",
       "      <td>0|1</td>\n",
       "      <td>66</td>\n",
       "      <td>64</td>\n",
       "      <td>97397768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:94108839-97430502</td>\n",
       "      <td>chr12_97397768.A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205160</th>\n",
       "      <td>chr12</td>\n",
       "      <td>97397885</td>\n",
       "      <td>55.36</td>\n",
       "      <td>0|1</td>\n",
       "      <td>55</td>\n",
       "      <td>65</td>\n",
       "      <td>97397768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:94108839-97430502</td>\n",
       "      <td>chr12_97397768.A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205161</th>\n",
       "      <td>chr12</td>\n",
       "      <td>97397979</td>\n",
       "      <td>61.68</td>\n",
       "      <td>1|0</td>\n",
       "      <td>61</td>\n",
       "      <td>65</td>\n",
       "      <td>97397768</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr12:94108839-97430502</td>\n",
       "      <td>chr12_97397768.A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205162</th>\n",
       "      <td>chr12</td>\n",
       "      <td>97398043</td>\n",
       "      <td>58.55</td>\n",
       "      <td>0|1</td>\n",
       "      <td>58</td>\n",
       "      <td>65</td>\n",
       "      <td>97397768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:94108839-97430502</td>\n",
       "      <td>chr12_97397768.A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210749</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101499537</td>\n",
       "      <td>60.54</td>\n",
       "      <td>1|0</td>\n",
       "      <td>60</td>\n",
       "      <td>53</td>\n",
       "      <td>97397768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:98032956-115177215</td>\n",
       "      <td>chr12_97397768.B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210750</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101499948</td>\n",
       "      <td>58.30</td>\n",
       "      <td>1|0</td>\n",
       "      <td>58</td>\n",
       "      <td>52</td>\n",
       "      <td>97397768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:98032956-115177215</td>\n",
       "      <td>chr12_97397768.B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210753</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101500504</td>\n",
       "      <td>56.29</td>\n",
       "      <td>1|0</td>\n",
       "      <td>56</td>\n",
       "      <td>54</td>\n",
       "      <td>97397768</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:98032956-115177215</td>\n",
       "      <td>chr12_97397768.B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210755</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101504306</td>\n",
       "      <td>70.21</td>\n",
       "      <td>0|1</td>\n",
       "      <td>70</td>\n",
       "      <td>60</td>\n",
       "      <td>97397768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:98032956-115177215</td>\n",
       "      <td>chr12_97397768.B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210756</th>\n",
       "      <td>chr12</td>\n",
       "      <td>101504452</td>\n",
       "      <td>56.68</td>\n",
       "      <td>1|0</td>\n",
       "      <td>56</td>\n",
       "      <td>61</td>\n",
       "      <td>97397768</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:98032956-115177215</td>\n",
       "      <td>chr12_97397768.B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3597 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS   QUAL   GT  GQ  DP        PS REF ALT  \\\n",
       "205158  chr12   97397768  62.16  0|1  62  64  97397768   G   A   \n",
       "205159  chr12   97397787  66.45  0|1  66  64  97397768   A   G   \n",
       "205160  chr12   97397885  55.36  0|1  55  65  97397768   G   A   \n",
       "205161  chr12   97397979  61.68  1|0  61  65  97397768   T   C   \n",
       "205162  chr12   97398043  58.55  0|1  58  65  97397768   A   G   \n",
       "...       ...        ...    ...  ...  ..  ..       ...  ..  ..   \n",
       "210749  chr12  101499537  60.54  1|0  60  53  97397768   A   G   \n",
       "210750  chr12  101499948  58.30  1|0  58  52  97397768   A   G   \n",
       "210753  chr12  101500504  56.29  1|0  56  54  97397768   C   T   \n",
       "210755  chr12  101504306  70.21  0|1  70  60  97397768   G   A   \n",
       "210756  chr12  101504452  56.68  1|0  56  61  97397768   C   G   \n",
       "\n",
       "                             ROI              PS_R  \n",
       "205158   chr12:94108839-97430502  chr12_97397768.A  \n",
       "205159   chr12:94108839-97430502  chr12_97397768.A  \n",
       "205160   chr12:94108839-97430502  chr12_97397768.A  \n",
       "205161   chr12:94108839-97430502  chr12_97397768.A  \n",
       "205162   chr12:94108839-97430502  chr12_97397768.A  \n",
       "...                          ...               ...  \n",
       "210749  chr12:98032956-115177215  chr12_97397768.B  \n",
       "210750  chr12:98032956-115177215  chr12_97397768.B  \n",
       "210753  chr12:98032956-115177215  chr12_97397768.B  \n",
       "210755  chr12:98032956-115177215  chr12_97397768.B  \n",
       "210756  chr12:98032956-115177215  chr12_97397768.B  \n",
       "\n",
       "[3597 rows x 11 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"PS\"] == \"97397768\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3c940cd-aa5f-4fb6-a18f-a1052729e8e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_ps_tags(df):\n",
    "    ### This function adds A B C etc to sub-haploblocks (haploblocks that are defined by multiple ROIs)\n",
    "    new_ps = [\"\"] * len(df)\n",
    "    print(df)\n",
    "    for ps, group in df.groupby(\"PS\", dropna=False):\n",
    "        print(ps)\n",
    "        print(group)\n",
    "        rois = group[\"ROI\"].dropna().unique()\n",
    "        print(rois)\n",
    "        ### Single or no ROI → PS unchanged\n",
    "        if len(rois) <= 1:\n",
    "            print(group.index)\n",
    "            for idx in group.index:\n",
    "                print(idx)\n",
    "                print(ps)\n",
    "                new_ps[idx] = ps\n",
    "            continue\n",
    "        ### Multiple ROIs → assign suffixes A, B, C...\n",
    "        roi_to_suffix = {\n",
    "            roi: string.ascii_uppercase[i]\n",
    "            for i, roi in enumerate(sorted(rois))}\n",
    "        for idx, row in group.iterrows():\n",
    "            if pd.isna(row[\"ROI\"]):\n",
    "                new_ps[idx] = ps\n",
    "            else:\n",
    "                new_ps[idx] = f\"{ps}.{roi_to_suffix[row['ROI']]}\"\n",
    "    return new_ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e36a37be-4bdd-453f-8d74-76b9e413f529",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>DP</th>\n",
       "      <th>PS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ROI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10183</td>\n",
       "      <td>8.77</td>\n",
       "      <td>1|0</td>\n",
       "      <td>8</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10185</td>\n",
       "      <td>7.52</td>\n",
       "      <td>1|0</td>\n",
       "      <td>7</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10194</td>\n",
       "      <td>10.42</td>\n",
       "      <td>1|0</td>\n",
       "      <td>10</td>\n",
       "      <td>52</td>\n",
       "      <td>10183</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10491</td>\n",
       "      <td>5.05</td>\n",
       "      <td>1|0</td>\n",
       "      <td>5</td>\n",
       "      <td>55</td>\n",
       "      <td>10183</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr12</td>\n",
       "      <td>10649</td>\n",
       "      <td>14.34</td>\n",
       "      <td>1|0</td>\n",
       "      <td>14</td>\n",
       "      <td>71</td>\n",
       "      <td>10183</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:1-11895386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268495</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133241504</td>\n",
       "      <td>54.35</td>\n",
       "      <td>0|1</td>\n",
       "      <td>54</td>\n",
       "      <td>44</td>\n",
       "      <td>133158779</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268496</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133244280</td>\n",
       "      <td>58.20</td>\n",
       "      <td>0|1</td>\n",
       "      <td>58</td>\n",
       "      <td>47</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268498</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133249425</td>\n",
       "      <td>53.87</td>\n",
       "      <td>0|1</td>\n",
       "      <td>53</td>\n",
       "      <td>49</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268499</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133253066</td>\n",
       "      <td>55.29</td>\n",
       "      <td>0|1</td>\n",
       "      <td>55</td>\n",
       "      <td>49</td>\n",
       "      <td>133158779</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268505</th>\n",
       "      <td>chr12</td>\n",
       "      <td>133264934</td>\n",
       "      <td>11.47</td>\n",
       "      <td>1|0</td>\n",
       "      <td>11</td>\n",
       "      <td>44</td>\n",
       "      <td>133158779</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr12:128308614-133275309</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>124679 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS   QUAL   GT  GQ  DP         PS REF ALT  \\\n",
       "3       chr12      10183   8.77  1|0   8  52      10183   T   A   \n",
       "4       chr12      10185   7.52  1|0   7  52      10183   C   A   \n",
       "5       chr12      10194  10.42  1|0  10  52      10183   A   T   \n",
       "17      chr12      10491   5.05  1|0   5  55      10183   A   T   \n",
       "18      chr12      10649  14.34  1|0  14  71      10183   G   A   \n",
       "...       ...        ...    ...  ...  ..  ..        ...  ..  ..   \n",
       "268495  chr12  133241504  54.35  0|1  54  44  133158779   A   G   \n",
       "268496  chr12  133244280  58.20  0|1  58  47  133158779   G   A   \n",
       "268498  chr12  133249425  53.87  0|1  53  49  133158779   G   T   \n",
       "268499  chr12  133253066  55.29  0|1  55  49  133158779   G   T   \n",
       "268505  chr12  133264934  11.47  1|0  11  44  133158779   T   C   \n",
       "\n",
       "                              ROI  \n",
       "3                chr12:1-11895386  \n",
       "4                chr12:1-11895386  \n",
       "5                chr12:1-11895386  \n",
       "17               chr12:1-11895386  \n",
       "18               chr12:1-11895386  \n",
       "...                           ...  \n",
       "268495  chr12:128308614-133275309  \n",
       "268496  chr12:128308614-133275309  \n",
       "268498  chr12:128308614-133275309  \n",
       "268499  chr12:128308614-133275309  \n",
       "268505  chr12:128308614-133275309  \n",
       "\n",
       "[124679 rows x 10 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### RUN scripts\n",
    "\n",
    "# OMIM test set of 50000 variants\n",
    "variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/test_nh.vcf\"\n",
    "all_variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/SAMPLE_renamed.vcf.gz\"\n",
    "phased_variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/phased_ROI_HG2.vcf.gz\"\n",
    "bed=\"/hpc/umc_laat/gvandersluis/data/Ref_HG/HG_OMIM_ROI_merged.bed\"\n",
    "samp=\"2\"\n",
    "vers=\"OMIM\"\n",
    "grouped = hb_maker(variants,bed) # maar van all_variants de header version van de variants\n",
    "grouped\n",
    "# grouped.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{vers}_ROI/final_df.csv\", index=False)\n",
    "\n",
    "# grouped.sort_values([\"chromosome\", \"PS_tag\"]).head(20)\n",
    "# grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af3b4700-b920-4de9-a2e5-686ad4107f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0f77d35a-b918-4b3d-b653-352bc3a51395",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pysam.libcbcf.VariantFile at 0x7fb777ed2950>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "674324ec-f5b0-4b05-82ad-c676d016f233",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PS_tag</th>\n",
       "      <th>START_HB</th>\n",
       "      <th>END_HB</th>\n",
       "      <th>informative_variants</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>QUAL_mean</th>\n",
       "      <th>GENES_in_HB</th>\n",
       "      <th>HB_length</th>\n",
       "      <th>total_VARS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>16514442</td>\n",
       "      <td>16514442</td>\n",
       "      <td>21189725</td>\n",
       "      <td>5755</td>\n",
       "      <td>chr12</td>\n",
       "      <td>53.974622</td>\n",
       "      <td>[GOT2P4, GOT2P4-LMO3, LMO3, LMO3-LOC105369677,...</td>\n",
       "      <td>4675283</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      PS_tag  START_HB    END_HB  informative_variants chromosome  QUAL_mean  \\\n",
       "47  16514442  16514442  21189725                  5755      chr12  53.974622   \n",
       "\n",
       "                                          GENES_in_HB  HB_length total_VARS  \n",
       "47  [GOT2P4, GOT2P4-LMO3, LMO3, LMO3-LOC105369677,...    4675283       None  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped[grouped[\"PS_tag\"].astype(int) == 16514442]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "519ff287-343e-43f7-878b-85791f7efa5f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['apptainer', 'exec', '-B', '/hpc/:/hpc/', '/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif', 'bcftools', 'index', '-t', '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/16514442_phased_vars.vcf'], returncode=0)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "out_vcf = f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/16514442_BM.vcf\"\n",
    "cmd = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", \"chr12:16514442-21189725\", \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/HG002_BM_SSANDT_rn.vcf\", \"-Oz\", \"-o\", out_vcf]\n",
    "subprocess.run(cmd, check=True)\n",
    "indexer = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"index\", \"-t\", out_vcf]\n",
    "subprocess.run(indexer, check=True)\n",
    "\n",
    "### Make specific region version of the phased vcf file (Prepare file for whatshap compare)\n",
    "out_ph_vars_vcf = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/16514442_phased_vars.vcf\"\n",
    "cmd_1 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", \"chr12:16514442-21189725\", \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/phased_ROI_HG2.vcf.gz\", \"-Oz\", \"-o\", out_ph_vars_vcf]\n",
    "subprocess.run(cmd_1, check=True)\n",
    "indexer_1 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"index\", \"-t\", out_ph_vars_vcf]\n",
    "subprocess.run(indexer_1, check=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "64e46ed1-2e91-4670-9fcd-dc4cc2746974",
   "metadata": {},
   "outputs": [
    {
     "ename": "UnicodeDecodeError",
     "evalue": "'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mUnicodeDecodeError\u001b[39m                        Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[28]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[43mpd\u001b[49m\u001b[43m.\u001b[49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43m/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/16514442_phased_vars.vcf\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/hpc/umc_laat/gvandersluis/software/miniforge3/envs/NIPD/lib/python3.14/site-packages/pandas/io/parsers/readers.py:1026\u001b[39m, in \u001b[36mread_csv\u001b[39m\u001b[34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[39m\n\u001b[32m   1013\u001b[39m kwds_defaults = _refine_defaults_read(\n\u001b[32m   1014\u001b[39m     dialect,\n\u001b[32m   1015\u001b[39m     delimiter,\n\u001b[32m   (...)\u001b[39m\u001b[32m   1022\u001b[39m     dtype_backend=dtype_backend,\n\u001b[32m   1023\u001b[39m )\n\u001b[32m   1024\u001b[39m kwds.update(kwds_defaults)\n\u001b[32m-> \u001b[39m\u001b[32m1026\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/hpc/umc_laat/gvandersluis/software/miniforge3/envs/NIPD/lib/python3.14/site-packages/pandas/io/parsers/readers.py:620\u001b[39m, in \u001b[36m_read\u001b[39m\u001b[34m(filepath_or_buffer, kwds)\u001b[39m\n\u001b[32m    617\u001b[39m _validate_names(kwds.get(\u001b[33m\"\u001b[39m\u001b[33mnames\u001b[39m\u001b[33m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[32m    619\u001b[39m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m620\u001b[39m parser = \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    622\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[32m    623\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/hpc/umc_laat/gvandersluis/software/miniforge3/envs/NIPD/lib/python3.14/site-packages/pandas/io/parsers/readers.py:1620\u001b[39m, in \u001b[36mTextFileReader.__init__\u001b[39m\u001b[34m(self, f, engine, **kwds)\u001b[39m\n\u001b[32m   1617\u001b[39m     \u001b[38;5;28mself\u001b[39m.options[\u001b[33m\"\u001b[39m\u001b[33mhas_index_names\u001b[39m\u001b[33m\"\u001b[39m] = kwds[\u001b[33m\"\u001b[39m\u001b[33mhas_index_names\u001b[39m\u001b[33m\"\u001b[39m]\n\u001b[32m   1619\u001b[39m \u001b[38;5;28mself\u001b[39m.handles: IOHandles | \u001b[38;5;28;01mNone\u001b[39;00m = \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[32m-> \u001b[39m\u001b[32m1620\u001b[39m \u001b[38;5;28mself\u001b[39m._engine = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/hpc/umc_laat/gvandersluis/software/miniforge3/envs/NIPD/lib/python3.14/site-packages/pandas/io/parsers/readers.py:1898\u001b[39m, in \u001b[36mTextFileReader._make_engine\u001b[39m\u001b[34m(self, f, engine)\u001b[39m\n\u001b[32m   1895\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(msg)\n\u001b[32m   1897\u001b[39m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m-> \u001b[39m\u001b[32m1898\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mmapping\u001b[49m\u001b[43m[\u001b[49m\u001b[43mengine\u001b[49m\u001b[43m]\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43moptions\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   1899\u001b[39m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m:\n\u001b[32m   1900\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m.handles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/hpc/umc_laat/gvandersluis/software/miniforge3/envs/NIPD/lib/python3.14/site-packages/pandas/io/parsers/c_parser_wrapper.py:93\u001b[39m, in \u001b[36mCParserWrapper.__init__\u001b[39m\u001b[34m(self, src, **kwds)\u001b[39m\n\u001b[32m     90\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m kwds[\u001b[33m\"\u001b[39m\u001b[33mdtype_backend\u001b[39m\u001b[33m\"\u001b[39m] == \u001b[33m\"\u001b[39m\u001b[33mpyarrow\u001b[39m\u001b[33m\"\u001b[39m:\n\u001b[32m     91\u001b[39m     \u001b[38;5;66;03m# Fail here loudly instead of in cython after reading\u001b[39;00m\n\u001b[32m     92\u001b[39m     import_optional_dependency(\u001b[33m\"\u001b[39m\u001b[33mpyarrow\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m---> \u001b[39m\u001b[32m93\u001b[39m \u001b[38;5;28mself\u001b[39m._reader = \u001b[43mparsers\u001b[49m\u001b[43m.\u001b[49m\u001b[43mTextReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43msrc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     95\u001b[39m \u001b[38;5;28mself\u001b[39m.unnamed_cols = \u001b[38;5;28mself\u001b[39m._reader.unnamed_cols\n\u001b[32m     97\u001b[39m \u001b[38;5;66;03m# error: Cannot determine type of 'names'\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mpandas/_libs/parsers.pyx:574\u001b[39m, in \u001b[36mpandas._libs.parsers.TextReader.__cinit__\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mpandas/_libs/parsers.pyx:663\u001b[39m, in \u001b[36mpandas._libs.parsers.TextReader._get_header\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mpandas/_libs/parsers.pyx:874\u001b[39m, in \u001b[36mpandas._libs.parsers.TextReader._tokenize_rows\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mpandas/_libs/parsers.pyx:891\u001b[39m, in \u001b[36mpandas._libs.parsers.TextReader._check_tokenize_status\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mpandas/_libs/parsers.pyx:2053\u001b[39m, in \u001b[36mpandas._libs.parsers.raise_parser_error\u001b[39m\u001b[34m()\u001b[39m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m<frozen codecs>:325\u001b[39m, in \u001b[36mBufferedIncrementalDecoder.decode\u001b[39m\u001b[34m(self, input, final)\u001b[39m\n\u001b[32m    322\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mdecode\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;28minput\u001b[39m, final=\u001b[38;5;28;01mFalse\u001b[39;00m):\n\u001b[32m    323\u001b[39m     \u001b[38;5;66;03m# decode input (taking the buffer into account)\u001b[39;00m\n\u001b[32m    324\u001b[39m     data = \u001b[38;5;28mself\u001b[39m.buffer + \u001b[38;5;28minput\u001b[39m\n\u001b[32m--> \u001b[39m\u001b[32m325\u001b[39m     (result, consumed) = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_buffer_decode\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfinal\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    326\u001b[39m     \u001b[38;5;66;03m# keep undecoded input until the next call\u001b[39;00m\n\u001b[32m    327\u001b[39m     \u001b[38;5;28mself\u001b[39m.buffer = data[consumed:]\n",
      "\u001b[31mUnicodeDecodeError\u001b[39m: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte"
     ]
    }
   ],
   "source": [
    "pd.read_csv(\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/16514442_phased_vars.vcf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1e67d352-1f6b-4007-b188-a6635fa18e78",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:1000552-1079306\n",
      "1000552 \t Done\n",
      "1\n",
      "chr1:10117019-10283590\n",
      "10117019 \t Done\n",
      "2\n",
      "chr1:10326684-10370590\n",
      "10326684 \t Done\n",
      "3\n",
      "chr1:10424359-10512026\n",
      "10424359 \t Done\n",
      "4\n",
      "chr1:10590287-11036365\n",
      "10590287 \t Done\n",
      "5\n",
      "chr1:11083631-11206194\n",
      "11083631 \t Done\n",
      "6\n",
      "chr1:1118005-1174690\n",
      "1118005 \t Done\n",
      "7\n",
      "chr1:11264462-11356791\n",
      "11264462 \t Done\n",
      "8\n",
      "chr1:11407560-12218875\n",
      "11407560 \t Done\n",
      "9\n",
      "chr1:1216550-1295621\n",
      "1216550 \t Done\n",
      "10\n",
      "chr1:12269238-12902628\n",
      "12269238 \t Done\n",
      "11\n",
      "chr1:13021831-13050669\n",
      "13021831 \t Done\n",
      "12\n",
      "chr1:13104329-13136798\n",
      "13104329 \t Done\n",
      "13\n",
      "chr1:13183818-13265675\n",
      "13183818 \t Done\n",
      "14\n",
      "chr1:13330460-13884604\n",
      "13330460 \t Done\n",
      "15\n",
      "chr1:13942278-16569139\n",
      "13942278 \t Done\n",
      "16\n",
      "chr1:1406214-2432709\n",
      "1406214 \t Done\n",
      "17\n",
      "chr1:16011572-16683109\n",
      "16011572 \t Done\n",
      "18\n",
      "chr1:16755040-16773626\n",
      "16755040 \t Done\n",
      "19\n",
      "chr1:16859997-18039368\n",
      "16859997 \t Done\n",
      "20\n",
      "chr1:18096829-18283908\n",
      "18096829 \t Done\n",
      "21\n",
      "chr1:18349782-19648300\n",
      "18349782 \t Done\n",
      "22\n",
      "chr1:19700469-19999168\n",
      "19700469 \t Done\n",
      "23\n",
      "chr1:20049791-20176268\n",
      "20049791 \t Done\n",
      "24\n",
      "chr1:20279975-21970280\n",
      "20279975 \t Done\n",
      "25\n",
      "chr1:22011988-24673789\n",
      "22011988 \t Done\n",
      "26\n",
      "chr1:2477222-2700365\n",
      "2477222 \t Done\n",
      "27\n",
      "chr1:24778986-24826144\n",
      "24778986 \t Done\n",
      "28\n",
      "chr1:24870382-25691376\n",
      "24870382 \t Done\n",
      "29\n",
      "chr1:25744418-25828593\n",
      "25744418 \t Done\n",
      "30\n",
      "chr1:25932053-26059806\n",
      "25932053 \t Done\n",
      "31\n",
      "chr1:26115759-26620400\n",
      "26115759 \t Done\n",
      "32\n",
      "chr1:26647427-26878509\n",
      "26647427 \t Done\n",
      "33\n",
      "chr1:26924610-27535940\n",
      "26924610 \t Done\n",
      "34\n",
      "chr1:2748830-4371255\n",
      "2748830 \t Done\n",
      "35\n",
      "chr1:27581862-27743697\n",
      "27581862 \t Done\n",
      "36\n",
      "chr1:27882794-29528951\n",
      "27882794 \t Done\n",
      "37\n",
      "chr1:29555950-29771256\n",
      "29555950 \t Done\n",
      "38\n",
      "chr1:30225136-31231795\n",
      "30225136 \t Done\n",
      "39\n",
      "chr1:31298668-31382150\n",
      "31298668 \t Done\n",
      "40\n",
      "chr1:31432048-31806485\n",
      "31432048 \t Done\n",
      "41\n",
      "chr1:31889339-31924723\n",
      "31889339 \t Done\n",
      "42\n",
      "chr1:31978437-32315452\n",
      "31978437 \t Done\n",
      "43\n",
      "chr1:32350309-33175655\n",
      "32350309 \t Done\n",
      "44\n",
      "chr1:33241955-33311656\n",
      "33241955 \t Done\n",
      "45\n",
      "chr1:33371444-33626090\n",
      "33371444 \t Done\n",
      "46\n",
      "chr1:33670536-33928305\n",
      "33670536 \t Done\n",
      "47\n",
      "chr1:33978177-33987928\n",
      "33978177 \t Done\n",
      "48\n",
      "chr1:34072879-34978924\n",
      "34072879 \t Done\n",
      "49\n",
      "chr1:35024693-35046157\n",
      "35024693 \t Done\n",
      "50\n",
      "chr1:35205756-35690863\n",
      "35205756 \t Done\n",
      "51\n",
      "chr1:35810601-35863844\n",
      "35810601 \t Done\n",
      "52\n",
      "chr1:35910532-37165145\n",
      "35910532 \t Done\n",
      "53\n",
      "chr1:37210447-38447915\n",
      "37210447 \t Done\n",
      "54\n",
      "chr1:38485459-39842380\n",
      "38485459 \t Done\n",
      "55\n",
      "chr1:39889337-40987208\n",
      "39889337 \t Done\n",
      "56\n",
      "chr1:41027387-41182743\n",
      "41027387 \t Done\n",
      "57\n",
      "chr1:41249565-41264052\n",
      "41249565 \t Done\n",
      "58\n",
      "chr1:41306391-42757741\n",
      "41306391 \t Done\n",
      "59\n",
      "chr1:42830784-42983816\n",
      "42830784 \t Done\n",
      "60\n",
      "chr1:43026517-43410634\n",
      "43026517 \t Done\n",
      "61\n",
      "chr1:43449966-43462481\n",
      "43449966 \t Done\n",
      "62\n",
      "chr1:43528657-43721408\n",
      "43528657 \t Done\n",
      "63\n",
      "chr1:43780685-44806068\n",
      "43780685 \t Done\n",
      "64\n",
      "chr1:44865590-47244001\n",
      "44865590 \t Done\n",
      "65\n",
      "chr1:4655077-4784883\n",
      "4655077 \t Done\n",
      "66\n",
      "chr1:47323841-47326102\n",
      "47323841 \t Done\n",
      "67\n",
      "chr1:47388417-47442801\n",
      "47388417 \t Done\n",
      "68\n",
      "chr1:47489942-47891364\n",
      "47489942 \t Done\n",
      "69\n",
      "chr1:47945756-47945756\n",
      "47945756 \t Done\n",
      "70\n",
      "chr1:48055478-48631763\n",
      "48055478 \t Done\n",
      "71\n",
      "chr1:48685493-48697255\n",
      "48685493 \t Done\n",
      "72\n",
      "chr1:48750699-48764188\n",
      "48750699 \t Done\n",
      "73\n",
      "chr1:48820290-48893675\n",
      "48820290 \t Done\n",
      "74\n",
      "chr1:48964070-48977976\n",
      "48964070 \t Done\n",
      "75\n",
      "chr1:49063809-49140917\n",
      "49063809 \t Done\n",
      "76\n",
      "chr1:49210457-49213515\n",
      "49210457 \t Done\n",
      "77\n",
      "chr1:49259495-49280714\n",
      "49259495 \t Done\n",
      "78\n",
      "chr1:49359434-49378716\n",
      "49359434 \t Done\n",
      "79\n",
      "chr1:49419430-49425939\n",
      "49419430 \t Done\n",
      "80\n",
      "chr1:49528668-49556374\n",
      "49528668 \t Done\n",
      "81\n",
      "chr1:49589513-49623147\n",
      "49589513 \t Done\n",
      "82\n",
      "chr1:49696255-49756959\n",
      "49696255 \t Done\n",
      "83\n",
      "chr1:49800454-50061567\n",
      "49800454 \t Done\n",
      "84\n",
      "chr1:50126929-51119014\n",
      "50126929 \t Done\n",
      "85\n",
      "chr1:51168714-51219386\n",
      "51165081 \t Done\n",
      "86\n",
      "chr1:51329032-51381950\n",
      "51329032 \t Done\n",
      "87\n",
      "chr1:51480627-51480627\n",
      "51430991 \t Done\n",
      "88\n",
      "chr1:51531037-51549511\n",
      "51531037 \t Done\n",
      "89\n",
      "chr1:51601065-52181982\n",
      "51601065 \t Done\n",
      "90\n",
      "chr1:52220858-52225859\n",
      "52220858 \t Done\n",
      "91\n",
      "chr1:52290658-52449913\n",
      "52290658 \t Done\n",
      "92\n",
      "chr1:52502722-52768904\n",
      "52502722 \t Done\n",
      "93\n",
      "chr1:52805920-52815758\n",
      "52805920 \t Done\n",
      "94\n",
      "chr1:52868384-52884219\n",
      "52868384 \t Done\n",
      "95\n",
      "chr1:52934953-53514369\n",
      "52934953 \t Done\n",
      "96\n",
      "chr1:53566222-53572984\n",
      "53566222 \t Done\n",
      "97\n",
      "chr1:53643837-55548130\n",
      "53643837 \t Done\n",
      "98\n",
      "chr1:56185733-59816040\n",
      "56185733 \t Done\n",
      "99\n",
      "chr1:5641831-8828783\n",
      "5641831 \t Done\n",
      "100\n",
      "chr1:59878944-59925583\n",
      "59878944 \t Done\n",
      "101\n",
      "chr1:60000160-60529364\n",
      "60000160 \t Done\n",
      "102\n",
      "chr1:60865937-60911122\n",
      "60865937 \t Done\n",
      "103\n",
      "chr1:60976337-61660645\n",
      "60976337 \t Done\n",
      "104\n",
      "chr1:61699062-63423169\n",
      "61699062 \t Done\n",
      "105\n",
      "chr1:63462167-63993114\n",
      "63462167 \t Done\n",
      "106\n",
      "chr1:64053551-64176389\n",
      "64053551 \t Done\n",
      "107\n",
      "chr1:64289323-64627930\n",
      "64289323 \t Done\n",
      "108\n",
      "chr1:64682701-65097469\n",
      "64682701 \t Done\n",
      "109\n",
      "chr1:65189420-65886307\n",
      "65189420 \t Done\n",
      "110\n",
      "chr1:655624-693461\n",
      "655624 \t Done\n",
      "111\n",
      "chr1:65949811-65968467\n",
      "65949811 \t Done\n",
      "112\n",
      "chr1:66038974-66987352\n",
      "66038974 \t Done\n",
      "113\n",
      "chr1:67057639-67131127\n",
      "67057639 \t Done\n",
      "114\n",
      "chr1:67183813-68443430\n",
      "67183813 \t Done\n",
      "115\n",
      "chr1:68514106-68555167\n",
      "68514106 \t Done\n",
      "116\n",
      "chr1:68703258-68985614\n",
      "68703258 \t Done\n",
      "117\n",
      "chr1:69573135-70213670\n",
      "69573135 \t Done\n",
      "118\n",
      "chr1:70253199-70333879\n",
      "70253199 \t Done\n",
      "119\n",
      "chr1:70381627-72101326\n",
      "70381627 \t Done\n",
      "120\n",
      "chr1:72152899-72152899\n",
      "72152899 \t Done\n",
      "121\n",
      "chr1:72228330-72280708\n",
      "72228330 \t Done\n",
      "122\n",
      "chr1:73717583-73769029\n",
      "73717583 \t Done\n",
      "123\n",
      "chr1:73832178-73914594\n",
      "73832178 \t Done\n",
      "124\n",
      "chr1:73957395-74319198\n",
      "73957395 \t Done\n",
      "125\n",
      "chr1:74392171-74553677\n",
      "74392171 \t Done\n",
      "126\n",
      "chr1:74616302-75065252\n",
      "74616302 \t Done\n",
      "127\n",
      "chr1:75107430-75764840\n",
      "75107430 \t Done\n",
      "128\n",
      "chr1:75824232-75886416\n",
      "75824232 \t Done\n",
      "129\n",
      "chr1:76011522-76027171\n",
      "76011522 \t Done\n",
      "130\n",
      "chr1:76085843-76239575\n",
      "76085843 \t Done\n",
      "131\n",
      "chr1:76280059-76634006\n",
      "76280059 \t Done\n",
      "132\n",
      "chr1:76893531-77032002\n",
      "76893531 \t Done\n",
      "133\n",
      "chr1:77089796-77914181\n",
      "77089796 \t Done\n",
      "134\n",
      "chr1:77971870-78345989\n",
      "77971870 \t Done\n",
      "135\n",
      "chr1:78409421-78990830\n",
      "78409421 \t Done\n",
      "136\n",
      "chr1:814327-984039\n",
      "814327 \t Done\n",
      "137\n",
      "chr1:8911215-9710807\n",
      "8911215 \t Done\n",
      "138\n",
      "chr1:9764000-9788141\n",
      "9764000 \t Done\n",
      "139\n",
      "chr1:9829909-9884231\n",
      "9829909 \t Done\n",
      "140\n",
      "chr1:9958448-10048500\n",
      "9958448 \t Done\n",
      "141\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/ROI_eval.tsv'"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bm_roi, grouped = run_switch(grouped, all_variants, samp, vers, phased_variants)\n",
    "bm_roi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1b1685b3-c5b6-4cfc-99a5-ba7ca8fbc398",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#sample</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dataset_name0</th>\n",
       "      <th>dataset_name1</th>\n",
       "      <th>file_name0</th>\n",
       "      <th>file_name1</th>\n",
       "      <th>intersection_blocks</th>\n",
       "      <th>covered_variants</th>\n",
       "      <th>all_assessed_pairs</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>...</th>\n",
       "      <th>largestblock_switches</th>\n",
       "      <th>largestblock_switch_rate</th>\n",
       "      <th>largestblock_switchflips</th>\n",
       "      <th>largestblock_switchflip_rate</th>\n",
       "      <th>largestblock_hamming</th>\n",
       "      <th>largestblock_hamming_rate</th>\n",
       "      <th>largestblock_diff_genotypes</th>\n",
       "      <th>largestblock_diff_genotypes_rate</th>\n",
       "      <th>het_variants0</th>\n",
       "      <th>only_snvs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>1000552</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>10117019</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>10326684</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>10424359</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>10590287</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>178</td>\n",
       "      <td>177</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>276</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>4655077</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>101</td>\n",
       "      <td>100</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.020000</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>151</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>30225136</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>864</td>\n",
       "      <td>863</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.002317</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.001159</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001157</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1313</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>56185733</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>2363</td>\n",
       "      <td>2362</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000847</td>\n",
       "      <td>2/0</td>\n",
       "      <td>0.000847</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000846</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4014</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>5641831</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>1774</td>\n",
       "      <td>1773</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001128</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2877</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr1</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>78409421</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>335</td>\n",
       "      <td>334</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.005988</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.002994</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002985</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>556</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    #sample chromosome dataset_name0  dataset_name1  \\\n",
       "0     hg002       chr1     BENCHMARK        1000552   \n",
       "1     hg002       chr1     BENCHMARK       10117019   \n",
       "2     hg002       chr1     BENCHMARK       10326684   \n",
       "3     hg002       chr1     BENCHMARK       10424359   \n",
       "4     hg002       chr1     BENCHMARK       10590287   \n",
       "..      ...        ...           ...            ...   \n",
       "64    hg002       chr1     BENCHMARK        4655077   \n",
       "37    hg002       chr1     BENCHMARK       30225136   \n",
       "97    hg002       chr1     BENCHMARK       56185733   \n",
       "98    hg002       chr1     BENCHMARK        5641831   \n",
       "133   hg002       chr1     BENCHMARK       78409421   \n",
       "\n",
       "                                            file_name0  \\\n",
       "0    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "1    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "2    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "3    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "4    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "..                                                 ...   \n",
       "64   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "37   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "97   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "98   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "133  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "\n",
       "                                            file_name1  intersection_blocks  \\\n",
       "0    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "1    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "2    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "3    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "4    /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "..                                                 ...                  ...   \n",
       "64   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "37   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "97   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "98   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "133  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "\n",
       "     covered_variants  all_assessed_pairs  all_switches  ...  \\\n",
       "0                   3                   2             0  ...   \n",
       "1                  13                  12             0  ...   \n",
       "2                   2                   1             0  ...   \n",
       "3                   7                   6             0  ...   \n",
       "4                 178                 177             0  ...   \n",
       "..                ...                 ...           ...  ...   \n",
       "64                101                 100             2  ...   \n",
       "37                864                 863             2  ...   \n",
       "97               2363                2362             2  ...   \n",
       "98               1774                1773             2  ...   \n",
       "133               335                 334             2  ...   \n",
       "\n",
       "     largestblock_switches largestblock_switch_rate  largestblock_switchflips  \\\n",
       "0                        0                 0.000000                       0/0   \n",
       "1                        0                 0.000000                       0/0   \n",
       "2                        0                 0.000000                       0/0   \n",
       "3                        0                 0.000000                       0/0   \n",
       "4                        0                 0.000000                       0/0   \n",
       "..                     ...                      ...                       ...   \n",
       "64                       2                 0.020000                       0/1   \n",
       "37                       2                 0.002317                       0/1   \n",
       "97                       2                 0.000847                       2/0   \n",
       "98                       2                 0.001128                       0/1   \n",
       "133                      2                 0.005988                       0/1   \n",
       "\n",
       "     largestblock_switchflip_rate  largestblock_hamming  \\\n",
       "0                        0.000000                     0   \n",
       "1                        0.000000                     0   \n",
       "2                        0.000000                     0   \n",
       "3                        0.000000                     0   \n",
       "4                        0.000000                     0   \n",
       "..                            ...                   ...   \n",
       "64                       0.010000                     1   \n",
       "37                       0.001159                     1   \n",
       "97                       0.000847                     2   \n",
       "98                       0.000564                     1   \n",
       "133                      0.002994                     1   \n",
       "\n",
       "     largestblock_hamming_rate  largestblock_diff_genotypes  \\\n",
       "0                     0.000000                            0   \n",
       "1                     0.000000                            0   \n",
       "2                     0.000000                            0   \n",
       "3                     0.000000                            0   \n",
       "4                     0.000000                            0   \n",
       "..                         ...                          ...   \n",
       "64                    0.009901                            0   \n",
       "37                    0.001157                            0   \n",
       "97                    0.000846                            0   \n",
       "98                    0.000564                            0   \n",
       "133                   0.002985                            0   \n",
       "\n",
       "     largestblock_diff_genotypes_rate  het_variants0  only_snvs  \n",
       "0                                 0.0              5          0  \n",
       "1                                 0.0             19          0  \n",
       "2                                 0.0              4          0  \n",
       "3                                 0.0             12          0  \n",
       "4                                 0.0            276          0  \n",
       "..                                ...            ...        ...  \n",
       "64                                0.0            151          0  \n",
       "37                                0.0           1313          0  \n",
       "97                                0.0           4014          0  \n",
       "98                                0.0           2877          0  \n",
       "133                               0.0            556          0  \n",
       "\n",
       "[139 rows x 28 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sw_e = pd.read_csv(\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/ROI_eval.tsv\", sep=\"\\t\")\n",
    "sw_e.sort_values(\"all_switches\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e2651580-8aac-4b49-a98e-0abc701ae642",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PS_tag</th>\n",
       "      <th>START_HB</th>\n",
       "      <th>END_HB</th>\n",
       "      <th>informative_variants</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>QUAL_mean</th>\n",
       "      <th>GENES_in_HB</th>\n",
       "      <th>HB_length</th>\n",
       "      <th>total_VARS</th>\n",
       "      <th>het_variants0</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>all_switch_rate</th>\n",
       "      <th>all_switchflips</th>\n",
       "      <th>all_switchflip_rate</th>\n",
       "      <th>blockwise_hamming_rate</th>\n",
       "      <th>Accuracy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>30225136</td>\n",
       "      <td>30225136</td>\n",
       "      <td>31231795</td>\n",
       "      <td>1134</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.033616</td>\n",
       "      <td>[LOC105378617-LOC105378618, LOC105378618, LOC1...</td>\n",
       "      <td>1006659</td>\n",
       "      <td>None</td>\n",
       "      <td>1313.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.002317</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.001159</td>\n",
       "      <td>0.001157</td>\n",
       "      <td>99.9988425925926%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>56185733</td>\n",
       "      <td>56185733</td>\n",
       "      <td>59816040</td>\n",
       "      <td>3314</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.832764</td>\n",
       "      <td>[LOC105378741-RPSAP20, RPSAP20-LINC01767, LINC...</td>\n",
       "      <td>3630307</td>\n",
       "      <td>None</td>\n",
       "      <td>4014.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000847</td>\n",
       "      <td>2/0</td>\n",
       "      <td>0.000847</td>\n",
       "      <td>0.000846</td>\n",
       "      <td>99.99915361828185%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>5641831</td>\n",
       "      <td>5641831</td>\n",
       "      <td>8828783</td>\n",
       "      <td>2394</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.887686</td>\n",
       "      <td>[LOC105376686-MIR4689, MIR4689, NPHP4, KCNAB2,...</td>\n",
       "      <td>3186952</td>\n",
       "      <td>None</td>\n",
       "      <td>2877.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.001128</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>0.000564</td>\n",
       "      <td>99.99943630214206%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>78409421</td>\n",
       "      <td>78409421</td>\n",
       "      <td>78990830</td>\n",
       "      <td>479</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.100125</td>\n",
       "      <td>[RNA5SP23-PTGFR, PTGFR, PTGFR-IFI44L, IFI44L, ...</td>\n",
       "      <td>581409</td>\n",
       "      <td>None</td>\n",
       "      <td>556.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.005988</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.002994</td>\n",
       "      <td>0.002985</td>\n",
       "      <td>99.99701492537314%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>16755040</td>\n",
       "      <td>16755040</td>\n",
       "      <td>16773626</td>\n",
       "      <td>22</td>\n",
       "      <td>chr1</td>\n",
       "      <td>47.836818</td>\n",
       "      <td>[MST1L]</td>\n",
       "      <td>18586</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>655624</td>\n",
       "      <td>655624</td>\n",
       "      <td>693461</td>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>48.973333</td>\n",
       "      <td>[MTCO3P12-WBP1LP6, OR4F16]</td>\n",
       "      <td>37837</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       PS_tag  START_HB    END_HB  informative_variants chromosome  QUAL_mean  \\\n",
       "38   30225136  30225136  31231795                  1134       chr1  57.033616   \n",
       "98   56185733  56185733  59816040                  3314       chr1  56.832764   \n",
       "99    5641831   5641831   8828783                  2394       chr1  56.887686   \n",
       "135  78409421  78409421  78990830                   479       chr1  57.100125   \n",
       "18   16755040  16755040  16773626                    22       chr1  47.836818   \n",
       "110    655624    655624    693461                     3       chr1  48.973333   \n",
       "\n",
       "                                           GENES_in_HB  HB_length total_VARS  \\\n",
       "38   [LOC105378617-LOC105378618, LOC105378618, LOC1...    1006659       None   \n",
       "98   [LOC105378741-RPSAP20, RPSAP20-LINC01767, LINC...    3630307       None   \n",
       "99   [LOC105376686-MIR4689, MIR4689, NPHP4, KCNAB2,...    3186952       None   \n",
       "135  [RNA5SP23-PTGFR, PTGFR, PTGFR-IFI44L, IFI44L, ...     581409       None   \n",
       "18                                             [MST1L]      18586       None   \n",
       "110                         [MTCO3P12-WBP1LP6, OR4F16]      37837       None   \n",
       "\n",
       "     het_variants0  all_switches  all_switch_rate all_switchflips  \\\n",
       "38          1313.0           2.0         0.002317             0/1   \n",
       "98          4014.0           2.0         0.000847             2/0   \n",
       "99          2877.0           2.0         0.001128             0/1   \n",
       "135          556.0           2.0         0.005988             0/1   \n",
       "18             NaN           NaN              NaN             NaN   \n",
       "110            NaN           NaN              NaN             NaN   \n",
       "\n",
       "     all_switchflip_rate  blockwise_hamming_rate            Accuracy  \n",
       "38              0.001159                0.001157   99.9988425925926%  \n",
       "98              0.000847                0.000846  99.99915361828185%  \n",
       "99              0.000564                0.000564  99.99943630214206%  \n",
       "135             0.002994                0.002985  99.99701492537314%  \n",
       "18                   NaN                     NaN                 NaN  \n",
       "110                  NaN                     NaN                 NaN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sw_e = pd.read_csv(\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/OMIM_ROI/ROI_eval.tsv\", sep=\"\\t\")\n",
    "sw_e.sort_values(\"all_switches\")\n",
    "\n",
    "### Format the switch error file\n",
    "sw_e = sw_e[[\"dataset_name1\",\"het_variants0\",\"all_switches\",\"all_switch_rate\",\"all_switchflips\",\"all_switchflip_rate\",\"blockwise_hamming_rate\"]]\n",
    "sw_e[\"Accuracy\"] = (100-sw_e[\"blockwise_hamming_rate\"]).astype(str)+\"%\"\n",
    "sw_e.drop(columns=\"blockwise_hamming_rate\")\n",
    "sw_e.rename(columns={\"dataset_name1\": \"PS_tag\"}, inplace=True)\n",
    "sw_e = pd.merge(grouped, sw_e, on=\"PS_tag\", how=\"left\")\n",
    "sw_e.sort_values(\"all_switches\").tail(6)\n",
    "# ### Merge Hapoblocks with their switch errors\n",
    "# basis_df[\"GENE_l\"] = basis_df[\"GENE\"]+\"_\"+basis_df[\"HB_START\"].astype(str)    \n",
    "# switch_df = pd.merge(basis_df, sw_e, on=\"GENE_l\", how=\"left\")\n",
    "# switch_df[\"difference_het-vars_BM_vs_data\"] = abs(switch_df[\"PHASED_VARIANTS\"]-switch_df[\"het_variants0\"])\n",
    "# switch_df = switch_df.drop(columns=\"GENE_l\")\n",
    "\n",
    "#     ### Write merged dataframe to file\n",
    "#     switch_df.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{version}_ROI/final_df.csv\", index=False)\n",
    "    \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:NIPD]",
   "language": "python",
   "name": "conda-env-NIPD-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
