{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "aab540c1-4038-4035-97ff-5bc89b9837f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "### Imports\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import os\n",
    "import pysam\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7c49818c-2b48-4a55-a568-dbbc80cdc744",
   "metadata": {},
   "outputs": [],
   "source": [
    "def hb_maker(phased_vars_file):\n",
    "    ### Read the filtered vcf file (filtered on ROI of the OMIM genes & QUAL >= 40) and get the desired columns\n",
    "    df_vcf = pd.read_csv(phased_vars_file, sep=\"\\t\", names=['CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT', \"sample\"])\n",
    "    df_vcf[\"GENE\"] = df_vcf['INFO'].str.split(\"|\").str[4]\n",
    "    df_splitted = df_vcf['sample'].str.split(\":\")\n",
    "    df_vcf[\"GT\"] = df_splitted.str[0]\n",
    "    df_vcf[\"DP\"] = df_splitted.str[2]\n",
    "    df_vcf[\"PS\"] = df_splitted.str[5]\n",
    "\n",
    "    ### Only keep the phased variants (the variants with a phase tag)\n",
    "    phased = df_vcf.loc[df_vcf['PS'].notna(), ['CHROM','POS','QUAL','GENE','GT','DP','PS','REF','ALT']]\n",
    "\n",
    "    ### Group by phase tag and get the min, max, and amount of variants per haploblock\n",
    "    grouped = phased.groupby(['PS']).agg({'POS': ['min','max', 'count'],\n",
    "                                                    'CHROM': 'first',\n",
    "                                                    'QUAL': 'mean',\n",
    "                                                    'GENE': 'unique'}).reset_index()\n",
    "\n",
    "    ### Change the column names, type, and add desired columns\n",
    "    grouped.columns = ['_'.join(col).rstrip('_') if isinstance(col, tuple) else col\n",
    "                   for col in grouped.columns]\n",
    "    grouped.rename(columns={\"PS\":\"PS_tag\", \"POS_min\":\"START_HB\", \"POS_max\": \"END_HB\", \"POS_count\":\"informative_variants\", \"CHROM_first\":\"chromosome\", \"GENE_unique\":\"GENES_in_HB\"}, inplace=True)\n",
    "    grouped[\"PS_tag\"] = grouped[\"PS_tag\"].astype(int)\n",
    "    grouped[\"HB_length\"]=grouped[\"END_HB\"]-grouped[\"START_HB\"]\n",
    "    grouped[\"total_VARS\"] = None\n",
    "    return grouped\n",
    "\n",
    "\n",
    "def run_switch(haploblock_table, all_vars, sample, version, ph_vars):\n",
    "    ### A counter to check how many haploblocks are already compared to BM\n",
    "    cnt = 0 \n",
    "\n",
    "    ### Variables and Files for the output\n",
    "    folder = \"/hpc/umc_laat/gvandersluis/data/\"    \n",
    "    BM_vars = f\"{folder}Ont_data_nhung/HG00{sample}/HG00{sample}_BM_SSANDT_rn.vcf\"\n",
    "    BM_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/ROI_eval.tsv\"\n",
    "    open(BM_ROI, \"w\").close()\n",
    "    switches_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/switches_ROI.bed\"\n",
    "    open(switches_ROI, \"w\").close()\n",
    "\n",
    "    ### Loop through every haploblock\n",
    "    for idx, item in haploblock_table.iterrows():\n",
    "        ### Get the haploblock region\n",
    "        reg = item[\"chromosome\"]+\":\"+str(item[\"START_HB\"])+\"-\"+str(item[\"END_HB\"])\n",
    "        print(reg)\n",
    "        ### ADD TOTAL VARIANTS in the region from the unfiltered vcf file\n",
    "        cmd = f\"apptainer exec -B /hpc/:/hpc/ /hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif bcftools view -H -r {reg} {all_vars} | wc -l\"\n",
    "        ### Run the command\n",
    "        result = subprocess.run(cmd, shell=True, check=True, capture_output=True, text=True)\n",
    "        ### Get the count as integer and add the value to the column\n",
    "        count = int(result.stdout.strip())\n",
    "        grouped.loc[idx, \"total_VARS\"] = count\n",
    "        \n",
    "        ### WHATSHAP COMPARE\n",
    "        PStag = str(item[\"PS_tag\"])\n",
    "        if \"NaN\" not in reg:\n",
    "            ### Prepare the benchmark file for whatshap compare\n",
    "            ### make an output file for the filtered Benchmark file (filtered on region of interest) and run that code + index the vcf\n",
    "            out_vcf = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{PStag}_BM.vcf\"\n",
    "            cmd = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", reg, BM_vars, \"-Oz\", \"-o\", out_vcf]\n",
    "            subprocess.run(cmd, check=True)\n",
    "            indexer = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"index\", \"-t\", out_vcf]\n",
    "            subprocess.run(indexer, check=True)\n",
    "\n",
    "            ### Prepare output files for WHATSHAP compare\n",
    "            indexed = out_vcf+\".tbi\"\n",
    "            tsv_out = f\"{PStag}_eval.tsv\"\n",
    "            open(tsv_out, \"a+\").write(\"\")\n",
    "\n",
    "            ### If the output vcf file has variants (If there are variants found in the benchmark file of this particular region, then:)\n",
    "            if sum(1 for _ in pysam.VariantFile(out_vcf)) != 0:\n",
    "                ### Make file for the switch error bed file\n",
    "                bed_out = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{PStag}_switch.bed\"\n",
    "                ### RUN WHATSHAP COMPARE\n",
    "                cmd2 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/whatshap_v1.sif\", \"whatshap\", \"compare\",\"--switch-error-bed\",bed_out,\"--tsv-pairwise\", tsv_out, \"--names\", f\"BENCHMARK,{PStag}\", out_vcf, ph_vars]\n",
    "                subprocess.run(cmd2, check=True, capture_output=True)\n",
    "\n",
    "                ### If the whatsap compare output file is empty (first run), then write output\n",
    "                if os.path.getsize(BM_ROI) == 0:\n",
    "                    with open(BM_ROI, \"w\") as out:\n",
    "                        with open(tsv_out, \"r\") as inp:\n",
    "                            out.write(inp.read())\n",
    "                ### If whatshap compare output file is not empty anymore (every run after the first), then append output\n",
    "                else:\n",
    "                    with open(BM_ROI, \"a\") as out:\n",
    "                        with open(tsv_out, \"r\") as inp:\n",
    "                            next(inp) # Skip header\n",
    "                            out.write(inp.read())\n",
    "                ### Write or append switch error locations to the bed file\n",
    "                with open(switches_ROI, \"a+\") as out_b:\n",
    "                    with open(bed_out, \"r\") as inp_b:\n",
    "                        out_b.write(inp_b.read())\n",
    "            ### REMOVE temporary needed files\n",
    "                os.remove(bed_out)\n",
    "            os.remove(out_vcf)\n",
    "            os.remove(indexed)\n",
    "            os.remove(tsv_out)\n",
    "        print(PStag, \"\\t Done\")\n",
    "        \n",
    "        cnt += 1\n",
    "        print(cnt)\n",
    "    return grouped, BM_ROI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9e78426e-eba2-43a6-8bcd-cbb52882b1fd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PS_tag</th>\n",
       "      <th>START_HB</th>\n",
       "      <th>END_HB</th>\n",
       "      <th>informative_variants</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>QUAL_mean</th>\n",
       "      <th>GENES_in_HB</th>\n",
       "      <th>HB_length</th>\n",
       "      <th>total_VARS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>655624</td>\n",
       "      <td>655624</td>\n",
       "      <td>693461</td>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>48.973333</td>\n",
       "      <td>[MTCO3P12-WBP1LP6, OR4F16]</td>\n",
       "      <td>37837</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>814327</td>\n",
       "      <td>814327</td>\n",
       "      <td>984039</td>\n",
       "      <td>22</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55.885000</td>\n",
       "      <td>[FAM87B, LINC01128, FAM41C-LOC284600, LOC28460...</td>\n",
       "      <td>169712</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000552</td>\n",
       "      <td>1000552</td>\n",
       "      <td>1079306</td>\n",
       "      <td>28</td>\n",
       "      <td>chr1</td>\n",
       "      <td>48.257857</td>\n",
       "      <td>[HES4, AGRN, RNF223]</td>\n",
       "      <td>78754</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1118005</td>\n",
       "      <td>1118005</td>\n",
       "      <td>1174690</td>\n",
       "      <td>72</td>\n",
       "      <td>chr1</td>\n",
       "      <td>58.305278</td>\n",
       "      <td>[C1orf159, C1orf159-LINC01342, LINC01342, LINC...</td>\n",
       "      <td>56685</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1216550</td>\n",
       "      <td>1216550</td>\n",
       "      <td>1295621</td>\n",
       "      <td>12</td>\n",
       "      <td>chr1</td>\n",
       "      <td>54.552500</td>\n",
       "      <td>[TNFRSF4, C1QTNF12, SCNN1D, LINC01786, ACAP3]</td>\n",
       "      <td>79071</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1406214</td>\n",
       "      <td>1406214</td>\n",
       "      <td>2432709</td>\n",
       "      <td>989</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55.426845</td>\n",
       "      <td>[MRPL20-DT, LINC01770, ATAD3C, ATAD3B, ATAD3A,...</td>\n",
       "      <td>1026495</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>2477222</td>\n",
       "      <td>2477222</td>\n",
       "      <td>2700365</td>\n",
       "      <td>259</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.520965</td>\n",
       "      <td>[PLCH2, PANK4, HES5, HES5-TNFRSF14-AS1, TNFRSF...</td>\n",
       "      <td>223143</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>2748830</td>\n",
       "      <td>2748830</td>\n",
       "      <td>4371255</td>\n",
       "      <td>1536</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55.459102</td>\n",
       "      <td>[TTC34, LOC105378601, LOC112268220, LOC1079857...</td>\n",
       "      <td>1622425</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>4655077</td>\n",
       "      <td>4655077</td>\n",
       "      <td>4784883</td>\n",
       "      <td>133</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.993008</td>\n",
       "      <td>[AJAP1]</td>\n",
       "      <td>129806</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>5641831</td>\n",
       "      <td>5641831</td>\n",
       "      <td>8828783</td>\n",
       "      <td>2394</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.887686</td>\n",
       "      <td>[LOC105376686-MIR4689, MIR4689, NPHP4, KCNAB2,...</td>\n",
       "      <td>3186952</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>8911215</td>\n",
       "      <td>8911215</td>\n",
       "      <td>9710807</td>\n",
       "      <td>764</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.748128</td>\n",
       "      <td>[LOC112268261, LOC112268261-CA6, CA6, RN7SL451...</td>\n",
       "      <td>799592</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>9764000</td>\n",
       "      <td>9764000</td>\n",
       "      <td>9788141</td>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.555000</td>\n",
       "      <td>[CLSTN1]</td>\n",
       "      <td>24141</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>9829909</td>\n",
       "      <td>9829909</td>\n",
       "      <td>9884231</td>\n",
       "      <td>21</td>\n",
       "      <td>chr1</td>\n",
       "      <td>58.601429</td>\n",
       "      <td>[CLSTN1-CTNNBIP1, CTNNBIP1, LOC105376717]</td>\n",
       "      <td>54322</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>9958448</td>\n",
       "      <td>9958448</td>\n",
       "      <td>10048500</td>\n",
       "      <td>9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.754444</td>\n",
       "      <td>[NMNAT1, TMEM274P, MIR5697, RBP7, RBP7-UBE4B, ...</td>\n",
       "      <td>90052</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10117019</td>\n",
       "      <td>10117019</td>\n",
       "      <td>10283590</td>\n",
       "      <td>16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.503750</td>\n",
       "      <td>[UBE4B, UBE4B-KIF1B, KIF1B]</td>\n",
       "      <td>166571</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10326684</td>\n",
       "      <td>10326684</td>\n",
       "      <td>10370590</td>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>55.163333</td>\n",
       "      <td>[KIF1B, LOC105376725]</td>\n",
       "      <td>43906</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10424359</td>\n",
       "      <td>10424359</td>\n",
       "      <td>10512026</td>\n",
       "      <td>8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>59.305000</td>\n",
       "      <td>[PGD, CENPS, CORT, DFFA, PEX14]</td>\n",
       "      <td>87667</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10590287</td>\n",
       "      <td>10590287</td>\n",
       "      <td>11036365</td>\n",
       "      <td>215</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56.020326</td>\n",
       "      <td>[PEX14, CASZ1, LOC105376733, LOC105376733-HSPE...</td>\n",
       "      <td>446078</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>11083631</td>\n",
       "      <td>11083631</td>\n",
       "      <td>11206194</td>\n",
       "      <td>8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>58.996250</td>\n",
       "      <td>[EXOSC10, EXOSC10-AS1, MTOR, MTOR-AS1]</td>\n",
       "      <td>122563</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>11264462</td>\n",
       "      <td>11264462</td>\n",
       "      <td>11356791</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1</td>\n",
       "      <td>57.450851</td>\n",
       "      <td>[MTOR, UBIAD1, LOC105376737, LOC105376737-LOC1...</td>\n",
       "      <td>92329</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       PS_tag  START_HB    END_HB  informative_variants chromosome  QUAL_mean  \\\n",
       "110    655624    655624    693461                     3       chr1  48.973333   \n",
       "136    814327    814327    984039                    22       chr1  55.885000   \n",
       "0     1000552   1000552   1079306                    28       chr1  48.257857   \n",
       "6     1118005   1118005   1174690                    72       chr1  58.305278   \n",
       "9     1216550   1216550   1295621                    12       chr1  54.552500   \n",
       "16    1406214   1406214   2432709                   989       chr1  55.426845   \n",
       "26    2477222   2477222   2700365                   259       chr1  57.520965   \n",
       "34    2748830   2748830   4371255                  1536       chr1  55.459102   \n",
       "65    4655077   4655077   4784883                   133       chr1  56.993008   \n",
       "99    5641831   5641831   8828783                  2394       chr1  56.887686   \n",
       "137   8911215   8911215   9710807                   764       chr1  56.748128   \n",
       "138   9764000   9764000   9788141                     2       chr1  57.555000   \n",
       "139   9829909   9829909   9884231                    21       chr1  58.601429   \n",
       "140   9958448   9958448  10048500                     9       chr1  56.754444   \n",
       "1    10117019  10117019  10283590                    16       chr1  57.503750   \n",
       "2    10326684  10326684  10370590                     3       chr1  55.163333   \n",
       "3    10424359  10424359  10512026                     8       chr1  59.305000   \n",
       "4    10590287  10590287  11036365                   215       chr1  56.020326   \n",
       "5    11083631  11083631  11206194                     8       chr1  58.996250   \n",
       "7    11264462  11264462  11356791                    47       chr1  57.450851   \n",
       "\n",
       "                                           GENES_in_HB  HB_length total_VARS  \n",
       "110                         [MTCO3P12-WBP1LP6, OR4F16]      37837       None  \n",
       "136  [FAM87B, LINC01128, FAM41C-LOC284600, LOC28460...     169712       None  \n",
       "0                                 [HES4, AGRN, RNF223]      78754       None  \n",
       "6    [C1orf159, C1orf159-LINC01342, LINC01342, LINC...      56685       None  \n",
       "9        [TNFRSF4, C1QTNF12, SCNN1D, LINC01786, ACAP3]      79071       None  \n",
       "16   [MRPL20-DT, LINC01770, ATAD3C, ATAD3B, ATAD3A,...    1026495       None  \n",
       "26   [PLCH2, PANK4, HES5, HES5-TNFRSF14-AS1, TNFRSF...     223143       None  \n",
       "34   [TTC34, LOC105378601, LOC112268220, LOC1079857...    1622425       None  \n",
       "65                                             [AJAP1]     129806       None  \n",
       "99   [LOC105376686-MIR4689, MIR4689, NPHP4, KCNAB2,...    3186952       None  \n",
       "137  [LOC112268261, LOC112268261-CA6, CA6, RN7SL451...     799592       None  \n",
       "138                                           [CLSTN1]      24141       None  \n",
       "139          [CLSTN1-CTNNBIP1, CTNNBIP1, LOC105376717]      54322       None  \n",
       "140  [NMNAT1, TMEM274P, MIR5697, RBP7, RBP7-UBE4B, ...      90052       None  \n",
       "1                          [UBE4B, UBE4B-KIF1B, KIF1B]     166571       None  \n",
       "2                                [KIF1B, LOC105376725]      43906       None  \n",
       "3                      [PGD, CENPS, CORT, DFFA, PEX14]      87667       None  \n",
       "4    [PEX14, CASZ1, LOC105376733, LOC105376733-HSPE...     446078       None  \n",
       "5               [EXOSC10, EXOSC10-AS1, MTOR, MTOR-AS1]     122563       None  \n",
       "7    [MTOR, UBIAD1, LOC105376737, LOC105376737-LOC1...      92329       None  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### RUN scripts\n",
    "\n",
    "# OMIM test set of 50000 variants\n",
    "variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/test_nh.vcf\"\n",
    "all_variants=\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/SAMPLE_renamed.vcf.gz\"\n",
    "samp=\"2\"\n",
    "vers=\"OMIM\"\n",
    "grouped = hb_maker(variants) # maar van all_variants de header version van de variants\n",
    "grouped.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{vers}_ROI/final_df.csv\", index=False)\n",
    "\n",
    "grouped.sort_values([\"chromosome\", \"PS_tag\"]).head(20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e67d352-1f6b-4007-b188-a6635fa18e78",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1:1000552-1079306\n"
     ]
    }
   ],
   "source": [
    "bm_roi = run_switch(grouped, all_variants, samp, vers, all_variants)\n",
    "bm_roi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2651580-8aac-4b49-a98e-0abc701ae642",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "bm_roi.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{vers}_ROI/bm_roi_df.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d88291b-ef1b-497a-a4d8-5c3111f2f3d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "### ALLES HIERONDER IS CODE DIE IK EVT LATER NOG MOET GEBRUIKEN "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bb29c215-8c42-42f3-9f25-b42fb757fbc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "### ALGEMENE FUNCTIE\n",
    "\n",
    "def main(variants, ph_variants, goi, version, samp):\n",
    "    ### Make hapbloblocks in ROI\n",
    "    phased_hb = hb_maker(variants)\n",
    "    basis_df = phased_GOI(goi, phased_hb, samp)\n",
    "    \n",
    "    ### Run Switch Errors\n",
    "    bm_roi = switch_errors(basis_df, ph_variants, samp, version)\n",
    "\n",
    "    ### Format the switch error file\n",
    "    sw_e = pd.read_csv(bm_roi, sep=\"\\t\")\n",
    "    sw_e = sw_e[[\"dataset_name1\",\"het_variants0\",\"all_switches\",\"all_switch_rate\",\"all_switchflips\",\"all_switchflip_rate\",\"blockwise_hamming_rate\"]]\n",
    "    sw_e[\"Accuracy\"] = (100-sw_e[\"blockwise_hamming_rate\"]).astype(str)+\"%\"\n",
    "    sw_e.drop(columns=\"blockwise_hamming_rate\")\n",
    "    sw_e.rename(columns={\"dataset_name1\": \"GENE_l\"}, inplace=True)\n",
    "\n",
    "    ### Merge Hapoblocks with their switch errors\n",
    "    basis_df[\"GENE_l\"] = basis_df[\"GENE\"]+\"_\"+basis_df[\"HB_START\"].astype(str)    \n",
    "    switch_df = pd.merge(basis_df, sw_e, on=\"GENE_l\", how=\"left\")\n",
    "    switch_df[\"difference_het-vars_BM_vs_data\"] = abs(switch_df[\"PHASED_VARIANTS\"]-switch_df[\"het_variants0\"])\n",
    "    switch_df = switch_df.drop(columns=\"GENE_l\")\n",
    "\n",
    "    ### Write merged dataframe to file\n",
    "    switch_df.to_csv(f\"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG00{samp}/{version}_ROI/final_df.csv\", index=False)\n",
    "    \n",
    "    return basis_df, switch_df, sw_e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3aba2e6f-08c2-47af-b444-36972dbd6d43",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VCF filtered on PASS, QUAL >= 40 & ROI\n",
      "BRCA1_43183966 \t Done\n",
      "BRCA1_42861321 \t Done\n",
      "BRCA1_42808235 \t Done\n",
      "BRCA1_42607430 \t Done\n",
      "BRCA1_43322424 \t Done\n",
      "BRCA1_43010006 \t Done\n",
      "BRCA2_31865037 \t Done\n",
      "BRCA2_32712242 \t Done\n",
      "BRCA2_32095084 \t Done\n",
      "BRCA2_32180531 \t Done\n",
      "BRCA2_32508214 \t Done\n",
      "BRCA2_32602810 \t Done\n",
      "CFTR_117002433 \t Done\n",
      "CFTR_117777011 \t Done\n",
      "CRTAP_33204384 \t Done\n",
      "CRTAP_32635859 \t Done\n",
      "CRTAP_32934214 \t Done\n",
      "CYP21A2_32488753 \t Done\n",
      "CYP21A2_32000387 \t Done\n",
      "CYP21A2_31542190 \t Done\n",
      "CYP21A2_32515992 \t Done\n",
      "CYP21A2_32185426 \t Done\n",
      "HBA1_27803 \t Done\n",
      "HBB_4727925 \t Done\n",
      "HBB_5273948 \t Done\n",
      "HBB_5679844 \t Done\n",
      "MUSK_110240520 \t Done\n",
      "MUSK_110997014 \t Done\n",
      "PEX7_136636685 \t Done\n",
      "PEX7_136574402 \t Done\n",
      "POLG_88820301 \t Done\n",
      "POLG_89660113 \t Done\n",
      "SMN1_70485532 \t Done\n",
      "SMN1_71217981 \t Done\n",
      "SMN1_71016257 \t Done\n",
      "SMN1_71067872 \t Done\n",
      "SMN1_71317230 \t Done\n",
      "TSEN54_75022811 \t Done\n",
      "TSEN54_75977471 \t Done\n",
      "VCF filtered on ROI\n",
      "BRCA1_42607430 \t Done\n",
      "BRCA1_43183966 \t Done\n",
      "BRCA1_43240781 \t Done\n",
      "BRCA2_32712242 \t Done\n",
      "BRCA2_32602810 \t Done\n",
      "BRCA2_32508214 \t Done\n",
      "BRCA2_32180531 \t Done\n",
      "BRCA2_31865037 \t Done\n",
      "BRCA2_32095084 \t Done\n",
      "CFTR_117777011 \t Done\n",
      "CFTR_117002433 \t Done\n",
      "CRTAP_33204384 \t Done\n",
      "CRTAP_32934214 \t Done\n",
      "CRTAP_32635859 \t Done\n",
      "CYP21A2_31991875 \t Done\n",
      "CYP21A2_32185426 \t Done\n",
      "CYP21A2_32487383 \t Done\n",
      "CYP21A2_32501153 \t Done\n",
      "CYP21A2_31542190 \t Done\n",
      "HBA1_17889 \t Done\n",
      "HBB_4727925 \t Done\n",
      "HBB_5273948 \t Done\n",
      "MUSK_110239493 \t Done\n",
      "MUSK_110997014 \t Done\n",
      "PEX7_136636685 \t Done\n",
      "PEX7_136574402 \t Done\n",
      "POLG_88820301 \t Done\n",
      "POLG_89660113 \t Done\n",
      "SMN1_70463648 \t Done\n",
      "SMN1_70440650 \t Done\n",
      "SMN1_70559099 \t Done\n",
      "SMN1_70598701 \t Done\n",
      "SMN1_70707781 \t Done\n",
      "SMN1_70929044 \t Done\n",
      "TSEN54_75022811 \t Done\n",
      "TSEN54_75977471 \t Done\n",
      "RAW VCF from ONT filtered on ROI\n",
      "BRCA1_43240781 \t Done\n",
      "BRCA1_43183966 \t Done\n",
      "BRCA1_42607430 \t Done\n",
      "BRCA2_32602810 \t Done\n",
      "BRCA2_32712242 \t Done\n",
      "BRCA2_31865037 \t Done\n",
      "BRCA2_32508214 \t Done\n",
      "BRCA2_32180531 \t Done\n",
      "BRCA2_32095084 \t Done\n",
      "CFTR_117002433 \t Done\n",
      "CFTR_117777011 \t Done\n",
      "CRTAP_32934214 \t Done\n",
      "CRTAP_32635859 \t Done\n",
      "CRTAP_33204384 \t Done\n",
      "CYP21A2_32185426 \t Done\n",
      "CYP21A2_32487383 \t Done\n",
      "CYP21A2_32501153 \t Done\n",
      "CYP21A2_31991875 \t Done\n",
      "CYP21A2_31542190 \t Done\n",
      "HBA1_17889 \t Done\n",
      "HBB_5273948 \t Done\n",
      "HBB_4727925 \t Done\n",
      "MUSK_110239493 \t Done\n",
      "MUSK_110997014 \t Done\n",
      "PEX7_136636685 \t Done\n",
      "PEX7_136574402 \t Done\n",
      "POLG_88820301 \t Done\n",
      "POLG_89660113 \t Done\n",
      "SMN1_70440650 \t Done\n",
      "SMN1_70559099 \t Done\n",
      "SMN1_70598701 \t Done\n",
      "SMN1_70707781 \t Done\n",
      "SMN1_70929044 \t Done\n",
      "TSEN54_75022811 \t Done\n",
      "TSEN54_75977471 \t Done\n"
     ]
    }
   ],
   "source": [
    "\n",
    "### File Inputs\n",
    "\n",
    "# Regions of interest\n",
    "goi_file = \"/hpc/umc_laat/gvandersluis/data/Ref_HG/HG_annotation_ROI.bed\"\n",
    "\n",
    "## VCF Files\n",
    "\n",
    "# vcf filtered on qual > 40, PASS, and regions of interest\n",
    "filtered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI_nh.vcf\"\n",
    "ph_filtered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf\"\n",
    "\n",
    "# vcf filtered on regions of interest\n",
    "ununfiltered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/un_unphased_ROI_nh.vcf\"\n",
    "ph_ununfiltered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf\"\n",
    "\n",
    "\n",
    "# Raw VCF data, only filtered on ROI\n",
    "ONT = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/unphased_ROI/Sample_un_unfiltered_ROI_nh.vcf.gz\"\n",
    "ph_ONT = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/unphased_ROI/Sample_un_unfiltered_ROI.vcf.gz\"\n",
    "\n",
    "sample = \"2\"\n",
    "\n",
    "\n",
    "### RUN CODE\n",
    "print(\"VCF filtered on PASS, QUAL >= 40 & ROI\")\n",
    "filtered_Q_ROI = [filtered_v, ph_filtered_v, \"filtered\"]\n",
    "basis_f_q_roi, sw_f_q_roi, switch_f_q_roi = main(filtered_Q_ROI[0], filtered_Q_ROI[1], goi_file, filtered_Q_ROI[2], sample)\n",
    "\n",
    "print(\"VCF filtered on ROI\")\n",
    "filtered_ROI = [ununfiltered_v, ph_ununfiltered_v ,\"un_unfiltered\"]\n",
    "basis_f_roi, sw_f_roi, switch_f_roi = main(filtered_ROI[0], filtered_ROI[1], goi_file, filtered_ROI[2], sample)\n",
    "\n",
    "print(\"RAW VCF from ONT filtered on ROI\")\n",
    "RAW_ROI = [ONT, ph_ONT, \"unphased\"]\n",
    "raw_f_roi, sw_raw_f_roi, switch_raw_roi = main(RAW_ROI[0], RAW_ROI[1], goi_file, RAW_ROI[2], sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f6ef76f-ee3c-4a5a-ac43-8b2765790241",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:NIPD]",
   "language": "python",
   "name": "conda-env-NIPD-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
