{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aab540c1-4038-4035-97ff-5bc89b9837f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "### Imports\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "18a8216e-c644-48a9-8d8d-588e478e792c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### File Inputs\n",
    "\n",
    "# Regions of interest\n",
    "goi_file = \"/hpc/umc_laat/gvandersluis/data/Ref_HG/HG_annotation_ROI.bed\"\n",
    "\n",
    "## VCF Files\n",
    "\n",
    "# vcf filtered on qual > 40, PASS, and regions of interest\n",
    "filtered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI_nh.vcf\"\n",
    "\n",
    "# vcf filtered on regions of interest\n",
    "ununfiltered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/un_unphased_ROI_nh.vcf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ccb61cec-3e58-4151-a3d9-2f839348f557",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Base table code\n",
    "\n",
    "def hb_maker(phased_vars_file):\n",
    "    df_vcf = pd.read_csv(phased_vars_file, sep=\"\\t\", names=['CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT', \"sample\"])\n",
    "    df_vcf[\"GENE\"] = df_vcf['INFO'].str.split(\"|\").str[4]\n",
    "    df_splitted = df_vcf['sample'].str.split(\":\")\n",
    "    df_vcf[\"GT\"] = df_splitted.str[0]\n",
    "    df_vcf[\"DP\"] = df_splitted.str[2]\n",
    "    df_vcf[\"PS\"] = df_splitted.str[5]\n",
    " \n",
    "    # Min Quality of 10, this means there is no more than 1/10000 chance of an icorrect base call (>99.9% accuracy)\n",
    "    # & (pd.to_numeric(df_vcf['DP']) >= 20\n",
    "    phased = df_vcf.loc[df_vcf['PS'].notna(), ['CHROM','POS','QUAL','GENE','GT','DP','PS','REF','ALT']]\n",
    "    # phased[\"DP\"] = pd.to_numeric(phased[\"DP\"], errors=\"coerce\")\n",
    "    # phased = phased.sort_values(by=\"DP\")\n",
    "\n",
    "    grouped = phased.groupby(['PS']).agg({'POS': ['min','max', 'count'],\n",
    "                                                    'CHROM': 'first',\n",
    "                                                    'QUAL': 'mean',\n",
    "                                                    'GENE': 'unique'}).reset_index()\n",
    "    return grouped\n",
    "\n",
    "\n",
    "def phased_GOI(regions_file, phased_hps):\n",
    "    genes_of_interest = pd.read_csv(regions_file, sep=\"\\t\", names=[\"CHROM\",\"START\",\"END\",\"GENE\",\"INFO\",\"STRAND\"])\n",
    "    goi = genes_of_interest[\"GENE\"].unique()\n",
    "    genes_of_interest[\"POSITION\"] = genes_of_interest[\"CHROM\"].astype(str)+\":\"+genes_of_interest[\"START\"].astype(str)+\"-\"+genes_of_interest[\"END\"].astype(str)\n",
    "    #print(genes_of_interest)\n",
    "    # Make sure each GENE entry is a Python list\n",
    "    phased_hps[\"GENE\"] = phased_hps[(\"GENE\",\"unique\")].apply(lambda x: x.tolist())\n",
    "    #print(phased_hps)\n",
    "\n",
    "    # Filter rows where at least one gene in 'unique' is in GOI\n",
    "    filtered = phased_hps[phased_hps[(\"GENE\", \"unique\")].apply(lambda genes: any(g in goi for g in genes))].copy()\n",
    "\n",
    "    # Now, inside each row, keep only the genes that match the GOI\n",
    "    filtered[(\"GENE\", \"unique\")] = filtered[(\"GENE\", \"unique\")].apply(lambda genes: \", \".join([str(g) for g in genes if g in goi]))\n",
    "    # Remove sub columns\n",
    "    filtered.columns = ['_'.join([c for c in col if c]).strip() if isinstance(col, tuple) else col for col in filtered.columns.values]\n",
    "    # Now rename for clarity\n",
    "    filtered = filtered.rename(columns={'PS': 'PS',\n",
    "                                        'POS_min': 'POS_START',\n",
    "                                        'POS_max': 'POS_END',\n",
    "                                        'POS_count': 'PHASED_VARIANTS',\n",
    "                                        'CHROM_first': 'chromosome',\n",
    "                                        'QUAL_mean': 'QUAL_MEAN',\n",
    "                                        'GENE_unique': 'GENE'})\n",
    "\n",
    "    filtered[\"HB_LENGTH\"] = filtered[\"POS_END\"]-filtered[\"POS_START\"]\n",
    "\n",
    "    merged = pd.merge(filtered, genes_of_interest, on=\"GENE\", how=\"right\")[[\"GENE\",\"POSITION\",\"POS_START\",\"POS_END\",\"HB_LENGTH\",\"PHASED_VARIANTS\"]]\n",
    "   \n",
    "    int_cols = [\"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS\"]\n",
    "\n",
    "    for c in int_cols:\n",
    "        merged[c] = pd.to_numeric(merged[c], errors=\"coerce\").astype(\"Int64\")\n",
    "    merged = merged.astype(str).replace(\"<NA>\", \"NaN\")\n",
    "    return merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b51a6684-c5f6-4865-9f9c-47a753cfc4fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Code Calculate Switch Errors\n",
    "\n",
    "def switch_errors(bs_df, sample, version):\n",
    "    folder = \"/hpc/umc_laat/gvandersluis/data/\"\n",
    "    ph_vars = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/phased_ROI.vcf.gz\"\n",
    "    BM_vars = f\"{folder}Ont_data_nhung/HG00{sample}/HG00{sample}_BM_SSANDT_rn.vcf\"\n",
    "\n",
    "    BM_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/ROI_eval.tsv\"\n",
    "    open(BM_ROI, \"w\").close()\n",
    "\n",
    "\n",
    "    for idx, region in bs_df.iterrows():\n",
    "        print(region)\n",
    "        reg = region[\"POSITION\"].split(\":\")[0]+\":\"+region[\"POS_START\"]+\"-\"+region[\"POS_END\"]\n",
    "        print(reg)\n",
    "        gen = region[\"GENE\"]\n",
    "        if \"NaN\" not in reg:\n",
    "            out_vcf = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{gen}_BM.vcf\"\n",
    "            cmd = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", reg, BM_vars, \"-Oz\", \"-o\", out_vcf]\n",
    "            subprocess.run(cmd, check=True)\n",
    "            tsv_out = f\"{gen}_eval.tsv\"\n",
    "            cmd2 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/whatshap_v1.sif\", \"whatshap\", \"compare\",\"--tsv-pairwise\", tsv_out, \"--names\", f\"BENCHMARK,{gen}\", out_vcf, ph_vars]\n",
    "            subprocess.run(cmd2, check=True)\n",
    "            if os.path.getsize(BM_ROI) == 0:\n",
    "                with open(BM_ROI, \"w\") as out:\n",
    "                    with open(tsv_out, \"r\") as inp:\n",
    "                        out.write(inp.read())\n",
    "            else:\n",
    "                with open(BM_ROI, \"a\") as out:\n",
    "                    with open(tsv_out, \"r\") as inp:\n",
    "                        next(inp)\n",
    "                        out.write(inp.read())\n",
    "            os.remove(out_vcf)\n",
    "            os.remove(tsv_out)\n",
    "        print(gen, \"\\t Done\")\n",
    "    return BM_ROI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cf69877b-c61f-47b5-a1a8-955fd2a26c57",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VCF filtered on PASS, QUAL >= 40 & ROI\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  \n",
       "0              209  \n",
       "1              NaN  \n",
       "2               83  \n",
       "3              603  \n",
       "4              416  \n",
       "5              769  \n",
       "6              NaN  \n",
       "7               99  \n",
       "8              562  \n",
       "9              614  \n",
       "10               7  \n",
       "11             544  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Run Filtered Version\n",
    "\n",
    "print(\"VCF filtered on PASS, QUAL >= 40 & ROI\")\n",
    "phased_hb = hb_maker(filtered_v)\n",
    "basis_df = phased_GOI(goi_file, phased_hb)\n",
    "\n",
    "basis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7eca7e2d-152a-4759-a6ec-d77068c07920",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>PS</th>\n",
       "      <th colspan=\"3\" halign=\"left\">POS</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>count</th>\n",
       "      <th>first</th>\n",
       "      <th>mean</th>\n",
       "      <th>unique</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>110240520</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>769</td>\n",
       "      <td>chr9</td>\n",
       "      <td>56.340949</td>\n",
       "      <td>[TXN, LOC107987114, LOC107987114-TXNDC8, TXNDC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>110997014</td>\n",
       "      <td>110997014</td>\n",
       "      <td>111235428</td>\n",
       "      <td>271</td>\n",
       "      <td>chr9</td>\n",
       "      <td>56.029262</td>\n",
       "      <td>[LPAR1, LPAR1-RNY4P18, RNY4P18-LOC105376219, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117002433</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>416</td>\n",
       "      <td>chr7</td>\n",
       "      <td>57.192668</td>\n",
       "      <td>[ST7, MIR6132, ST7-AS2, ST7-OT3, LOC105375466,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>117777011</td>\n",
       "      <td>117777011</td>\n",
       "      <td>117970930</td>\n",
       "      <td>121</td>\n",
       "      <td>chr7</td>\n",
       "      <td>56.490083</td>\n",
       "      <td>[CTTNBP2, CTTNBP2-LOC105375470]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>136574402</td>\n",
       "      <td>136574402</td>\n",
       "      <td>136582242</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6</td>\n",
       "      <td>49.390000</td>\n",
       "      <td>[MAP3K5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>136636685</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>603</td>\n",
       "      <td>chr6</td>\n",
       "      <td>57.398756</td>\n",
       "      <td>[MAP3K5, LOC101928429, MAP3K5-PEX7, PEX7, SLC3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>27803</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>614</td>\n",
       "      <td>chr16</td>\n",
       "      <td>53.066221</td>\n",
       "      <td>[WASIR2, IL9RP3, IL9RP3-POLR3K, POLR3K, SNRNP2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>31542190</td>\n",
       "      <td>31542190</td>\n",
       "      <td>31906357</td>\n",
       "      <td>99</td>\n",
       "      <td>chr6</td>\n",
       "      <td>54.678687</td>\n",
       "      <td>[DDX39B, ATP6V1G2-DDX39B, NFKBIL1, LOC10028732...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>31865037</td>\n",
       "      <td>31865037</td>\n",
       "      <td>32011930</td>\n",
       "      <td>69</td>\n",
       "      <td>chr13</td>\n",
       "      <td>58.369420</td>\n",
       "      <td>[EEF1DP3, EEF1DP3-FRY-AS1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>32000387</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>83</td>\n",
       "      <td>chr6</td>\n",
       "      <td>57.614337</td>\n",
       "      <td>[C4A, TNXA, C4B, CYP21A2, TNXB, ATF6B, FKBPL, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>32095084</td>\n",
       "      <td>32095084</td>\n",
       "      <td>32123384</td>\n",
       "      <td>4</td>\n",
       "      <td>chr13</td>\n",
       "      <td>58.920000</td>\n",
       "      <td>[FRY]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>32180531</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>99</td>\n",
       "      <td>chr13</td>\n",
       "      <td>56.383434</td>\n",
       "      <td>[FRY, ZAR1L, BRCA2, N4BP2L1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>32185426</td>\n",
       "      <td>32185426</td>\n",
       "      <td>32486211</td>\n",
       "      <td>803</td>\n",
       "      <td>chr6</td>\n",
       "      <td>55.601357</td>\n",
       "      <td>[PBX2, NOTCH4, NOTCH4-TSBP1-AS1, TSBP1-AS1, TS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>32488753</td>\n",
       "      <td>32488753</td>\n",
       "      <td>32490582</td>\n",
       "      <td>13</td>\n",
       "      <td>chr6</td>\n",
       "      <td>43.536154</td>\n",
       "      <td>[HLA-DRB9-HLA-DRB5]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>32508214</td>\n",
       "      <td>32508214</td>\n",
       "      <td>32544937</td>\n",
       "      <td>4</td>\n",
       "      <td>chr13</td>\n",
       "      <td>60.080000</td>\n",
       "      <td>[N4BP2L2, N4BP2L2-IT2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>32515992</td>\n",
       "      <td>32515992</td>\n",
       "      <td>32540235</td>\n",
       "      <td>153</td>\n",
       "      <td>chr6</td>\n",
       "      <td>49.961046</td>\n",
       "      <td>[HLA-DRB5, HLA-DRB5-RNU1-61P]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>32602810</td>\n",
       "      <td>32602810</td>\n",
       "      <td>32665173</td>\n",
       "      <td>8</td>\n",
       "      <td>chr13</td>\n",
       "      <td>57.875000</td>\n",
       "      <td>[PDS5B]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>32635859</td>\n",
       "      <td>32635859</td>\n",
       "      <td>32894167</td>\n",
       "      <td>146</td>\n",
       "      <td>chr3</td>\n",
       "      <td>57.658151</td>\n",
       "      <td>[RPL30P4-CNOT10, CNOT10, CNOT10-RPL23AP43, RPL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>32712242</td>\n",
       "      <td>32712242</td>\n",
       "      <td>32834472</td>\n",
       "      <td>125</td>\n",
       "      <td>chr13</td>\n",
       "      <td>57.105840</td>\n",
       "      <td>[PDS5B, PDS5B-LINC00423, LINC00423]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>32934214</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>209</td>\n",
       "      <td>chr3</td>\n",
       "      <td>58.141627</td>\n",
       "      <td>[TRIM71-CCR4, GLB1, LOC107986072, LOC107986073...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>33204384</td>\n",
       "      <td>33204384</td>\n",
       "      <td>33628767</td>\n",
       "      <td>291</td>\n",
       "      <td>chr3</td>\n",
       "      <td>55.954433</td>\n",
       "      <td>[SUSD5, SUSD5-FBXL2, FBXL2, UBP1, UBP1-RNA5SP1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>42607430</td>\n",
       "      <td>42607430</td>\n",
       "      <td>42767578</td>\n",
       "      <td>71</td>\n",
       "      <td>chr17</td>\n",
       "      <td>58.045211</td>\n",
       "      <td>[TUBG1, TUBG1-ATP5MGP7, TUBG2, PLEKHH3, CNTNAP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>42808235</td>\n",
       "      <td>42808235</td>\n",
       "      <td>42811886</td>\n",
       "      <td>4</td>\n",
       "      <td>chr17</td>\n",
       "      <td>54.315000</td>\n",
       "      <td>[BECN1, CNTD1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>42861321</td>\n",
       "      <td>42861321</td>\n",
       "      <td>42921612</td>\n",
       "      <td>13</td>\n",
       "      <td>chr17</td>\n",
       "      <td>55.360769</td>\n",
       "      <td>[AOC3, LINC00671, G6PC1, G6PC1-RNY4P2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>43010006</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>7</td>\n",
       "      <td>chr17</td>\n",
       "      <td>56.210000</td>\n",
       "      <td>[IFI35, BRCA1, NBR2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>43183966</td>\n",
       "      <td>43183966</td>\n",
       "      <td>43231858</td>\n",
       "      <td>8</td>\n",
       "      <td>chr17</td>\n",
       "      <td>54.307500</td>\n",
       "      <td>[NBR1, TMEM106A, CCDC200]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>43322424</td>\n",
       "      <td>43322424</td>\n",
       "      <td>43607251</td>\n",
       "      <td>272</td>\n",
       "      <td>chr17</td>\n",
       "      <td>56.124081</td>\n",
       "      <td>[RNU2-1-LOC105371785, LOC105371785, LINC00910,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>4727925</td>\n",
       "      <td>4727925</td>\n",
       "      <td>5186352</td>\n",
       "      <td>783</td>\n",
       "      <td>chr11</td>\n",
       "      <td>57.396973</td>\n",
       "      <td>[MMP26, OR51F1, OR52R1, OR51F2, OR51S1, OR51H1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>5273948</td>\n",
       "      <td>5273948</td>\n",
       "      <td>5626809</td>\n",
       "      <td>1056</td>\n",
       "      <td>chr11</td>\n",
       "      <td>55.534706</td>\n",
       "      <td>[HBE1, HBE1-OR51AB1P, OR51AB1P, OR51AB1P-OR51B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>5679844</td>\n",
       "      <td>5679844</td>\n",
       "      <td>5708350</td>\n",
       "      <td>61</td>\n",
       "      <td>chr11</td>\n",
       "      <td>57.410328</td>\n",
       "      <td>[TRIM5, TRIM22]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>70485532</td>\n",
       "      <td>70485532</td>\n",
       "      <td>70537151</td>\n",
       "      <td>362</td>\n",
       "      <td>chr5</td>\n",
       "      <td>46.389033</td>\n",
       "      <td>[LOC441081, GUSBP15, LOC107986356]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>70929044</td>\n",
       "      <td>71217981</td>\n",
       "      <td>71217981</td>\n",
       "      <td>1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>41.820000</td>\n",
       "      <td>[GUSBP17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>71016257</td>\n",
       "      <td>71016257</td>\n",
       "      <td>71029957</td>\n",
       "      <td>22</td>\n",
       "      <td>chr5</td>\n",
       "      <td>46.763636</td>\n",
       "      <td>[NAIP]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>71067872</td>\n",
       "      <td>71067872</td>\n",
       "      <td>71139948</td>\n",
       "      <td>14</td>\n",
       "      <td>chr5</td>\n",
       "      <td>48.572857</td>\n",
       "      <td>[GTF2H2, OCLNP1, SNORD13B-2, CDH12P4-GUSBP9]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>71317230</td>\n",
       "      <td>71317230</td>\n",
       "      <td>71437569</td>\n",
       "      <td>65</td>\n",
       "      <td>chr5</td>\n",
       "      <td>54.846308</td>\n",
       "      <td>[LINC02197, LOC107986421, LOC105379026, LOC105...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>75022811</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>544</td>\n",
       "      <td>chr17</td>\n",
       "      <td>57.189007</td>\n",
       "      <td>[MRPL58, MRPL58-RNU6-362P, KCTD2, TRR-CCT1-1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>75977471</td>\n",
       "      <td>75977471</td>\n",
       "      <td>76019738</td>\n",
       "      <td>40</td>\n",
       "      <td>chr17</td>\n",
       "      <td>56.460500</td>\n",
       "      <td>[TEN1-CDK3, ACOX1, CDK3, TEN1, EVPL]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>88820301</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>562</td>\n",
       "      <td>chr15</td>\n",
       "      <td>56.616886</td>\n",
       "      <td>[ACAN, HAPLN3, MFGE8, MFGE8-KRT18P47, KRT18P47...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>89660113</td>\n",
       "      <td>89660113</td>\n",
       "      <td>89819867</td>\n",
       "      <td>66</td>\n",
       "      <td>chr15</td>\n",
       "      <td>57.554545</td>\n",
       "      <td>[KIF7, PLIN1, PEX11A, WDR93, MESP1, MRPL15P1-M...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           PS        POS                   CHROM       QUAL  \\\n",
       "                     min        max count  first       mean   \n",
       "0   110240520  110240520  110955599   769   chr9  56.340949   \n",
       "1   110997014  110997014  111235428   271   chr9  56.029262   \n",
       "2   117002433  117002433  117727876   416   chr7  57.192668   \n",
       "3   117777011  117777011  117970930   121   chr7  56.490083   \n",
       "4   136574402  136574402  136582242     2   chr6  49.390000   \n",
       "5   136636685  136636685  137361584   603   chr6  57.398756   \n",
       "6       27803      27803     646952   614  chr16  53.066221   \n",
       "7    31542190   31542190   31906357    99   chr6  54.678687   \n",
       "8    31865037   31865037   32011930    69  chr13  58.369420   \n",
       "9    32000387   32000387   32137224    83   chr6  57.614337   \n",
       "10   32095084   32095084   32123384     4  chr13  58.920000   \n",
       "11   32180531   32180531   32415657    99  chr13  56.383434   \n",
       "12   32185426   32185426   32486211   803   chr6  55.601357   \n",
       "13   32488753   32488753   32490582    13   chr6  43.536154   \n",
       "14   32508214   32508214   32544937     4  chr13  60.080000   \n",
       "15   32515992   32515992   32540235   153   chr6  49.961046   \n",
       "16   32602810   32602810   32665173     8  chr13  57.875000   \n",
       "17   32635859   32635859   32894167   146   chr3  57.658151   \n",
       "18   32712242   32712242   32834472   125  chr13  57.105840   \n",
       "19   32934214   32934214   33148829   209   chr3  58.141627   \n",
       "20   33204384   33204384   33628767   291   chr3  55.954433   \n",
       "21   42607430   42607430   42767578    71  chr17  58.045211   \n",
       "22   42808235   42808235   42811886     4  chr17  54.315000   \n",
       "23   42861321   42861321   42921612    13  chr17  55.360769   \n",
       "24   43010006   43010006   43132602     7  chr17  56.210000   \n",
       "25   43183966   43183966   43231858     8  chr17  54.307500   \n",
       "26   43322424   43322424   43607251   272  chr17  56.124081   \n",
       "27    4727925    4727925    5186352   783  chr11  57.396973   \n",
       "28    5273948    5273948    5626809  1056  chr11  55.534706   \n",
       "29    5679844    5679844    5708350    61  chr11  57.410328   \n",
       "30   70485532   70485532   70537151   362   chr5  46.389033   \n",
       "31   70929044   71217981   71217981     1   chr5  41.820000   \n",
       "32   71016257   71016257   71029957    22   chr5  46.763636   \n",
       "33   71067872   71067872   71139948    14   chr5  48.572857   \n",
       "34   71317230   71317230   71437569    65   chr5  54.846308   \n",
       "35   75022811   75022811   75870128   544  chr17  57.189007   \n",
       "36   75977471   75977471   76019738    40  chr17  56.460500   \n",
       "37   88820301   88820301   89569212   562  chr15  56.616886   \n",
       "38   89660113   89660113   89819867    66  chr15  57.554545   \n",
       "\n",
       "                                                 GENE  \n",
       "                                               unique  \n",
       "0   [TXN, LOC107987114, LOC107987114-TXNDC8, TXNDC...  \n",
       "1   [LPAR1, LPAR1-RNY4P18, RNY4P18-LOC105376219, L...  \n",
       "2   [ST7, MIR6132, ST7-AS2, ST7-OT3, LOC105375466,...  \n",
       "3                     [CTTNBP2, CTTNBP2-LOC105375470]  \n",
       "4                                            [MAP3K5]  \n",
       "5   [MAP3K5, LOC101928429, MAP3K5-PEX7, PEX7, SLC3...  \n",
       "6   [WASIR2, IL9RP3, IL9RP3-POLR3K, POLR3K, SNRNP2...  \n",
       "7   [DDX39B, ATP6V1G2-DDX39B, NFKBIL1, LOC10028732...  \n",
       "8                          [EEF1DP3, EEF1DP3-FRY-AS1]  \n",
       "9   [C4A, TNXA, C4B, CYP21A2, TNXB, ATF6B, FKBPL, ...  \n",
       "10                                              [FRY]  \n",
       "11                       [FRY, ZAR1L, BRCA2, N4BP2L1]  \n",
       "12  [PBX2, NOTCH4, NOTCH4-TSBP1-AS1, TSBP1-AS1, TS...  \n",
       "13                                [HLA-DRB9-HLA-DRB5]  \n",
       "14                             [N4BP2L2, N4BP2L2-IT2]  \n",
       "15                      [HLA-DRB5, HLA-DRB5-RNU1-61P]  \n",
       "16                                            [PDS5B]  \n",
       "17  [RPL30P4-CNOT10, CNOT10, CNOT10-RPL23AP43, RPL...  \n",
       "18                [PDS5B, PDS5B-LINC00423, LINC00423]  \n",
       "19  [TRIM71-CCR4, GLB1, LOC107986072, LOC107986073...  \n",
       "20  [SUSD5, SUSD5-FBXL2, FBXL2, UBP1, UBP1-RNA5SP1...  \n",
       "21  [TUBG1, TUBG1-ATP5MGP7, TUBG2, PLEKHH3, CNTNAP...  \n",
       "22                                     [BECN1, CNTD1]  \n",
       "23             [AOC3, LINC00671, G6PC1, G6PC1-RNY4P2]  \n",
       "24                               [IFI35, BRCA1, NBR2]  \n",
       "25                          [NBR1, TMEM106A, CCDC200]  \n",
       "26  [RNU2-1-LOC105371785, LOC105371785, LINC00910,...  \n",
       "27  [MMP26, OR51F1, OR52R1, OR51F2, OR51S1, OR51H1...  \n",
       "28  [HBE1, HBE1-OR51AB1P, OR51AB1P, OR51AB1P-OR51B...  \n",
       "29                                    [TRIM5, TRIM22]  \n",
       "30                 [LOC441081, GUSBP15, LOC107986356]  \n",
       "31                                          [GUSBP17]  \n",
       "32                                             [NAIP]  \n",
       "33       [GTF2H2, OCLNP1, SNORD13B-2, CDH12P4-GUSBP9]  \n",
       "34  [LINC02197, LOC107986421, LOC105379026, LOC105...  \n",
       "35  [MRPL58, MRPL58-RNU6-362P, KCTD2, TRR-CCT1-1, ...  \n",
       "36               [TEN1-CDK3, ACOX1, CDK3, TEN1, EVPL]  \n",
       "37  [ACAN, HAPLN3, MFGE8, MFGE8-KRT18P47, KRT18P47...  \n",
       "38  [KIF7, PLIN1, PEX11A, WDR93, MESP1, MRPL15P1-M...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phased_hb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06513a60-640c-4cac-8849-23ae714f9156",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0c8a2ca9-ecb4-4ff0-ad7e-e1e263d14a31",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'switch_errors' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mNameError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m bm_roi = \u001b[43mswitch_errors\u001b[49m(basis_df, \u001b[33m\"\u001b[39m\u001b[33m2\u001b[39m\u001b[33m\"\u001b[39m, \u001b[33m\"\u001b[39m\u001b[33mfiltered\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m      2\u001b[39m sw_e = pd.read_csv(bm_roi, sep=\u001b[33m\"\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[33m\"\u001b[39m)\n\u001b[32m      4\u001b[39m sw_e = sw_e[[\u001b[33m\"\u001b[39m\u001b[33mdataset_name1\u001b[39m\u001b[33m\"\u001b[39m,\u001b[33m\"\u001b[39m\u001b[33mall_switches\u001b[39m\u001b[33m\"\u001b[39m]]\n",
      "\u001b[31mNameError\u001b[39m: name 'switch_errors' is not defined"
     ]
    }
   ],
   "source": [
    "### Run Switch Errors\n",
    "\n",
    "bm_roi = switch_errors(basis_df, \"2\", \"filtered\")\n",
    "sw_e = pd.read_csv(bm_roi, sep=\"\\t\")\n",
    "\n",
    "sw_e = sw_e[[\"dataset_name1\",\"all_switches\"]]\n",
    "sw_e.rename(columns={\"dataset_name1\": \"GENE\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15a4e82e-0d45-4e7f-82d5-9ce80d9d4cf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "sw_e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e9f2e314-258d-4d73-b7e2-f4c16ef7d954",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              209           0.0  \n",
       "1              NaN           NaN  \n",
       "2               83           0.0  \n",
       "3              603           0.0  \n",
       "4              416           0.0  \n",
       "5              769           0.0  \n",
       "6              NaN           NaN  \n",
       "7               99           0.0  \n",
       "8              562           0.0  \n",
       "9              614           0.0  \n",
       "10               7           0.0  \n",
       "11             544           2.0  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "switch_df = pd.merge(basis_df, sw_e, on=\"GENE\", how=\"left\")\n",
    "switch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8115e8b2-b6ea-4eca-8579-d11ae64a5913",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d54588aa-9567-4d19-ab10-4b3d9702186a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VCF filtered on ROI\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  \n",
       "0              231  \n",
       "1              399  \n",
       "2              102  \n",
       "3              663  \n",
       "4              462  \n",
       "5              893  \n",
       "6              NaN  \n",
       "7              106  \n",
       "8              633  \n",
       "9              949  \n",
       "10             120  \n",
       "11             604  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Run Ununfiltered version\n",
    "\n",
    "print(\"VCF filtered on ROI\")\n",
    "phased_hb_uf = hb_maker(ununfiltered_v)\n",
    "basis_df_uf = phased_GOI(goi_file, phased_hb_uf)\n",
    "\n",
    "basis_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "297aacd2-6b34-4a18-9ea5-f753aaceb567",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENE                                CRTAP\n",
      "POSITION           chr3:32630864-33630864\n",
      "POS_START                        32934214\n",
      "POS_END                          33148829\n",
      "HB_LENGTH                          214615\n",
      "PHASED_VARIANTS                       231\n",
      "Name: 0, dtype: object\n",
      "chr3:32934214-33148829\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf\n",
      "      CRTAP = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr3 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       264 /       424\n",
      "              CRTAP:       907 /      1394\n",
      "              UNION:       913 /      1402\n",
      "       INTERSECTION:       258 /       416\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CRTAP:\n",
      "         common heterozygous variants:       258\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        99\n",
      "        non-singleton blocks in CRTAP:         1\n",
      "                 --> covered variants:       222\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        89\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        88\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        88\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "CRTAP \t Done\n",
      "GENE                                 SMN1\n",
      "POSITION           chr5:70439486-71439486\n",
      "POS_START                        70929044\n",
      "POS_END                          71437569\n",
      "HB_LENGTH                          508525\n",
      "PHASED_VARIANTS                       399\n",
      "Name: 1, dtype: object\n",
      "chr5:70929044-71437569\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf\n",
      "       SMN1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr5 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:        74 /       108\n",
      "               SMN1:      1931 /      2496\n",
      "              UNION:      1934 /      2499\n",
      "       INTERSECTION:        71 /       105\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> SMN1:\n",
      "         common heterozygous variants:        71\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:         7\n",
      "         non-singleton blocks in SMN1:         1\n",
      "                 --> covered variants:        64\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:         6\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:         5\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:         5\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "SMN1 \t Done\n",
      "GENE                              CYP21A2\n",
      "POSITION           chr6:31540309-32540309\n",
      "POS_START                        31991875\n",
      "POS_END                          32137224\n",
      "HB_LENGTH                          145349\n",
      "PHASED_VARIANTS                       102\n",
      "Name: 2, dtype: object\n",
      "chr6:31991875-32137224\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf\n",
      "    CYP21A2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       118 /       187\n",
      "            CYP21A2:      4048 /      7613\n",
      "              UNION:      4054 /      7620\n",
      "       INTERSECTION:       112 /       180\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CYP21A2:\n",
      "         common heterozygous variants:       112\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       112\n",
      "      non-singleton blocks in CYP21A2:         1\n",
      "                 --> covered variants:       102\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       102\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       101\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       101\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "CYP21A2 \t Done\n",
      "GENE                                   PEX7\n",
      "POSITION           chr6:136368573-137368573\n",
      "POS_START                         136636685\n",
      "POS_END                           137361584\n",
      "HB_LENGTH                            724899\n",
      "PHASED_VARIANTS                         663\n",
      "Name: 3, dtype: object\n",
      "chr6:136636685-137361584\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf\n",
      "       PEX7 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       731 /      1151\n",
      "               PEX7:      4048 /      7613\n",
      "              UNION:      4063 /      7629\n",
      "       INTERSECTION:       716 /      1135\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> PEX7:\n",
      "         common heterozygous variants:       716\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       639\n",
      "         non-singleton blocks in PEX7:         1\n",
      "                 --> covered variants:       635\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       570\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       569\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       569\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "PEX7 \t Done\n",
      "GENE                                   CFTR\n",
      "POSITION           chr7:117001545-118001545\n",
      "POS_START                         117002433\n",
      "POS_END                           117727876\n",
      "HB_LENGTH                            725443\n",
      "PHASED_VARIANTS                         462\n",
      "Name: 4, dtype: object\n",
      "chr7:117002433-117727876\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf\n",
      "       CFTR = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr7 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       514 /       707\n",
      "               CFTR:       727 /      1062\n",
      "              UNION:       738 /      1082\n",
      "       INTERSECTION:       503 /       687\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CFTR:\n",
      "         common heterozygous variants:       503\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       455\n",
      "         non-singleton blocks in CFTR:         1\n",
      "                 --> covered variants:       431\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       401\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       400\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       400\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "CFTR \t Done\n",
      "GENE                                   MUSK\n",
      "POSITION           chr9:110237668-111237668\n",
      "POS_START                         110239493\n",
      "POS_END                           110955599\n",
      "HB_LENGTH                            716106\n",
      "PHASED_VARIANTS                         893\n",
      "Name: 5, dtype: object\n",
      "chr9:110239493-110955599\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf\n",
      "       MUSK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr9 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:      1005 /      1346\n",
      "               MUSK:      1436 /      1979\n",
      "              UNION:      1466 /      2011\n",
      "       INTERSECTION:       975 /      1314\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> MUSK:\n",
      "         common heterozygous variants:       975\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       743\n",
      "         non-singleton blocks in MUSK:         1\n",
      "                 --> covered variants:       834\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       647\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       646\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       646\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "MUSK \t Done\n",
      "GENE                                 HBB\n",
      "POSITION           chr11:4727429-5727429\n",
      "POS_START                            NaN\n",
      "POS_END                              NaN\n",
      "HB_LENGTH                            NaN\n",
      "PHASED_VARIANTS                      NaN\n",
      "Name: 6, dtype: object\n",
      "chr11:NaN-NaN\n",
      "HBB \t Done\n",
      "GENE                                 BRCA2\n",
      "POSITION           chr13:31857677-32857677\n",
      "POS_START                         32180531\n",
      "POS_END                           32430770\n",
      "HB_LENGTH                           250239\n",
      "PHASED_VARIANTS                        106\n",
      "Name: 7, dtype: object\n",
      "chr13:32180531-32430770\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf\n",
      "      BRCA2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr13 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       119 /       318\n",
      "              BRCA2:       428 /      1167\n",
      "              UNION:       430 /      1175\n",
      "       INTERSECTION:       117 /       310\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA2:\n",
      "         common heterozygous variants:       117\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        93\n",
      "        non-singleton blocks in BRCA2:         1\n",
      "                 --> covered variants:       105\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        84\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        83\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        83\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "BRCA2 \t Done\n",
      "GENE                                  POLG\n",
      "POSITION           chr15:88820029-89820029\n",
      "POS_START                         88820301\n",
      "POS_END                           89569212\n",
      "HB_LENGTH                           748911\n",
      "PHASED_VARIANTS                        633\n",
      "Name: 8, dtype: object\n",
      "chr15:88820301-89569212\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf\n",
      "       POLG = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr15 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       666 /       985\n",
      "               POLG:       880 /      1774\n",
      "              UNION:       895 /      1796\n",
      "       INTERSECTION:       651 /       963\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> POLG:\n",
      "         common heterozygous variants:       651\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       528\n",
      "         non-singleton blocks in POLG:         1\n",
      "                 --> covered variants:       574\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       471\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       470\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       470\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "POLG \t Done\n",
      "GENE                         HBA1\n",
      "POSITION           chr16:1-677101\n",
      "POS_START                   17889\n",
      "POS_END                    648721\n",
      "HB_LENGTH                  630832\n",
      "PHASED_VARIANTS               949\n",
      "Name: 9, dtype: object\n",
      "chr16:17889-648721\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf\n",
      "       HBA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr16 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       544 /       992\n",
      "               HBA1:      1111 /      1897\n",
      "              UNION:      1136 /      1928\n",
      "       INTERSECTION:       519 /       961\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> HBA1:\n",
      "         common heterozygous variants:       519\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       354\n",
      "         non-singleton blocks in HBA1:         1\n",
      "                 --> covered variants:       464\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       320\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       319\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       319\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HBA1 \t Done\n",
      "GENE                                 BRCA1\n",
      "POSITION           chr17:42607268-43607268\n",
      "POS_START                         42607430\n",
      "POS_END                           43132602\n",
      "HB_LENGTH                           525172\n",
      "PHASED_VARIANTS                        120\n",
      "Name: 10, dtype: object\n",
      "chr17:42607430-43132602\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf\n",
      "      BRCA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       119 /       361\n",
      "              BRCA1:      1392 /      3033\n",
      "              UNION:      1396 /      3043\n",
      "       INTERSECTION:       115 /       351\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA1:\n",
      "         common heterozygous variants:       115\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       101\n",
      "        non-singleton blocks in BRCA1:         1\n",
      "                 --> covered variants:       100\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        88\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        87\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        87\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "BRCA1 \t Done\n",
      "GENE                                TSEN54\n",
      "POSITION           chr17:75020340-76020340\n",
      "POS_START                         75022811\n",
      "POS_END                           75907509\n",
      "HB_LENGTH                           884698\n",
      "PHASED_VARIANTS                        604\n",
      "Name: 11, dtype: object\n",
      "chr17:75022811-75907509\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf\n",
      "     TSEN54 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       679 /      1307\n",
      "             TSEN54:      1392 /      3033\n",
      "              UNION:      1412 /      3070\n",
      "       INTERSECTION:       659 /      1270\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> TSEN54:\n",
      "         common heterozygous variants:       659\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       521\n",
      "       non-singleton blocks in TSEN54:         1\n",
      "                 --> covered variants:       569\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       452\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       451\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.44%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.22%\n",
      "          Block-wise Hamming distance:         1\n",
      "      Block-wise Hamming distance [%]:     0.22%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       451\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.44%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.22%\n",
      "                     Hamming distance:         1\n",
      "                 Hamming distance [%]:     0.22%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "TSEN54 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              231           0.0  \n",
       "1              399           0.0  \n",
       "2              102           0.0  \n",
       "3              663           0.0  \n",
       "4              462           0.0  \n",
       "5              893           0.0  \n",
       "6              NaN           NaN  \n",
       "7              106           0.0  \n",
       "8              633           0.0  \n",
       "9              949           0.0  \n",
       "10             120           0.0  \n",
       "11             604           2.0  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bm_roi_uf = switch_errors(basis_df_uf, \"2\", \"un_unfiltered\")\n",
    "\n",
    "sw_e_uf = pd.read_csv(bm_roi_uf, sep=\"\\t\")\n",
    "\n",
    "sw_e_uf = sw_e_uf[[\"dataset_name1\",\"all_switches\"]]\n",
    "sw_e_uf.rename(columns={\"dataset_name1\": \"GENE\"}, inplace=True)\n",
    "\n",
    "switch_df_uf = pd.merge(basis_df_uf, sw_e_uf, on=\"GENE\", how=\"left\")\n",
    "switch_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f66ed712-dc40-48a1-a527-3a05f9eff80b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              231           0.0  \n",
       "1              399           0.0  \n",
       "2              102           0.0  \n",
       "3              663           0.0  \n",
       "4              462           0.0  \n",
       "5              893           0.0  \n",
       "6              NaN           NaN  \n",
       "7              106           0.0  \n",
       "8              633           0.0  \n",
       "9              949           0.0  \n",
       "10             120           0.0  \n",
       "11             604           2.0  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# int_cols = [\"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS\"]\n",
    "\n",
    "# for c in int_cols:\n",
    "#     switch_df_uf[c] = pd.to_numeric(switch_df_uf[c], errors=\"coerce\").astype(\"Int64\")\n",
    "# switch_df_uf = switch_df_uf.astype(str).replace(\"<NA>\", \"NaN\")\n",
    "\n",
    "\n",
    "switch_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0ac4e50a-4a38-4ca0-9fc2-5f5716d0afa0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              209           0.0  \n",
       "1              NaN           NaN  \n",
       "2               83           0.0  \n",
       "3              603           0.0  \n",
       "4              416           0.0  \n",
       "5              769           0.0  \n",
       "6              NaN           NaN  \n",
       "7               99           0.0  \n",
       "8              562           0.0  \n",
       "9              614           0.0  \n",
       "10               7           0.0  \n",
       "11             544           2.0  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "switch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "30690ba1-241f-4bd7-9f7f-a625d7401d15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#sample</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dataset_name0</th>\n",
       "      <th>dataset_name1</th>\n",
       "      <th>file_name0</th>\n",
       "      <th>file_name1</th>\n",
       "      <th>intersection_blocks</th>\n",
       "      <th>covered_variants</th>\n",
       "      <th>all_assessed_pairs</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>...</th>\n",
       "      <th>largestblock_switches</th>\n",
       "      <th>largestblock_switch_rate</th>\n",
       "      <th>largestblock_switchflips</th>\n",
       "      <th>largestblock_switchflip_rate</th>\n",
       "      <th>largestblock_hamming</th>\n",
       "      <th>largestblock_hamming_rate</th>\n",
       "      <th>largestblock_diff_genotypes</th>\n",
       "      <th>largestblock_diff_genotypes_rate</th>\n",
       "      <th>het_variants0</th>\n",
       "      <th>only_snvs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr3</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CRTAP</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>3</td>\n",
       "      <td>646</td>\n",
       "      <td>643</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>738</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr5</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>4</td>\n",
       "      <td>463</td>\n",
       "      <td>459</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>489</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr6</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>5</td>\n",
       "      <td>1151</td>\n",
       "      <td>1146</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1244</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr6</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>PEX7</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>605</td>\n",
       "      <td>603</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>663</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr7</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>537</td>\n",
       "      <td>535</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>610</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr9</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>MUSK</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>1040</td>\n",
       "      <td>1038</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1181</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr11</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>HBB</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>3</td>\n",
       "      <td>1900</td>\n",
       "      <td>1897</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2075</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr13</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>6</td>\n",
       "      <td>309</td>\n",
       "      <td>303</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>354</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr15</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>POLG</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>628</td>\n",
       "      <td>626</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>707</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr16</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>HBA1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>614</td>\n",
       "      <td>613</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>674</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr17</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>6</td>\n",
       "      <td>375</td>\n",
       "      <td>369</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>406</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr17</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>TSEN54</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>584</td>\n",
       "      <td>582</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>656</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #sample chromosome dataset_name0 dataset_name1  \\\n",
       "0    hg002       chr3     BENCHMARK         CRTAP   \n",
       "1    hg002       chr5     BENCHMARK          SMN1   \n",
       "2    hg002       chr6     BENCHMARK       CYP21A2   \n",
       "3    hg002       chr6     BENCHMARK          PEX7   \n",
       "4    hg002       chr7     BENCHMARK          CFTR   \n",
       "5    hg002       chr9     BENCHMARK          MUSK   \n",
       "6    hg002      chr11     BENCHMARK           HBB   \n",
       "7    hg002      chr13     BENCHMARK         BRCA2   \n",
       "8    hg002      chr15     BENCHMARK          POLG   \n",
       "9    hg002      chr16     BENCHMARK          HBA1   \n",
       "10   hg002      chr17     BENCHMARK         BRCA1   \n",
       "11   hg002      chr17     BENCHMARK        TSEN54   \n",
       "\n",
       "                                           file_name0  \\\n",
       "0   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "1   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "2   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "3   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "4   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "5   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "6   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "7   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "8   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "9   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "10  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "11  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "\n",
       "                                           file_name1  intersection_blocks  \\\n",
       "0   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    3   \n",
       "1   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    4   \n",
       "2   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    5   \n",
       "3   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "4   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "5   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "6   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    3   \n",
       "7   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    6   \n",
       "8   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "9   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "10  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    6   \n",
       "11  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "\n",
       "    covered_variants  all_assessed_pairs  all_switches  ...  \\\n",
       "0                646                 643             0  ...   \n",
       "1                463                 459             0  ...   \n",
       "2               1151                1146             0  ...   \n",
       "3                605                 603             0  ...   \n",
       "4                537                 535             0  ...   \n",
       "5               1040                1038             0  ...   \n",
       "6               1900                1897             0  ...   \n",
       "7                309                 303             0  ...   \n",
       "8                628                 626             0  ...   \n",
       "9                614                 613             0  ...   \n",
       "10               375                 369             0  ...   \n",
       "11               584                 582             0  ...   \n",
       "\n",
       "    largestblock_switches largestblock_switch_rate  largestblock_switchflips  \\\n",
       "0                       0                      0.0                       0/0   \n",
       "1                       0                      0.0                       0/0   \n",
       "2                       0                      0.0                       0/0   \n",
       "3                       0                      0.0                       0/0   \n",
       "4                       0                      0.0                       0/0   \n",
       "5                       0                      0.0                       0/0   \n",
       "6                       0                      0.0                       0/0   \n",
       "7                       0                      0.0                       0/0   \n",
       "8                       0                      0.0                       0/0   \n",
       "9                       0                      0.0                       0/0   \n",
       "10                      0                      0.0                       0/0   \n",
       "11                      0                      0.0                       0/0   \n",
       "\n",
       "    largestblock_switchflip_rate  largestblock_hamming  \\\n",
       "0                            0.0                     0   \n",
       "1                            0.0                     0   \n",
       "2                            0.0                     0   \n",
       "3                            0.0                     0   \n",
       "4                            0.0                     0   \n",
       "5                            0.0                     0   \n",
       "6                            0.0                     0   \n",
       "7                            0.0                     0   \n",
       "8                            0.0                     0   \n",
       "9                            0.0                     0   \n",
       "10                           0.0                     0   \n",
       "11                           0.0                     0   \n",
       "\n",
       "    largestblock_hamming_rate  largestblock_diff_genotypes  \\\n",
       "0                         0.0                            0   \n",
       "1                         0.0                            0   \n",
       "2                         0.0                            0   \n",
       "3                         0.0                            0   \n",
       "4                         0.0                            0   \n",
       "5                         0.0                            0   \n",
       "6                         0.0                            0   \n",
       "7                         0.0                            0   \n",
       "8                         0.0                            0   \n",
       "9                         0.0                            0   \n",
       "10                        0.0                            0   \n",
       "11                        0.0                            0   \n",
       "\n",
       "    largestblock_diff_genotypes_rate  het_variants0  only_snvs  \n",
       "0                                0.0            738          0  \n",
       "1                                0.0            489          0  \n",
       "2                                0.0           1244          0  \n",
       "3                                0.0            663          0  \n",
       "4                                0.0            610          0  \n",
       "5                                0.0           1181          0  \n",
       "6                                0.0           2075          0  \n",
       "7                                0.0            354          0  \n",
       "8                                0.0            707          0  \n",
       "9                                0.0            674          0  \n",
       "10                               0.0            406          0  \n",
       "11                               0.0            656          0  \n",
       "\n",
       "[12 rows x 28 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(bm_roi, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "128bbbcc-805f-4dd5-bcd6-38d3592d20fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#sample</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dataset_name0</th>\n",
       "      <th>dataset_name1</th>\n",
       "      <th>file_name0</th>\n",
       "      <th>file_name1</th>\n",
       "      <th>intersection_blocks</th>\n",
       "      <th>covered_variants</th>\n",
       "      <th>all_assessed_pairs</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>...</th>\n",
       "      <th>largestblock_switches</th>\n",
       "      <th>largestblock_switch_rate</th>\n",
       "      <th>largestblock_switchflips</th>\n",
       "      <th>largestblock_switchflip_rate</th>\n",
       "      <th>largestblock_hamming</th>\n",
       "      <th>largestblock_hamming_rate</th>\n",
       "      <th>largestblock_diff_genotypes</th>\n",
       "      <th>largestblock_diff_genotypes_rate</th>\n",
       "      <th>het_variants0</th>\n",
       "      <th>only_snvs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr3</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CRTAP</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>3</td>\n",
       "      <td>706</td>\n",
       "      <td>703</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>907</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr5</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>5</td>\n",
       "      <td>1588</td>\n",
       "      <td>1583</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1931</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr6</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>5</td>\n",
       "      <td>2858</td>\n",
       "      <td>2853</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3236</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr6</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>PEX7</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>666</td>\n",
       "      <td>664</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>812</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr7</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>593</td>\n",
       "      <td>591</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>727</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr9</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>MUSK</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>1187</td>\n",
       "      <td>1185</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1436</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr11</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>HBB</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>2041</td>\n",
       "      <td>2039</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2316</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr13</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>6</td>\n",
       "      <td>342</td>\n",
       "      <td>336</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>428</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr15</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>POLG</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>704</td>\n",
       "      <td>702</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>880</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr16</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>HBA1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>1</td>\n",
       "      <td>949</td>\n",
       "      <td>948</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1111</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr17</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>3</td>\n",
       "      <td>462</td>\n",
       "      <td>459</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>585</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>hg002</td>\n",
       "      <td>chr17</td>\n",
       "      <td>BENCHMARK</td>\n",
       "      <td>TSEN54</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>/hpc/umc_laat/gvandersluis/data/Ont_data_nhung...</td>\n",
       "      <td>2</td>\n",
       "      <td>646</td>\n",
       "      <td>644</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>807</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #sample chromosome dataset_name0 dataset_name1  \\\n",
       "0    hg002       chr3     BENCHMARK         CRTAP   \n",
       "1    hg002       chr5     BENCHMARK          SMN1   \n",
       "2    hg002       chr6     BENCHMARK       CYP21A2   \n",
       "3    hg002       chr6     BENCHMARK          PEX7   \n",
       "4    hg002       chr7     BENCHMARK          CFTR   \n",
       "5    hg002       chr9     BENCHMARK          MUSK   \n",
       "6    hg002      chr11     BENCHMARK           HBB   \n",
       "7    hg002      chr13     BENCHMARK         BRCA2   \n",
       "8    hg002      chr15     BENCHMARK          POLG   \n",
       "9    hg002      chr16     BENCHMARK          HBA1   \n",
       "10   hg002      chr17     BENCHMARK         BRCA1   \n",
       "11   hg002      chr17     BENCHMARK        TSEN54   \n",
       "\n",
       "                                           file_name0  \\\n",
       "0   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "1   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "2   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "3   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "4   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "5   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "6   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "7   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "8   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "9   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "10  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "11  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...   \n",
       "\n",
       "                                           file_name1  intersection_blocks  \\\n",
       "0   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    3   \n",
       "1   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    5   \n",
       "2   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    5   \n",
       "3   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "4   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "5   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "6   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "7   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    6   \n",
       "8   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "9   /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    1   \n",
       "10  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    3   \n",
       "11  /hpc/umc_laat/gvandersluis/data/Ont_data_nhung...                    2   \n",
       "\n",
       "    covered_variants  all_assessed_pairs  all_switches  ...  \\\n",
       "0                706                 703             0  ...   \n",
       "1               1588                1583             0  ...   \n",
       "2               2858                2853             0  ...   \n",
       "3                666                 664             0  ...   \n",
       "4                593                 591             0  ...   \n",
       "5               1187                1185             0  ...   \n",
       "6               2041                2039             0  ...   \n",
       "7                342                 336             0  ...   \n",
       "8                704                 702             0  ...   \n",
       "9                949                 948             0  ...   \n",
       "10               462                 459             0  ...   \n",
       "11               646                 644             0  ...   \n",
       "\n",
       "    largestblock_switches largestblock_switch_rate  largestblock_switchflips  \\\n",
       "0                       0                      0.0                       0/0   \n",
       "1                       0                      0.0                       0/0   \n",
       "2                       0                      0.0                       0/0   \n",
       "3                       0                      0.0                       0/0   \n",
       "4                       0                      0.0                       0/0   \n",
       "5                       0                      0.0                       0/0   \n",
       "6                       0                      0.0                       0/0   \n",
       "7                       0                      0.0                       0/0   \n",
       "8                       0                      0.0                       0/0   \n",
       "9                       0                      0.0                       0/0   \n",
       "10                      0                      0.0                       0/0   \n",
       "11                      0                      0.0                       0/0   \n",
       "\n",
       "    largestblock_switchflip_rate  largestblock_hamming  \\\n",
       "0                            0.0                     0   \n",
       "1                            0.0                     0   \n",
       "2                            0.0                     0   \n",
       "3                            0.0                     0   \n",
       "4                            0.0                     0   \n",
       "5                            0.0                     0   \n",
       "6                            0.0                     0   \n",
       "7                            0.0                     0   \n",
       "8                            0.0                     0   \n",
       "9                            0.0                     0   \n",
       "10                           0.0                     0   \n",
       "11                           0.0                     0   \n",
       "\n",
       "    largestblock_hamming_rate  largestblock_diff_genotypes  \\\n",
       "0                         0.0                            0   \n",
       "1                         0.0                            0   \n",
       "2                         0.0                            0   \n",
       "3                         0.0                            0   \n",
       "4                         0.0                            0   \n",
       "5                         0.0                            0   \n",
       "6                         0.0                            0   \n",
       "7                         0.0                            0   \n",
       "8                         0.0                            0   \n",
       "9                         0.0                            0   \n",
       "10                        0.0                            0   \n",
       "11                        0.0                            0   \n",
       "\n",
       "    largestblock_diff_genotypes_rate  het_variants0  only_snvs  \n",
       "0                                0.0            907          0  \n",
       "1                                0.0           1931          0  \n",
       "2                                0.0           3236          0  \n",
       "3                                0.0            812          0  \n",
       "4                                0.0            727          0  \n",
       "5                                0.0           1436          0  \n",
       "6                                0.0           2316          0  \n",
       "7                                0.0            428          0  \n",
       "8                                0.0            880          0  \n",
       "9                                0.0           1111          0  \n",
       "10                               0.0            585          0  \n",
       "11                               0.0            807          0  \n",
       "\n",
       "[12 rows x 28 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(bm_roi_uf, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "148fabd3-5e9c-485b-8484-95707f61a3d6",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (2934525902.py, line 4)",
     "output_type": "error",
     "traceback": [
      "  \u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 4\u001b[39m\n\u001b[31m    \u001b[39m\u001b[31mbcftools view -r 'chr3:32934214-33148829' phased_ROI.vcf.gz | grep -E '0\\|1|1\\|1|1\\|0|0\\|0' | wc\u001b[39m\n             ^\n\u001b[31mSyntaxError\u001b[39m\u001b[31m:\u001b[39m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "### Interesting Code Lines\n",
    "\n",
    "# find how many phased variant per selected region\n",
    "bcftools view -r 'chr3:32934214-33148829' phased_ROI.vcf.gz | grep -E '0\\|1|1\\|1|1\\|0|0\\|0' | wc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef0633cf-83c8-4a9e-a14f-73f68d231953",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:NIPD]",
   "language": "python",
   "name": "conda-env-NIPD-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
