{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "aab540c1-4038-4035-97ff-5bc89b9837f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "\n",
    "### Imports\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18a8216e-c644-48a9-8d8d-588e478e792c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### File Inputs\n",
    "\n",
    "# Regions of interest\n",
    "goi_file = \"/hpc/umc_laat/gvandersluis/data/Ref_HG/HG_annotation_ROI.bed\"\n",
    "\n",
    "## VCF Files\n",
    "\n",
    "# vcf filtered on qual > 40, PASS, and regions of interest\n",
    "filtered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI_nh.vcf\"\n",
    "\n",
    "# vcf filtered on regions of interest\n",
    "ununfiltered_v = \"/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/un_unphased_ROI_nh.vcf\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ccb61cec-3e58-4151-a3d9-2f839348f557",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Base table code\n",
    "\n",
    "def hb_maker(phased_vars_file):\n",
    "    df_vcf = pd.read_csv(phased_vars_file, sep=\"\\t\", names=['CHROM','POS','ID','REF','ALT','QUAL','FILTER','INFO','FORMAT', \"sample\"])\n",
    "    df_vcf[\"GENE\"] = df_vcf['INFO'].str.split(\"|\").str[4]\n",
    "    df_splitted = df_vcf['sample'].str.split(\":\")\n",
    "    df_vcf[\"GT\"] = df_splitted.str[0]\n",
    "    df_vcf[\"DP\"] = df_splitted.str[2]\n",
    "    df_vcf[\"PS\"] = df_splitted.str[5]\n",
    "    # Min Quality of 10, this means there is no more than 1/10000 chance of an incorrect base call (>99.9% accuracy)\n",
    "    # & (pd.to_numeric(df_vcf['DP']) >= 20\n",
    "    phased = df_vcf.loc[df_vcf['PS'].notna(), ['CHROM','POS','QUAL','GENE','GT','DP','PS','REF','ALT']]\n",
    "    # phased[\"DP\"] = pd.to_numeric(phased[\"DP\"], errors=\"coerce\")\n",
    "    # phased = phased.sort_values(by=\"DP\")\n",
    "\n",
    "\n",
    "    grouped = phased.groupby(['PS']).agg({'POS': ['min','max', 'count'],\n",
    "                                                    'CHROM': 'first',\n",
    "                                                    'QUAL': 'mean',\n",
    "                                                    'GENE': 'unique'}).reset_index()\n",
    "\n",
    "    return grouped\n",
    "\n",
    "\n",
    "def phased_GOI(regions_file, phased_hps):\n",
    "    genes_of_interest = pd.read_csv(regions_file, sep=\"\\t\", names=[\"CHROM\",\"START\",\"END\",\"GENE\",\"INFO\",\"STRAND\"])\n",
    "    goi = genes_of_interest[\"GENE\"].unique()\n",
    "    genes_of_interest[\"POSITION\"] = genes_of_interest[\"CHROM\"].astype(str)+\":\"+genes_of_interest[\"START\"].astype(str)+\"-\"+genes_of_interest[\"END\"].astype(str)\n",
    "    #print(genes_of_interest)\n",
    "    # Make sure each GENE entry is a Python list\n",
    "    phased_hps[\"GENE\"] = phased_hps[(\"GENE\",\"unique\")].apply(lambda x: x.tolist())\n",
    "    #print(phased_hps)\n",
    "\n",
    "    # Filter rows where at least one gene in 'unique' is in GOI\n",
    "    filtered = phased_hps[phased_hps[(\"GENE\", \"unique\")].apply(lambda genes: any(g in goi for g in genes))].copy()\n",
    "\n",
    "    # Now, inside each row, keep only the genes that match the GOI\n",
    "    filtered[(\"GENE\", \"unique\")] = filtered[(\"GENE\", \"unique\")].apply(lambda genes: \", \".join([str(g) for g in genes if g in goi]))\n",
    "    # Remove sub columns\n",
    "    filtered.columns = ['_'.join([c for c in col if c]).strip() if isinstance(col, tuple) else col for col in filtered.columns.values]\n",
    "    # Now rename for clarity\n",
    "    filtered = filtered.rename(columns={'PS': 'PS',\n",
    "                                        'POS_min': 'POS_START',\n",
    "                                        'POS_max': 'POS_END',\n",
    "                                        'POS_count': 'PHASED_VARIANTS',\n",
    "                                        'CHROM_first': 'chromosome',\n",
    "                                        'QUAL_mean': 'QUAL_MEAN',\n",
    "                                        'GENE_unique': 'GENE'})\n",
    "\n",
    "    filtered[\"HB_LENGTH\"] = filtered[\"POS_END\"]-filtered[\"POS_START\"]\n",
    "\n",
    "    merged = pd.merge(filtered, genes_of_interest, on=\"GENE\", how=\"right\")[[\"GENE\",\"POSITION\",\"POS_START\",\"POS_END\",\"HB_LENGTH\",\"PHASED_VARIANTS\"]]\n",
    "    int_cols = [\"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS\"]\n",
    "\n",
    "    for c in int_cols:\n",
    "        merged[c] = pd.to_numeric(merged[c], errors=\"coerce\").astype(\"Int64\")\n",
    "    merged = merged.astype(str).replace(\"<NA>\", \"NaN\")\n",
    "    return merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "b51a6684-c5f6-4865-9f9c-47a753cfc4fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Code Calculate Switch Errors\n",
    "\n",
    "def switch_errors(bs_df, sample, version):\n",
    "    folder = \"/hpc/umc_laat/gvandersluis/data/\"\n",
    "    ph_vars = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/phased_ROI.vcf.gz\"\n",
    "    BM_vars = f\"{folder}Ont_data_nhung/HG00{sample}/HG00{sample}_BM_SSANDT_rn.vcf\"\n",
    "\n",
    "    BM_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/ROI_eval.tsv\"\n",
    "    open(BM_ROI, \"w\").close()\n",
    "    switches_ROI = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/switches_ROI.bed\"\n",
    "    open(switches_ROI, \"w\").close()\n",
    "\n",
    "    for idx, region in bs_df.iterrows():\n",
    "        reg = region[\"POSITION\"].split(\":\")[0]+\":\"+region[\"POS_START\"]+\"-\"+region[\"POS_END\"]\n",
    "        gen = region[\"GENE\"]\n",
    "        if \"NaN\" not in reg:\n",
    "            out_vcf = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{gen}_BM.vcf\"\n",
    "            cmd = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/bcftools_v1.9-1-deb_cv1.sif\", \"bcftools\", \"view\", \"-r\", reg, BM_vars, \"-Oz\", \"-o\", out_vcf]\n",
    "            subprocess.run(cmd, check=True)\n",
    "            tsv_out = f\"{gen}_eval.tsv\"\n",
    "            bed_out = f\"{folder}Ont_data_nhung/HG00{sample}/{version}_ROI/{gen}_switch.bed\"\n",
    "            cmd2 = [\"apptainer\", \"exec\", \"-B\", \"/hpc/:/hpc/\", \"/hpc/umc_laat/gvandersluis/software/whatshap_v1.sif\", \"whatshap\", \"compare\",\"--switch-error-bed\",bed_out,\"--tsv-pairwise\", tsv_out, \"--names\", f\"BENCHMARK,{gen}\", out_vcf, ph_vars]\n",
    "            subprocess.run(cmd2, check=True)\n",
    "            print(\"HII\")\n",
    "            print(open(bed_out, \"r\").read())\n",
    "            if os.path.getsize(BM_ROI) == 0:\n",
    "                with open(BM_ROI, \"w\") as out:\n",
    "                    with open(tsv_out, \"r\") as inp:\n",
    "                        out.write(inp.read())\n",
    "            else:\n",
    "                with open(BM_ROI, \"a\") as out:\n",
    "                    with open(tsv_out, \"r\") as inp:\n",
    "                        next(inp)\n",
    "                        out.write(inp.read())\n",
    "            #if os.path.getsize(switches_ROI) == 0 and os.path.getsize(bed_out) != 0:\n",
    "            with open(switches_ROI, \"a+\") as out_b:\n",
    "                with open(bed_out, \"r\") as inp_b:\n",
    "                    out_b.write(inp_b.read())\n",
    "            os.remove(out_vcf)\n",
    "            os.remove(tsv_out)\n",
    "            os.remove(bed_out)\n",
    "        print(gen, \"\\t Done\")\n",
    "    return BM_ROI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "cf69877b-c61f-47b5-a1a8-955fd2a26c57",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VCF filtered on PASS, QUAL >= 40 & ROI\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  \n",
       "0              209  \n",
       "1              NaN  \n",
       "2               83  \n",
       "3              603  \n",
       "4              416  \n",
       "5              769  \n",
       "6              NaN  \n",
       "7               99  \n",
       "8              562  \n",
       "9              614  \n",
       "10               7  \n",
       "11             544  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Run Filtered Version\n",
    "\n",
    "print(\"VCF filtered on PASS, QUAL >= 40 & ROI\")\n",
    "phased_hb = hb_maker(filtered_v)\n",
    "basis_df = phased_GOI(goi_file, phased_hb)\n",
    "\n",
    "basis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "0c8a2ca9-ecb4-4ff0-ad7e-e1e263d14a31",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CRTAP_BM.vcf\n",
      "      CRTAP = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr3 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       264 /       424\n",
      "              CRTAP:       738 /       774\n",
      "              UNION:       773 /       957\n",
      "       INTERSECTION:       229 /       241\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CRTAP:\n",
      "         common heterozygous variants:       229\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        90\n",
      "        non-singleton blocks in CRTAP:         1\n",
      "                 --> covered variants:       204\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        83\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        82\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        82\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CRTAP \t Done\n",
      "SMN1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CYP21A2_BM.vcf\n",
      "    CYP21A2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       115 /       178\n",
      "            CYP21A2:      1907 /      2282\n",
      "              UNION:      1935 /      2368\n",
      "       INTERSECTION:        87 /        92\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CYP21A2:\n",
      "         common heterozygous variants:        87\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        87\n",
      "      non-singleton blocks in CYP21A2:         1\n",
      "                 --> covered variants:        83\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        83\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        82\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        82\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CYP21A2 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/PEX7_BM.vcf\n",
      "       PEX7 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       731 /      1151\n",
      "               PEX7:      1907 /      2282\n",
      "              UNION:      1998 /      2748\n",
      "       INTERSECTION:       640 /       685\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> PEX7:\n",
      "         common heterozygous variants:       640\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       576\n",
      "         non-singleton blocks in PEX7:         1\n",
      "                 --> covered variants:       593\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       534\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       533\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       533\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "PEX7 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/CFTR_BM.vcf\n",
      "       CFTR = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr7 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       514 /       707\n",
      "               CFTR:       610 /       644\n",
      "              UNION:       673 /       881\n",
      "       INTERSECTION:       451 /       470\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CFTR:\n",
      "         common heterozygous variants:       451\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       416\n",
      "         non-singleton blocks in CFTR:         1\n",
      "                 --> covered variants:       402\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       376\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       375\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       375\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CFTR \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/MUSK_BM.vcf\n",
      "       MUSK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr9 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:      1004 /      1345\n",
      "               MUSK:      1181 /      1243\n",
      "              UNION:      1348 /      1709\n",
      "       INTERSECTION:       837 /       879\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> MUSK:\n",
      "         common heterozygous variants:       837\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       644\n",
      "         non-singleton blocks in MUSK:         1\n",
      "                 --> covered variants:       737\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       575\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       574\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       574\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "MUSK \t Done\n",
      "HBB \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA2_BM.vcf\n",
      "      BRCA2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr13 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       118 /       309\n",
      "              BRCA2:       354 /       415\n",
      "              UNION:       367 /       606\n",
      "       INTERSECTION:       105 /       118\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA2:\n",
      "         common heterozygous variants:       105\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        84\n",
      "        non-singleton blocks in BRCA2:         1\n",
      "                 --> covered variants:        98\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        79\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        78\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        78\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "BRCA2 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/POLG_BM.vcf\n",
      "       POLG = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr15 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       666 /       985\n",
      "               POLG:       707 /       798\n",
      "              UNION:       798 /      1173\n",
      "       INTERSECTION:       575 /       610\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> POLG:\n",
      "         common heterozygous variants:       575\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       469\n",
      "         non-singleton blocks in POLG:         1\n",
      "                 --> covered variants:       528\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       433\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       432\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       432\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "POLG \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/HBA1_BM.vcf\n",
      "       HBA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr16 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       543 /       989\n",
      "               HBA1:       674 /       734\n",
      "              UNION:       791 /      1269\n",
      "       INTERSECTION:       426 /       454\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> HBA1:\n",
      "         common heterozygous variants:       426\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       309\n",
      "         non-singleton blocks in HBA1:         1\n",
      "                 --> covered variants:       388\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       282\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       281\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       281\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "HBA1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/BRCA1_BM.vcf\n",
      "      BRCA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:        12 /        27\n",
      "              BRCA1:      1062 /      1188\n",
      "              UNION:      1066 /      1205\n",
      "       INTERSECTION:         8 /        10\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA1:\n",
      "         common heterozygous variants:         8\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:         6\n",
      "        non-singleton blocks in BRCA1:         1\n",
      "                 --> covered variants:         7\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:         5\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:         4\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:         4\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "BRCA1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/TSEN54_BM.vcf\n",
      "     TSEN54 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/filtered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       676 /      1303\n",
      "             TSEN54:      1062 /      1188\n",
      "              UNION:      1149 /      1864\n",
      "       INTERSECTION:       589 /       627\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> TSEN54:\n",
      "         common heterozygous variants:       589\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       475\n",
      "       non-singleton blocks in TSEN54:         1\n",
      "                 --> covered variants:       528\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       423\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       422\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.47%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.24%\n",
      "          Block-wise Hamming distance:         1\n",
      "      Block-wise Hamming distance [%]:     0.24%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       422\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.47%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.24%\n",
      "                     Hamming distance:         1\n",
      "                 Hamming distance [%]:     0.24%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "chr17\t75620008\t75620159\tBENCHMARK<-->TSEN54\n",
      "chr17\t75620159\t75621024\tBENCHMARK<-->TSEN54\n",
      "\n",
      "TSEN54 \t Done\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              209           0.0  \n",
       "1              NaN           NaN  \n",
       "2               83           0.0  \n",
       "3              603           0.0  \n",
       "4              416           0.0  \n",
       "5              769           0.0  \n",
       "6              NaN           NaN  \n",
       "7               99           0.0  \n",
       "8              562           0.0  \n",
       "9              614           0.0  \n",
       "10               7           0.0  \n",
       "11             544           2.0  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Run Switch Errors\n",
    "\n",
    "bm_roi = switch_errors(basis_df, \"2\", \"filtered\")\n",
    "sw_e = pd.read_csv(bm_roi, sep=\"\\t\")\n",
    "\n",
    "sw_e = sw_e[[\"dataset_name1\",\"all_switches\"]]\n",
    "sw_e.rename(columns={\"dataset_name1\": \"GENE\"}, inplace=True)\n",
    "switch_df = pd.merge(basis_df, sw_e, on=\"GENE\", how=\"left\")\n",
    "switch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "d54588aa-9567-4d19-ab10-4b3d9702186a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VCF filtered on ROI\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  \n",
       "0              231  \n",
       "1              399  \n",
       "2              102  \n",
       "3              663  \n",
       "4              462  \n",
       "5              893  \n",
       "6              NaN  \n",
       "7              106  \n",
       "8              633  \n",
       "9              949  \n",
       "10             120  \n",
       "11             604  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Run Ununfiltered version\n",
    "\n",
    "print(\"VCF filtered on ROI\")\n",
    "phased_hb_uf = hb_maker(ununfiltered_v)\n",
    "basis_df_uf = phased_GOI(goi_file, phased_hb_uf)\n",
    "\n",
    "basis_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "297aacd2-6b34-4a18-9ea5-f753aaceb567",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CRTAP_BM.vcf\n",
      "      CRTAP = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr3 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       264 /       424\n",
      "              CRTAP:       907 /      1394\n",
      "              UNION:       913 /      1402\n",
      "       INTERSECTION:       258 /       416\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CRTAP:\n",
      "         common heterozygous variants:       258\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        99\n",
      "        non-singleton blocks in CRTAP:         1\n",
      "                 --> covered variants:       222\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        89\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        88\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        88\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CRTAP \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/SMN1_BM.vcf\n",
      "       SMN1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr5 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:        74 /       108\n",
      "               SMN1:      1931 /      2496\n",
      "              UNION:      1934 /      2499\n",
      "       INTERSECTION:        71 /       105\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> SMN1:\n",
      "         common heterozygous variants:        71\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:         7\n",
      "         non-singleton blocks in SMN1:         1\n",
      "                 --> covered variants:        64\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:         6\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:         5\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:         5\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "SMN1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CYP21A2_BM.vcf\n",
      "    CYP21A2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       118 /       187\n",
      "            CYP21A2:      4048 /      7613\n",
      "              UNION:      4054 /      7620\n",
      "       INTERSECTION:       112 /       180\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CYP21A2:\n",
      "         common heterozygous variants:       112\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       112\n",
      "      non-singleton blocks in CYP21A2:         1\n",
      "                 --> covered variants:       102\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       102\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       101\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       101\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CYP21A2 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/PEX7_BM.vcf\n",
      "       PEX7 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr6 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       731 /      1151\n",
      "               PEX7:      4048 /      7613\n",
      "              UNION:      4063 /      7629\n",
      "       INTERSECTION:       716 /      1135\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> PEX7:\n",
      "         common heterozygous variants:       716\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       639\n",
      "         non-singleton blocks in PEX7:         1\n",
      "                 --> covered variants:       635\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       570\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       569\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       569\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "PEX7 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr7\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/CFTR_BM.vcf\n",
      "       CFTR = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr7 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       514 /       707\n",
      "               CFTR:       727 /      1062\n",
      "              UNION:       738 /      1082\n",
      "       INTERSECTION:       503 /       687\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> CFTR:\n",
      "         common heterozygous variants:       503\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       455\n",
      "         non-singleton blocks in CFTR:         1\n",
      "                 --> covered variants:       431\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       401\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       400\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       400\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "CFTR \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr9\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/MUSK_BM.vcf\n",
      "       MUSK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr9 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:      1005 /      1346\n",
      "               MUSK:      1436 /      1979\n",
      "              UNION:      1466 /      2011\n",
      "       INTERSECTION:       975 /      1314\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> MUSK:\n",
      "         common heterozygous variants:       975\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       743\n",
      "         non-singleton blocks in MUSK:         1\n",
      "                 --> covered variants:       834\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       647\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       646\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       646\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "MUSK \t Done\n",
      "HBB \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA2_BM.vcf\n",
      "      BRCA2 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr13 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       119 /       318\n",
      "              BRCA2:       428 /      1167\n",
      "              UNION:       430 /      1175\n",
      "       INTERSECTION:       117 /       310\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA2:\n",
      "         common heterozygous variants:       117\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:        93\n",
      "        non-singleton blocks in BRCA2:         1\n",
      "                 --> covered variants:       105\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        84\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        83\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        83\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "BRCA2 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr15\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/POLG_BM.vcf\n",
      "       POLG = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr15 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       666 /       985\n",
      "               POLG:       880 /      1774\n",
      "              UNION:       895 /      1796\n",
      "       INTERSECTION:       651 /       963\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> POLG:\n",
      "         common heterozygous variants:       651\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       528\n",
      "         non-singleton blocks in POLG:         1\n",
      "                 --> covered variants:       574\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       471\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       470\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       470\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "POLG \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr16\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/HBA1_BM.vcf\n",
      "       HBA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr16 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       544 /       992\n",
      "               HBA1:      1111 /      1897\n",
      "              UNION:      1136 /      1928\n",
      "       INTERSECTION:       519 /       961\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> HBA1:\n",
      "         common heterozygous variants:       519\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       354\n",
      "         non-singleton blocks in HBA1:         1\n",
      "                 --> covered variants:       464\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       320\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       319\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       319\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "HBA1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/BRCA1_BM.vcf\n",
      "      BRCA1 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       119 /       361\n",
      "              BRCA1:      1392 /      3033\n",
      "              UNION:      1396 /      3043\n",
      "       INTERSECTION:       115 /       351\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> BRCA1:\n",
      "         common heterozygous variants:       115\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       101\n",
      "        non-singleton blocks in BRCA1:         1\n",
      "                 --> covered variants:       100\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:        88\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:        87\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "          Block-wise Hamming distance:         0\n",
      "      Block-wise Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:        87\n",
      "                        switch errors:         0\n",
      "                    switch error rate:     0.00%\n",
      "            switch/flip decomposition:       0/0\n",
      "                     switch/flip rate:     0.00%\n",
      "                     Hamming distance:         0\n",
      "                 Hamming distance [%]:     0.00%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "\n",
      "BRCA1 \t Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf'\n",
      "Reading phasing from '/hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz'\n",
      "Chromosomes present in all VCFs: chr17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparing phasings for sample hg002\n",
      "FILENAMES\n",
      "  BENCHMARK = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/TSEN54_BM.vcf\n",
      "     TSEN54 = /hpc/umc_laat/gvandersluis/data/Ont_data_nhung/HG002/un_unfiltered_ROI/phased_ROI.vcf.gz\n",
      "---------------- Chromosome chr17 ----------------\n",
      "VARIANT COUNTS (heterozygous / all): \n",
      "          BENCHMARK:       679 /      1307\n",
      "             TSEN54:      1392 /      3033\n",
      "              UNION:      1412 /      3070\n",
      "       INTERSECTION:       659 /      1270\n",
      "PAIRWISE COMPARISON: BENCHMARK <--> TSEN54:\n",
      "         common heterozygous variants:       659\n",
      "         (restricting to these below)\n",
      "    non-singleton blocks in BENCHMARK:         1\n",
      "                 --> covered variants:       521\n",
      "       non-singleton blocks in TSEN54:         1\n",
      "                 --> covered variants:       569\n",
      "    non-singleton intersection blocks:         1\n",
      "                 --> covered variants:       452\n",
      "              ALL INTERSECTION BLOCKS: ---------\n",
      "    phased pairs of variants assessed:       451\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.44%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.22%\n",
      "          Block-wise Hamming distance:         1\n",
      "      Block-wise Hamming distance [%]:     0.22%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "           LARGEST INTERSECTION BLOCK: ---------\n",
      "    phased pairs of variants assessed:       451\n",
      "                        switch errors:         2\n",
      "                    switch error rate:     0.44%\n",
      "            switch/flip decomposition:       0/1\n",
      "                     switch/flip rate:     0.22%\n",
      "                     Hamming distance:         1\n",
      "                 Hamming distance [%]:     0.22%\n",
      "                  Different genotypes:         0\n",
      "              Different genotypes [%]:     0.00%\n",
      "HII\n",
      "chr17\t75620008\t75620159\tBENCHMARK<-->TSEN54\n",
      "chr17\t75620159\t75621024\tBENCHMARK<-->TSEN54\n",
      "\n",
      "TSEN54 \t Done\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              231           0.0  \n",
       "1              399           0.0  \n",
       "2              102           0.0  \n",
       "3              663           0.0  \n",
       "4              462           0.0  \n",
       "5              893           0.0  \n",
       "6              NaN           NaN  \n",
       "7              106           0.0  \n",
       "8              633           0.0  \n",
       "9              949           0.0  \n",
       "10             120           0.0  \n",
       "11             604           2.0  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bm_roi_uf = switch_errors(basis_df_uf, \"2\", \"un_unfiltered\")\n",
    "\n",
    "sw_e_uf = pd.read_csv(bm_roi_uf, sep=\"\\t\")\n",
    "\n",
    "sw_e_uf = sw_e_uf[[\"dataset_name1\",\"all_switches\"]]\n",
    "sw_e_uf.rename(columns={\"dataset_name1\": \"GENE\"}, inplace=True)\n",
    "\n",
    "switch_df_uf = pd.merge(basis_df_uf, sw_e_uf, on=\"GENE\", how=\"left\")\n",
    "switch_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "f66ed712-dc40-48a1-a527-3a05f9eff80b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>231</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>70929044</td>\n",
       "      <td>71437569</td>\n",
       "      <td>508525</td>\n",
       "      <td>399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>31991875</td>\n",
       "      <td>32137224</td>\n",
       "      <td>145349</td>\n",
       "      <td>102</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>663</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>462</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110239493</td>\n",
       "      <td>110955599</td>\n",
       "      <td>716106</td>\n",
       "      <td>893</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32430770</td>\n",
       "      <td>250239</td>\n",
       "      <td>106</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>633</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>17889</td>\n",
       "      <td>648721</td>\n",
       "      <td>630832</td>\n",
       "      <td>949</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>42607430</td>\n",
       "      <td>43132602</td>\n",
       "      <td>525172</td>\n",
       "      <td>120</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75907509</td>\n",
       "      <td>884698</td>\n",
       "      <td>604</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486   70929044   71437569    508525   \n",
       "2   CYP21A2    chr6:31540309-32540309   31991875   32137224    145349   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110239493  110955599    716106   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32430770    250239   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      17889     648721    630832   \n",
       "10    BRCA1   chr17:42607268-43607268   42607430   43132602    525172   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75907509    884698   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              231           0.0  \n",
       "1              399           0.0  \n",
       "2              102           0.0  \n",
       "3              663           0.0  \n",
       "4              462           0.0  \n",
       "5              893           0.0  \n",
       "6              NaN           NaN  \n",
       "7              106           0.0  \n",
       "8              633           0.0  \n",
       "9              949           0.0  \n",
       "10             120           0.0  \n",
       "11             604           2.0  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# int_cols = [\"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS\"]\n",
    "\n",
    "# for c in int_cols:\n",
    "#     switch_df_uf[c] = pd.to_numeric(switch_df_uf[c], errors=\"coerce\").astype(\"Int64\")\n",
    "# switch_df_uf = switch_df_uf.astype(str).replace(\"<NA>\", \"NaN\")\n",
    "\n",
    "\n",
    "switch_df_uf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0ac4e50a-4a38-4ca0-9fc2-5f5716d0afa0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS  all_switches  \n",
       "0              209           0.0  \n",
       "1              NaN           NaN  \n",
       "2               83           0.0  \n",
       "3              603           0.0  \n",
       "4              416           0.0  \n",
       "5              769           0.0  \n",
       "6              NaN           NaN  \n",
       "7               99           0.0  \n",
       "8              562           0.0  \n",
       "9              614           0.0  \n",
       "10               7           0.0  \n",
       "11             544           2.0  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "switch_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b781766c-3dd0-4035-9d70-0be76a15d8b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>POS_START</th>\n",
       "      <th>POS_END</th>\n",
       "      <th>HB_LENGTH</th>\n",
       "      <th>PHASED_VARIANTS</th>\n",
       "      <th>FILTERED_VARIANTS</th>\n",
       "      <th>all_switches</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRTAP</td>\n",
       "      <td>chr3:32630864-33630864</td>\n",
       "      <td>32934214</td>\n",
       "      <td>33148829</td>\n",
       "      <td>214615</td>\n",
       "      <td>209</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMN1</td>\n",
       "      <td>chr5:70439486-71439486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>chr6:31540309-32540309</td>\n",
       "      <td>32000387</td>\n",
       "      <td>32137224</td>\n",
       "      <td>136837</td>\n",
       "      <td>83</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PEX7</td>\n",
       "      <td>chr6:136368573-137368573</td>\n",
       "      <td>136636685</td>\n",
       "      <td>137361584</td>\n",
       "      <td>724899</td>\n",
       "      <td>603</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>chr7:117001545-118001545</td>\n",
       "      <td>117002433</td>\n",
       "      <td>117727876</td>\n",
       "      <td>725443</td>\n",
       "      <td>416</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MUSK</td>\n",
       "      <td>chr9:110237668-111237668</td>\n",
       "      <td>110240520</td>\n",
       "      <td>110955599</td>\n",
       "      <td>715079</td>\n",
       "      <td>769</td>\n",
       "      <td>124</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HBB</td>\n",
       "      <td>chr11:4727429-5727429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>chr13:31857677-32857677</td>\n",
       "      <td>32180531</td>\n",
       "      <td>32415657</td>\n",
       "      <td>235126</td>\n",
       "      <td>99</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLG</td>\n",
       "      <td>chr15:88820029-89820029</td>\n",
       "      <td>88820301</td>\n",
       "      <td>89569212</td>\n",
       "      <td>748911</td>\n",
       "      <td>562</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HBA1</td>\n",
       "      <td>chr16:1-677101</td>\n",
       "      <td>27803</td>\n",
       "      <td>646952</td>\n",
       "      <td>619149</td>\n",
       "      <td>614</td>\n",
       "      <td>335</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>chr17:42607268-43607268</td>\n",
       "      <td>43010006</td>\n",
       "      <td>43132602</td>\n",
       "      <td>122596</td>\n",
       "      <td>7</td>\n",
       "      <td>113</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TSEN54</td>\n",
       "      <td>chr17:75020340-76020340</td>\n",
       "      <td>75022811</td>\n",
       "      <td>75870128</td>\n",
       "      <td>847317</td>\n",
       "      <td>544</td>\n",
       "      <td>60</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE                  POSITION  POS_START    POS_END HB_LENGTH  \\\n",
       "0     CRTAP    chr3:32630864-33630864   32934214   33148829    214615   \n",
       "1      SMN1    chr5:70439486-71439486        NaN        NaN       NaN   \n",
       "2   CYP21A2    chr6:31540309-32540309   32000387   32137224    136837   \n",
       "3      PEX7  chr6:136368573-137368573  136636685  137361584    724899   \n",
       "4      CFTR  chr7:117001545-118001545  117002433  117727876    725443   \n",
       "5      MUSK  chr9:110237668-111237668  110240520  110955599    715079   \n",
       "6       HBB     chr11:4727429-5727429        NaN        NaN       NaN   \n",
       "7     BRCA2   chr13:31857677-32857677   32180531   32415657    235126   \n",
       "8      POLG   chr15:88820029-89820029   88820301   89569212    748911   \n",
       "9      HBA1            chr16:1-677101      27803     646952    619149   \n",
       "10    BRCA1   chr17:42607268-43607268   43010006   43132602    122596   \n",
       "11   TSEN54   chr17:75020340-76020340   75022811   75870128    847317   \n",
       "\n",
       "   PHASED_VARIANTS FILTERED_VARIANTS all_switches  \n",
       "0              209                22            0  \n",
       "1              NaN               NaN          NaN  \n",
       "2               83                19            0  \n",
       "3              603                60            0  \n",
       "4              416                46            0  \n",
       "5              769               124            0  \n",
       "6              NaN               NaN          NaN  \n",
       "7               99                 7            0  \n",
       "8              562                71            0  \n",
       "9              614               335            0  \n",
       "10               7               113            0  \n",
       "11             544                60            2  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df = pd.merge(switch_df, switch_df_uf[[\"POSITION\",\"PHASED_VARIANTS\"]], how=\"left\", on=\"POSITION\")\n",
    "final_df[\"FILTERED_VARIANTS\"] = (pd.to_numeric(final_df[\"PHASED_VARIANTS_y\"], errors=\"coerce\") - pd.to_numeric(final_df[\"PHASED_VARIANTS_x\"], errors=\"coerce\"))\n",
    "final_df = final_df[[\"GENE\", \"POSITION\", \"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS_x\", \"FILTERED_VARIANTS\", \"all_switches\"]]\n",
    "final_df.rename(columns={\"PHASED_VARIANTS_x\": \"PHASED_VARIANTS\"}, inplace=True)\n",
    "\n",
    "int_cols = [\"POS_START\", \"POS_END\", \"HB_LENGTH\", \"PHASED_VARIANTS\", \"FILTERED_VARIANTS\", \"all_switches\"]\n",
    "\n",
    "for c in int_cols:\n",
    "    final_df[c] = pd.to_numeric(final_df[c], errors=\"coerce\").astype(\"Int64\")\n",
    "final_df = final_df.astype(str).replace(\"<NA>\", \"NaN\")\n",
    "\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "30690ba1-241f-4bd7-9f7f-a625d7401d15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dataset_name1</th>\n",
       "      <th>covered_variants</th>\n",
       "      <th>all_assessed_pairs</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>all_switch_rate</th>\n",
       "      <th>all_switchflips</th>\n",
       "      <th>all_switchflip_rate</th>\n",
       "      <th>blockwise_hamming</th>\n",
       "      <th>blockwise_hamming_rate</th>\n",
       "      <th>blockwise_diff_genotypes</th>\n",
       "      <th>blockwise_diff_genotypes_rate</th>\n",
       "      <th>largestblock_assessed_pairs</th>\n",
       "      <th>largestblock_switches</th>\n",
       "      <th>largestblock_switch_rate</th>\n",
       "      <th>largestblock_switchflips</th>\n",
       "      <th>largestblock_switchflip_rate</th>\n",
       "      <th>largestblock_hamming</th>\n",
       "      <th>largestblock_hamming_rate</th>\n",
       "      <th>het_variants0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3</td>\n",
       "      <td>CRTAP</td>\n",
       "      <td>83</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6</td>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>83</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>PEX7</td>\n",
       "      <td>534</td>\n",
       "      <td>533</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>533</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>376</td>\n",
       "      <td>375</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>375</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr9</td>\n",
       "      <td>MUSK</td>\n",
       "      <td>575</td>\n",
       "      <td>574</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>574</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr13</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>79</td>\n",
       "      <td>78</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>78</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr15</td>\n",
       "      <td>POLG</td>\n",
       "      <td>433</td>\n",
       "      <td>432</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>432</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr16</td>\n",
       "      <td>HBA1</td>\n",
       "      <td>282</td>\n",
       "      <td>281</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>281</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr17</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr17</td>\n",
       "      <td>TSEN54</td>\n",
       "      <td>423</td>\n",
       "      <td>422</td>\n",
       "      <td>2</td>\n",
       "      <td>0.004739</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.00237</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>422</td>\n",
       "      <td>2</td>\n",
       "      <td>0.004739</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.00237</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome dataset_name1  covered_variants  all_assessed_pairs  \\\n",
       "0       chr3         CRTAP                83                  82   \n",
       "1       chr6       CYP21A2                83                  82   \n",
       "2       chr6          PEX7               534                 533   \n",
       "3       chr7          CFTR               376                 375   \n",
       "4       chr9          MUSK               575                 574   \n",
       "5      chr13         BRCA2                79                  78   \n",
       "6      chr15          POLG               433                 432   \n",
       "7      chr16          HBA1               282                 281   \n",
       "8      chr17         BRCA1                 5                   4   \n",
       "9      chr17        TSEN54               423                 422   \n",
       "\n",
       "   all_switches  all_switch_rate all_switchflips  all_switchflip_rate  \\\n",
       "0             0         0.000000             0/0              0.00000   \n",
       "1             0         0.000000             0/0              0.00000   \n",
       "2             0         0.000000             0/0              0.00000   \n",
       "3             0         0.000000             0/0              0.00000   \n",
       "4             0         0.000000             0/0              0.00000   \n",
       "5             0         0.000000             0/0              0.00000   \n",
       "6             0         0.000000             0/0              0.00000   \n",
       "7             0         0.000000             0/0              0.00000   \n",
       "8             0         0.000000             0/0              0.00000   \n",
       "9             2         0.004739             0/1              0.00237   \n",
       "\n",
       "   blockwise_hamming  blockwise_hamming_rate  blockwise_diff_genotypes  \\\n",
       "0                  0                0.000000                         0   \n",
       "1                  0                0.000000                         0   \n",
       "2                  0                0.000000                         0   \n",
       "3                  0                0.000000                         0   \n",
       "4                  0                0.000000                         0   \n",
       "5                  0                0.000000                         0   \n",
       "6                  0                0.000000                         0   \n",
       "7                  0                0.000000                         0   \n",
       "8                  0                0.000000                         0   \n",
       "9                  1                0.002364                         0   \n",
       "\n",
       "   blockwise_diff_genotypes_rate  largestblock_assessed_pairs  \\\n",
       "0                            0.0                           82   \n",
       "1                            0.0                           82   \n",
       "2                            0.0                          533   \n",
       "3                            0.0                          375   \n",
       "4                            0.0                          574   \n",
       "5                            0.0                           78   \n",
       "6                            0.0                          432   \n",
       "7                            0.0                          281   \n",
       "8                            0.0                            4   \n",
       "9                            0.0                          422   \n",
       "\n",
       "   largestblock_switches  largestblock_switch_rate largestblock_switchflips  \\\n",
       "0                      0                  0.000000                      0/0   \n",
       "1                      0                  0.000000                      0/0   \n",
       "2                      0                  0.000000                      0/0   \n",
       "3                      0                  0.000000                      0/0   \n",
       "4                      0                  0.000000                      0/0   \n",
       "5                      0                  0.000000                      0/0   \n",
       "6                      0                  0.000000                      0/0   \n",
       "7                      0                  0.000000                      0/0   \n",
       "8                      0                  0.000000                      0/0   \n",
       "9                      2                  0.004739                      0/1   \n",
       "\n",
       "   largestblock_switchflip_rate  largestblock_hamming  \\\n",
       "0                       0.00000                     0   \n",
       "1                       0.00000                     0   \n",
       "2                       0.00000                     0   \n",
       "3                       0.00000                     0   \n",
       "4                       0.00000                     0   \n",
       "5                       0.00000                     0   \n",
       "6                       0.00000                     0   \n",
       "7                       0.00000                     0   \n",
       "8                       0.00000                     0   \n",
       "9                       0.00237                     1   \n",
       "\n",
       "   largestblock_hamming_rate  het_variants0  \n",
       "0                   0.000000            264  \n",
       "1                   0.000000            115  \n",
       "2                   0.000000            731  \n",
       "3                   0.000000            514  \n",
       "4                   0.000000           1004  \n",
       "5                   0.000000            118  \n",
       "6                   0.000000            666  \n",
       "7                   0.000000            543  \n",
       "8                   0.000000             12  \n",
       "9                   0.002364            676  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(bm_roi, sep=\"\\t\").drop([\"#sample\", \"dataset_name0\", \"file_name0\", \"intersection_blocks\",\"largestblock_diff_genotypes\",\"only_snvs\",\"largestblock_diff_genotypes_rate\",\"file_name1\"],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "128bbbcc-805f-4dd5-bcd6-38d3592d20fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>dataset_name1</th>\n",
       "      <th>covered_variants</th>\n",
       "      <th>all_assessed_pairs</th>\n",
       "      <th>all_switches</th>\n",
       "      <th>all_switch_rate</th>\n",
       "      <th>all_switchflips</th>\n",
       "      <th>all_switchflip_rate</th>\n",
       "      <th>blockwise_hamming</th>\n",
       "      <th>blockwise_hamming_rate</th>\n",
       "      <th>blockwise_diff_genotypes</th>\n",
       "      <th>blockwise_diff_genotypes_rate</th>\n",
       "      <th>largestblock_assessed_pairs</th>\n",
       "      <th>largestblock_switches</th>\n",
       "      <th>largestblock_switch_rate</th>\n",
       "      <th>largestblock_switchflips</th>\n",
       "      <th>largestblock_switchflip_rate</th>\n",
       "      <th>largestblock_hamming</th>\n",
       "      <th>largestblock_hamming_rate</th>\n",
       "      <th>het_variants0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr3</td>\n",
       "      <td>CRTAP</td>\n",
       "      <td>89</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>88</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>SMN1</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>CYP21A2</td>\n",
       "      <td>102</td>\n",
       "      <td>101</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>101</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6</td>\n",
       "      <td>PEX7</td>\n",
       "      <td>570</td>\n",
       "      <td>569</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>569</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr7</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>401</td>\n",
       "      <td>400</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>400</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr9</td>\n",
       "      <td>MUSK</td>\n",
       "      <td>647</td>\n",
       "      <td>646</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>646</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr13</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>84</td>\n",
       "      <td>83</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>83</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr15</td>\n",
       "      <td>POLG</td>\n",
       "      <td>471</td>\n",
       "      <td>470</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>470</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr16</td>\n",
       "      <td>HBA1</td>\n",
       "      <td>320</td>\n",
       "      <td>319</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>319</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr17</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>88</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>87</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr17</td>\n",
       "      <td>TSEN54</td>\n",
       "      <td>452</td>\n",
       "      <td>451</td>\n",
       "      <td>2</td>\n",
       "      <td>0.004435</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.002217</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002212</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>451</td>\n",
       "      <td>2</td>\n",
       "      <td>0.004435</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.002217</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002212</td>\n",
       "      <td>679</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome dataset_name1  covered_variants  all_assessed_pairs  \\\n",
       "0        chr3         CRTAP                89                  88   \n",
       "1        chr5          SMN1                 6                   5   \n",
       "2        chr6       CYP21A2               102                 101   \n",
       "3        chr6          PEX7               570                 569   \n",
       "4        chr7          CFTR               401                 400   \n",
       "5        chr9          MUSK               647                 646   \n",
       "6       chr13         BRCA2                84                  83   \n",
       "7       chr15          POLG               471                 470   \n",
       "8       chr16          HBA1               320                 319   \n",
       "9       chr17         BRCA1                88                  87   \n",
       "10      chr17        TSEN54               452                 451   \n",
       "\n",
       "    all_switches  all_switch_rate all_switchflips  all_switchflip_rate  \\\n",
       "0              0         0.000000             0/0             0.000000   \n",
       "1              0         0.000000             0/0             0.000000   \n",
       "2              0         0.000000             0/0             0.000000   \n",
       "3              0         0.000000             0/0             0.000000   \n",
       "4              0         0.000000             0/0             0.000000   \n",
       "5              0         0.000000             0/0             0.000000   \n",
       "6              0         0.000000             0/0             0.000000   \n",
       "7              0         0.000000             0/0             0.000000   \n",
       "8              0         0.000000             0/0             0.000000   \n",
       "9              0         0.000000             0/0             0.000000   \n",
       "10             2         0.004435             0/1             0.002217   \n",
       "\n",
       "    blockwise_hamming  blockwise_hamming_rate  blockwise_diff_genotypes  \\\n",
       "0                   0                0.000000                         0   \n",
       "1                   0                0.000000                         0   \n",
       "2                   0                0.000000                         0   \n",
       "3                   0                0.000000                         0   \n",
       "4                   0                0.000000                         0   \n",
       "5                   0                0.000000                         0   \n",
       "6                   0                0.000000                         0   \n",
       "7                   0                0.000000                         0   \n",
       "8                   0                0.000000                         0   \n",
       "9                   0                0.000000                         0   \n",
       "10                  1                0.002212                         0   \n",
       "\n",
       "    blockwise_diff_genotypes_rate  largestblock_assessed_pairs  \\\n",
       "0                             0.0                           88   \n",
       "1                             0.0                            5   \n",
       "2                             0.0                          101   \n",
       "3                             0.0                          569   \n",
       "4                             0.0                          400   \n",
       "5                             0.0                          646   \n",
       "6                             0.0                           83   \n",
       "7                             0.0                          470   \n",
       "8                             0.0                          319   \n",
       "9                             0.0                           87   \n",
       "10                            0.0                          451   \n",
       "\n",
       "    largestblock_switches  largestblock_switch_rate largestblock_switchflips  \\\n",
       "0                       0                  0.000000                      0/0   \n",
       "1                       0                  0.000000                      0/0   \n",
       "2                       0                  0.000000                      0/0   \n",
       "3                       0                  0.000000                      0/0   \n",
       "4                       0                  0.000000                      0/0   \n",
       "5                       0                  0.000000                      0/0   \n",
       "6                       0                  0.000000                      0/0   \n",
       "7                       0                  0.000000                      0/0   \n",
       "8                       0                  0.000000                      0/0   \n",
       "9                       0                  0.000000                      0/0   \n",
       "10                      2                  0.004435                      0/1   \n",
       "\n",
       "    largestblock_switchflip_rate  largestblock_hamming  \\\n",
       "0                       0.000000                     0   \n",
       "1                       0.000000                     0   \n",
       "2                       0.000000                     0   \n",
       "3                       0.000000                     0   \n",
       "4                       0.000000                     0   \n",
       "5                       0.000000                     0   \n",
       "6                       0.000000                     0   \n",
       "7                       0.000000                     0   \n",
       "8                       0.000000                     0   \n",
       "9                       0.000000                     0   \n",
       "10                      0.002217                     1   \n",
       "\n",
       "    largestblock_hamming_rate  het_variants0  \n",
       "0                    0.000000            264  \n",
       "1                    0.000000             74  \n",
       "2                    0.000000            118  \n",
       "3                    0.000000            731  \n",
       "4                    0.000000            514  \n",
       "5                    0.000000           1005  \n",
       "6                    0.000000            119  \n",
       "7                    0.000000            666  \n",
       "8                    0.000000            544  \n",
       "9                    0.000000            119  \n",
       "10                   0.002212            679  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(bm_roi_uf, sep=\"\\t\").drop([\"#sample\", \"dataset_name0\", \"file_name0\", \"intersection_blocks\",\"largestblock_diff_genotypes\",\"only_snvs\",\"largestblock_diff_genotypes_rate\",\"file_name1\"],axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "148fabd3-5e9c-485b-8484-95707f61a3d6",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (2934525902.py, line 4)",
     "output_type": "error",
     "traceback": [
      "  \u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 4\u001b[39m\n\u001b[31m    \u001b[39m\u001b[31mbcftools view -r 'chr3:32934214-33148829' phased_ROI.vcf.gz | grep -E '0\\|1|1\\|1|1\\|0|0\\|0' | wc\u001b[39m\n             ^\n\u001b[31mSyntaxError\u001b[39m\u001b[31m:\u001b[39m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "### Interesting Code Lines\n",
    "\n",
    "# find how many phased variant per selected region\n",
    "bcftools view -r 'chr3:32934214-33148829' phased_ROI.vcf.gz | grep -E '0\\|1|1\\|1|1\\|0|0\\|0' | wc\n",
    "\n",
    "bcftools view -H  -i 'FORMAT/PS=\".\"' -r 'chr17:75022811-75870128' phased_ROI.vcf.gz | wc\n",
    "bcftools view -H  -i 'FORMAT/GT!~\"/\"'  -r 'chr17:75022811-75870128' phased_ROI.vcf.gz | wc\n",
    "bcftools view -H  -i 'FORMAT/PS!=\".\"' -r 'chr17:75022811-75870128' phased_ROI.vcf.gz | wc\n",
    "bcftools view -H  -i 'FORMAT/GT~\"/\"'  -r 'chr17:75022811-75870128' phased_ROI.vcf.gz | wc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef0633cf-83c8-4a9e-a14f-73f68d231953",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:NIPD]",
   "language": "python",
   "name": "conda-env-NIPD-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
